<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Andrey Ziyatdinov, Helena Brunel" />

<meta name="date" content="2024-05-10" />

<title>Tutorial on solareclipser R package</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Tutorial on <code>solareclipser</code> R
package</h1>
<h4 class="author">Andrey Ziyatdinov, Helena Brunel</h4>
<h4 class="date">2024-05-10</h4>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(solareclipser)</span></code></pre></div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This vignette is a tutorial on the R package
<code>solareclipser</code>. The document contains a brief description of
the main statistical models (polygenic, association and linkage)
implemented in <code>SOLAR</code> and accessible via
<code>solareclipser</code>, installation instructions for both
<code>SOLAR</code> and <code>solareclipser</code>, reproducible examples
on synthetic data sets available within the <code>solareclipser</code>
package. The examples are intended to cover all the necessary steps in
the quantitative genetic analysis with <code>SOLAR</code>: data loading,
data exploration, running statistical models and interpreting the
obtained results by printing, summarizing and plotting the models in
R.</p>
<p>In human genetics research, the study of a complex disorder is
targeted to discover its underlying genetic architecture. Whereas simple
Mendelian diseases are results of a mutation in a single gene, for
example <a href="http://en.wikipedia.org/wiki/Huntington%27s_disease">Huntingtonâ€™s
disease</a>, complex disorders or phenotypes are the product of the
action of multiple genes and environmental factors as well as their
interactions. For instance, obesity is a complex disease which genetic
basis might be better dissected via measurable quantitative variation in
phenotypes, such as body-mass index, closely related to the disease
risk.</p>
<p>One of the study designs to identify the quantitative trait loci
(QTLs) that affect complex phenotypes is based on examination of related
individuals grouped by families (pedigrees). Pedigree-based designs
allow for both QTL linkage and association mappings. In addition, QTL
analysis with pedigrees possess more power to detect rare variants,
permits explicit control for population stratification in association
studies, and suit for examination of parent-of-origin and
sex-specificity effects.</p>
<p>The analysis of the pedigree data is traditionally based on variance
component (VC) models, also known as linear mixed models.
<code>SOLAR</code> is one of the well-known and widely used tools that
implements the variance component models and other routines needed for
the QTL analysis in extended pedigrees <span class="citation">[@Almasy1998]</span> (<a href="http://solar.txbiomedgenetics.org/">http://solar.txbiomedgenetics.org/</a>).
The name <code>SOLAR</code> stands for Sequential Oligogenic Linkage
Analysis Routines. The analyses available in <code>SOLAR</code> include
customized quantitative genetic analysis, advanced linkage analysis, SNP
association analysis (QTN, QTLD, and MGA). Other operations are
implemented for calculation of marker-specific or multipoint
identity-by-descent (IBD) matrices in pedigrees of arbitrary size and
complexity. The linkage analysis allows for multiple quantitative traits
and/or discrete traits which might involve multiple loci (oligogenic
analysis), dominance effects, household effects, and interactions.</p>
<p>There are other tools similar to <code>SOLAR</code>, such as
<code>Mendel</code>, <code>MMM</code>, <code>GEMMA</code> and
<code>FAST-LMM</code>, but the R package <code>solareclipser</code>
presented here has been especially developed to work with
<code>SOLAR</code>.</p>
<p><code>solareclipser</code> provides an interface from the program
<code>SOLAR</code> to the environment for statistical computing
<code>R</code>. On one side, main <code>SOLAR</code> procedures can be
called in <code>R</code> by one or a few commands from the
<code>solareclipser</code> package, that makes the analysis more
user-friendly with the same power of computations implemented in
<code>SOLAR</code>. On the other side, the <code>solareclipser</code>
user benefits from the infrastructure available with the R environment
and its packages, such as interactive data manipulation, data
visualization and parallel computation.</p>
</div>
<div id="methodology" class="section level1">
<h1>Methodology</h1>
<div id="linear-mixed-model" class="section level2">
<h2>Linear mixed model</h2>
<p>Consider that a particular trait, such as body-mass index, is
observed in <span class="math inline">\(n\)</span> individuals grouped
by families or clusters of related individuals. A <span class="math inline">\(n \times 1\)</span> vector <span class="math inline">\(y\)</span> contains the phenotypic values measured
for the trait. One assumes that these observations are described
adequately by a linear model with a <span class="math inline">\(p \times
1\)</span> vector of fixed effects (<span class="math inline">\(\beta\)</span>) and a <span class="math inline">\(q \times 1\)</span> vector of random effects
(<span class="math inline">\(u\)</span>) <span class="citation">[@Lynch1998, Chapter 26, pp. 746]</span>. In matrix
form,</p>
<p><span class="math display">\[ y = X \beta + Z u + e \]</span></p>
<p>where <span class="math inline">\(X\)</span> and <span class="math inline">\(Z\)</span> are respectively <span class="math inline">\(n \times p\)</span> and <span class="math inline">\(n \times q\)</span> incidence matrices, and <span class="math inline">\(e\)</span> is a <span class="math inline">\(n
\times 1\)</span> vector of residual deviations assumed to be
distributed independently of the random effects. Denote the <span class="math inline">\(n \times n\)</span> covariance matrix for the
vector <span class="math inline">\(e\)</span> by <span class="math inline">\(R\)</span> and the the <span class="math inline">\(q \times q\)</span> covariance matrix for the
vector <span class="math inline">\(u\)</span> by <span class="math inline">\(G\)</span>.</p>
<p>The first element of the vector <span class="math inline">\(\beta\)</span> is typically the population mean,
and other elements may be age, gender, diet and so on. The elements of
the incidence matrices are usually equal to 0 or 1, that indicate the
effect contribution to the individualsâ€™s phenotype. The model is
referred to as a <em>mixed model</em>, because this model jointly
accounts for fixed and random effects.</p>
<p>Giving the two terms of the random effects <span class="math inline">\(u\)</span> and <span class="math inline">\(e\)</span>, the first term typically accounts for
the contribution from random genetic effects such as additive genetic
values. The second term accounts for the residual variance, and one
usually assumes that residual errors have constant variance and are
uncorrelated, so that <span class="math inline">\(R\)</span> is a
diagonal matrix, with <span class="math inline">\(R = \sigma_e^2
I\)</span>.</p>
<p>Two more equations for the means and variances of the component
vectors of the mixed model can be derived on the basis of its
definition. Since <span class="math inline">\(E(u) = E(e) = 0\)</span>
by definition,</p>
<p><span class="math display">\[ E(y) = X \beta \]</span></p>
<p>Excluding the difference among individuals due to fixed effects and
calling the assumption that <span class="math inline">\(u\)</span> and
<span class="math inline">\(e\)</span> are uncorrelated, the covariance
matrix for the vector of observations <span class="math inline">\(y\)</span> is</p>
<p><span class="math display">\[ Var(y) = V = Z G Z^{T} + R
\]</span></p>
<p>Now the mixed model can be rewritten in a compact form</p>
<p><span class="math display">\[ y = X \beta + Z u + e\]</span></p>
<p><span class="math display">\[
\mbox{ where } \mbox{ } \mbox{ } u \sim (0, G) \mbox{ and } e \sim (0,
R) \\
\mbox{ implying } \mbox{ } \mbox{ } y \sim (X \beta, V) = (X \beta, Z G
Z^T + R)
\]</span></p>
<p>For the mixed model, <span class="math inline">\(y\)</span>, <span class="math inline">\(X\)</span> and <span class="math inline">\(Z\)</span> are observed variables, while <span class="math inline">\(\beta\)</span>, <span class="math inline">\(u\)</span>, <span class="math inline">\(R\)</span>
and <span class="math inline">\(G\)</span> are generally unknown
variables. Thus, the analysis consists of two complementary estimation
procedures: (1) estimation of the covariance matrices <span class="math inline">\(G\)</span> and <span class="math inline">\(R\)</span>, and (2) estimation of the vectors of
fixed and random effects, <span class="math inline">\(\beta\)</span> and
<span class="math inline">\(u\)</span>. The covariance matrices are
usually functions of a few unknown parameters or variance components.
For instance, the covariance matrix of the residuals errors typically
has the form <span class="math inline">\(R = \sigma_e^2 I\)</span>,
where <span class="math inline">\(\sigma_e^2\)</span> is the parameter
to be estimated.</p>
<p>The description of the two estimation procedures is out of scope of
this document, and the reader is referred to <span class="citation">[@Lynch1998, Chapter 26-27]</span>, where the necessary
theoretical material and accompanying examples are given. On the other
hand, the implementation of the estimation procedures in
<code>SOLAR</code> is discussed elsewhere <span class="citation">[@Almasy1998]</span>.</p>
<p>The next three sections will present <code>SOLAR</code> models, which
follow the general definition of linear mixed model given above, but
parametrization of the models and estimation procedures are implemented
in a <code>SOLAR</code>-specific way.</p>
</div>
<div id="polygenic-model-in-solar" class="section level2">
<h2>Polygenic model in <code>SOLAR</code></h2>
<p>The polygenic model is a basis for any quantitative genetic analysis
in <code>SOLAR</code>. The model includes fixed effects (<span class="math inline">\(\beta\)</span>) and three random effects (the
genetic additive effect <span class="math inline">\(g\)</span>, the
house-hold effect <span class="math inline">\(c\)</span>, and the
residual effect <span class="math inline">\(e\)</span>).</p>
<p><span class="math display">\[y = X\beta + g + c + e\]</span></p>
<p><span class="math display">\[\mbox{ where } \mbox{ } \mbox{ } g \sim
(0, \sigma_g^2 A)
  \mbox{ and } \mbox{ } \mbox{ } c \sim (0, \sigma_c^2 H)
  \mbox{ and } \mbox{ } \mbox{ } e \sim (0, \sigma_e^2 I)\]</span></p>
<p><span class="math display">\[ \mbox{ implying } \mbox{ } \mbox{ } y
\sim (X\beta, V) = (X\beta, \sigma_g^2 A + \sigma_c^2 H + \sigma_e^2
I)\]</span></p>
<p>The first <span class="math inline">\(g\)</span> term accounts for
the additive genetic effect with the covariance matrix <span class="math inline">\(A = 2 K\)</span>, where <span class="math inline">\(K\)</span> is the kinship matrix.
<code>SOLAR</code> estimates the <em>empirical</em> kinship matrix based
on the identifier (ID) fields in the pedigree data, such as individual
ID <code>id</code>, gender <code>sex</code>, family ID <code>fam</code>,
father ID <code>fa</code>, mother ID <code>mo</code>, and indicator of
identical individuals, e.g.Â monozygotic twins, <code>mztwin</code> <span class="citation">[@SOLARDOC, Appendix 1,
[file-pedigree](http://helix.nih.gov/Documentation/solar-6.6.2-doc/91.appendix_1_text.html#file-pedigree)
command]</span>.</p>
<p>The second <span class="math inline">\(c\)</span> term represents an
annotated environmental effect. It is named as a house-hold effect, but
it is generally related to the sharing of any non-genetic factor among
individuals. The house-hold effect is included in the analysis if the
pedigree data contain a house-hold ID field (the expected field name is
<code>hhid</code>). <code>SOLAR</code> automatically creates
<code>Pedigree-Household Groups</code> based on the <code>hhid</code>
identifiers. That means that pedigrees and household groups are merged
into larger groups, which contain every individual in the same pedigree
as well as every individual that has the same <code>hhid</code> as any
individual in the pedigree <span class="citation">[@SOLARDOC, Section
9.3 [Household Group
Analysis](http://helix.nih.gov/Documentation/solar-6.6.2-doc/09.chapter.html#household)]</span>.
As a consequence of this grouping operation by <code>SOLAR</code>, the
covariance matrix <span class="math inline">\(H\)</span> of the <span class="math inline">\(c\)</span> random effect contains the information
related to the pedigree-household groups rather than relationships given
by the <code>hhid</code> field.</p>
<p>The third <span class="math inline">\(e\)</span> term is the residual
error with the diagonal covariance matrix. It is important to note that
the <code>residual</code> command in <code>SOLAR</code> <span class="citation">[@SOLARDOC, Appendix 1,
[residual](http://helix.nih.gov/Documentation/solar-6.6.2-doc/91.appendix_1_text.html#residual)
command]</span> returns the covariate-free value of a trait under study,
expressed as <span class="math inline">\(y - X \beta\)</span>, and the
returned value is not the residual vector <span class="math inline">\(e\)</span> given in the equation of the polygenic
model.</p>
<p>The polygenic term <span class="math inline">\(g\)</span> represents
the additive genetic effect by default. The incorporation of the
dominance effect into <code>SOLAR</code> models is also possible, but it
is attributed to advanced modeling topics <span class="citation">[@SOLARDOC, Section 9.4 [Dominance
Analysis](http://helix.nih.gov/Documentation/solar-6.6.2-doc/09.chapter.html#dominance)]</span>.
The <code>SOLAR</code>â€™s developers suggest that the dominance variance
component is unlikely useful in quantitative trait mapping, particularly
in human genetic analysis, as only bi-lineal relatives contribute to
this component.</p>
<p><code>SOLAR</code> is able to work with discrete traits, which
possible values are coded as two consecutive integers, for example,
<code>0</code> and <code>1</code> <span class="citation">[@SOLARDOC,
Appendix 1,
[discrete-notes](http://helix.nih.gov/Documentation/solar-6.6.2-doc/91.appendix_1_text.html#discrete-notes)]</span>.
<code>SOLAR</code> uses a <em>liability threshold model</em> to handle
discrete traits <span class="citation">[@Duggirala1997]</span>, and more
details about implementation and practical notes are available in <span class="citation">[@SOLARDOC, Section 9.1 [Discrete
Traits](http://helix.nih.gov/Documentation/solar-6.6.2-doc/09.chapter.html)]</span>.
<code>SOLAR</code> authors recommend to work with quantitative traits
whenever it is possible. The user is able to disable the transformation
of discrete traits, if it is necessary.</p>
<p>The likelihood-based statistics, in particular Likelihood Ratio Tests
(LRTs), is used for the inference of the linear mixed models implemented
in <code>SOLAR</code>. The following table shows the alternative models
used to test the significance of the random effects <span class="math inline">\(g\)</span> and <span class="math inline">\(c\)</span>.</p>
<table>
<colgroup>
<col width="35%" />
<col width="39%" />
<col width="25%" />
</colgroup>
<thead>
<tr class="header">
<th>Alternative model</th>
<th>Free parameters</th>
<th>Restricted parameters</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>No polygenic effect</td>
<td><span class="math inline">\(\beta\)</span>, <span class="math inline">\(\sigma_c^2\)</span>, <span class="math inline">\(\sigma_e^2\)</span></td>
<td><span class="math inline">\(\sigma_g^2\)</span> = 0</td>
</tr>
<tr class="even">
<td>No house-hold effect</td>
<td><span class="math inline">\(\beta\)</span>, <span class="math inline">\(\sigma_g^2\)</span>, <span class="math inline">\(\sigma_e^2\)</span></td>
<td><span class="math inline">\(\sigma_c^2\)</span> = 0</td>
</tr>
</tbody>
</table>
<p>Under the assumption of multivariate normality, the LRT statistic is
asymptotically distributed as a <span class="math inline">\(\frac{1}{2}:\frac{1}{2}\)</span> mixture of of a
<span class="math inline">\(\chi_1^2\)</span> variable and a point mass
at the origin <span class="citation">[@Blangero2000]</span>. This
statement is applied to any LRT performed for parameters like <span class="math inline">\(\sigma_g^2\)</span>, which is greater or equal to
<code>0</code> by definition.</p>
<p><code>SOLAR</code> takes the decision of rejecting the house-hold
effect by considering two factors, the test results and the value of
<span class="math inline">\(\sigma_c^2\)</span> parameter. If value of
<span class="math inline">\(\sigma_c^2\)</span> is non-zero, the
house-hold effect is retained in the polygenic model, nevertheless the
effect is not significant. The user has an option to force the rejection
of the house-hold effect if it is necessary.</p>
<p>The tests applied to fixed effects (covariates) are also performed by
means of LRTs. In the table given below the alternative model for the
test of significance of <code>age</code> covariate is shown.</p>
<table>
<colgroup>
<col width="35%" />
<col width="39%" />
<col width="25%" />
</colgroup>
<thead>
<tr class="header">
<th>Alternative model</th>
<th>Free parameters</th>
<th>Restricted parameters</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>No age effect</td>
<td><span class="math inline">\(\sigma_g^2\)</span>, <span class="math inline">\(\sigma_c^2\)</span>, <span class="math inline">\(\sigma_e^2\)</span></td>
<td><span class="math inline">\(\beta_{age}\)</span> = 0</td>
</tr>
</tbody>
</table>
<div id="multivariate-polygenic-model-in-solar" class="section level3">
<h3>Multivariate polygenic model in <code>SOLAR</code></h3>
<p>In principle, one can estimate the multivariate polygenic model by
simply performing the univariate analysis in an appropriate way.
Consider two traits <span class="math inline">\(y_1\)</span> and <span class="math inline">\(y_2\)</span> measured in each of <span class="math inline">\(n\)</span> individuals, assuming that each trait
follows the same univariate polygenic analysis. Then the <span class="math inline">\(2 \times n\)</span> dimensional column vector of
observations is denoted by the stack of univariate vectors, and the
polygenic model has the matrix form:</p>
<p><span class="math display">\[
\begin{pmatrix}
  y_1 \\
  y_2
\end{pmatrix}_{2n \times 1} =
\begin{pmatrix}
  X_1 &amp; 0 \\
  0 &amp; X_2
\end{pmatrix}_{2n \times nk}
\begin{pmatrix}
  \beta_1 \\
  \beta_2
\end{pmatrix}_{2n \times 1}
+
\begin{pmatrix}
  g_1 \\
  g_2
\end{pmatrix}_{2n \times 1}
+
\begin{pmatrix}
  e_1 \\
  e_2
\end{pmatrix}_{2n \times 1}
\]</span></p>
<p><span class="math display">\[
G_{2n \times 2n} = C_{2 \times 2} \otimes A_{n \times n} =
\begin{pmatrix}
  c_{11} A &amp; c_{12} A \\
  c_{21} A &amp; c_{21} A
\end{pmatrix} =
\begin{pmatrix}
  \sigma_{1}^2 A &amp; \rho \sigma_{1} \sigma_{2} A \\
  \rho \sigma_{1} \sigma_{2} A &amp; \sigma_{2}^2 A
\end{pmatrix}_{g}
\]</span></p>
<p><span class="math display">\[
R_{2n \times 2n} = E_{2 \times 2} \otimes I_{n \times n} =
\begin{pmatrix}
  e_{11} I &amp; e_{12} I \\
  e_{21} I &amp; e_{21} I
\end{pmatrix} =
\begin{pmatrix}
  \sigma_{1}^2 I &amp; \rho \sigma_{1} \sigma_{2} I \\
  \rho \sigma_{1} \sigma_{2} I &amp; \sigma_{2}^2 I
\end{pmatrix}_{e}
\]</span></p>
<p>The model can be generalized to <span class="math inline">\(k\)</span> traits, and other random effects such
as house-hold can be added in the similar way. The <em>Kronecker
product</em> notation <span class="math inline">\(\otimes\)</span> is
convinient for denoting the covariance matrices. More details on the
multivariate analysis are given in <span class="citation">[@Lynch1998,
Chapter 26, Multiple Traits, pp. 774]</span>.</p>
<p>The multivariate analysis implemented in <code>SOLAR</code> allows
for any number of traits, but the tests for significance of the
correlation coefficients are performed only for the bivariate models.
The following table describes the main LRTs available in
<code>SOLAR</code> for the bivariate analysis.</p>
<table>
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th>Alternative model</th>
<th>Free parameters</th>
<th>Restricted parameters</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>No genetic correlation</td>
<td><span class="math inline">\((\sigma_1^2, \sigma_2^2)_g\)</span>,
<span class="math inline">\((\sigma_1^2, \sigma_2^2,
\rho)_e\)</span></td>
<td><span class="math inline">\(\rho_g = 0\)</span></td>
</tr>
<tr class="even">
<td>Complete pleiotropy</td>
<td><span class="math inline">\((\sigma_1^2, \sigma_2^2)_g\)</span>,
<span class="math inline">\((\sigma_1^2, \sigma_2^2,
\rho)_e\)</span></td>
<td><span class="math inline">\(\rho_g = 1\)</span></td>
</tr>
<tr class="odd">
<td>No environmental correlation</td>
<td><span class="math inline">\((\sigma_1^2, \sigma_2^2,
\rho)_g\)</span>, <span class="math inline">\((\sigma_1^2,
\sigma_2^2)_e\)</span></td>
<td><span class="math inline">\(\rho_e = 0\)</span></td>
</tr>
</tbody>
</table>
<p>It is important to note that the transformation of discrete traits in
the multivariate analysis is automatically performed by
<code>SOLAR</code>.</p>
</div>
</div>
<div id="linkage-model-in-solar" class="section level2">
<h2>Linkage model in <code>SOLAR</code></h2>
<p><code>SOLAR</code> introduces sequential oligogenic linkage analysis,
as stated in its name abbreviation. Consider the linkage model with a
single QTL, which is described by another random effect <span class="math inline">\(q\)</span>.</p>
<p><span class="math display">\[y = X\beta + g + c + q + e\]</span></p>
<p><span class="math display">\[
  \mbox{ where } \mbox{ } \mbox{ } g \sim (0, \sigma_g^2 A)
  \mbox{ and } \mbox{ } \mbox{ } c \sim (0, \sigma_c^2 H)
  \mbox{ and } \mbox{ } \mbox{ } q \sim (0, \sigma_q^2 M)
  \mbox{ and } \mbox{ } \mbox{ } e \sim (0, \sigma_e^2 I)
\]</span></p>
<p><span class="math display">\[ \mbox{ implying } \mbox{ } \mbox{ } y
\sim (X\beta, V) = (X\beta, \sigma_g^2 A + \sigma_c^2 H + \sigma_q^2 M +
\sigma_e^2)\]</span></p>
<p>The QTL under study <span class="math inline">\(q\)</span>
corresponds to a specific position in the genome and an
identity-by-descent (IBD) matrix. The genome-wide linkage analysis
assumes that the highly informative markers, such as micro-satellites,
are available in the study, in order to estimate the two-point or
multipoint IBD matrices by <code>SOLAR</code> or other specific program
like <code>MERLIN</code>. Once the IBD matrices are calculated, the
genome-wide scan is conducted by sequentially running the linkage model
for each genome position.</p>
<p>The alternative model for testing the significance of a single QTL
effect is the same as the polygenic model.</p>
<table>
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th>Alternative model</th>
<th>Free parameters</th>
<th>Restricted parameters</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>No linkage</td>
<td><span class="math inline">\(\beta, \sigma_g^2, \sigma_c^2\)</span>,
<span class="math inline">\(\sigma_e^2\)</span></td>
<td><span class="math inline">\(\sigma_q^2\)</span> = 0</td>
</tr>
</tbody>
</table>
<p>Traditionally, the logarithm (base 10) of the odds (LOD) values are
used to score the QTLs. If the LRT statistics for the linkage model is
denoted as <span class="math inline">\(\Lambda\)</span>, then the
correspondent LOD score is equal to <span class="math inline">\(\Lambda
/ (2 ln(10))\)</span>.</p>
<p><code>SOLAR</code> also allows for multi-pass linkage scan, that
means that any further scan can be conditioned to QTLs found in the
previous scans. For example, the second-pass scan conditioned to the
previously found QTL <span class="math inline">\(q_1\)</span> has the
form:</p>
<p><span class="math display">\[y = X\beta + g + c + q_1 + q +
e\]</span></p>
<p>The second-pass LRT test now looks like in the table below.</p>
<table>
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th>Alternative model</th>
<th>Free parameters</th>
<th>Restricted parameters</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>No linkage</td>
<td><span class="math inline">\(\beta, \sigma_g^2, \sigma_c^2\)</span>,
<span class="math inline">\(\sigma_{q_1}^2\)</span>, <span class="math inline">\(\sigma_e^2\)</span></td>
<td><span class="math inline">\(\sigma_q^2\)</span> = 0</td>
</tr>
</tbody>
</table>
<p>The linkage model is extended to deal with many traits or discrete
traits in the same way as for the polygenic model.</p>
<p>Additional information related to the linkage modeling in
<code>SOLAR</code> can be found in related publications <span class="citation">[@Almasy1998]</span>, <span class="citation">[@Almasy1997]</span>, <span class="citation">[@Williams1999]</span>, the <code>SOLAR</code> tutorial
<span class="citation">[@SOLARDOC, Chapter 3
[Tutorial](http://helix.nih.gov/Documentation/solar-6.6.2-doc/03.chapter.html)]</span>,
and the <code>SOLAR</code> manual, for example, <span class="citation">[@SOLARDOC, Apendix 1,
[multipoint](http://helix.nih.gov/Documentation/solar-6.6.2-doc/91.appendix_1_text.html#multipoint)
command]</span>.</p>
</div>
<div id="association-model-in-solar" class="section level2">
<h2>Association model in <code>SOLAR</code></h2>
<p>A basic association model in <code>SOLAR</code> is extended from the
polygenic model by simply adding a new covariate (fixed effect)
<code>snp</code>.</p>
<p><span class="math display">\[ y = X\beta + \beta_{snp} * snp + g + c
+ e\]</span></p>
<p><span class="math display">\[\mbox{ where } \mbox{ } \mbox{ } g \sim
(0, \sigma_g^2 A)
  \mbox{ and } \mbox{ } \mbox{ } c \sim (0, \sigma_c^2 H)
  \mbox{ and } \mbox{ } \mbox{ } e \sim (0, \sigma_e^2 I)\]</span></p>
<p><span class="math display">\[ \mbox{ implying } \mbox{ } \mbox{ } y
\sim (X\beta, V) = (X\beta, \sigma_g^2 A + \sigma_c^2 H + \sigma_e^2
I)\]</span></p>
<p>In the genome-wide association scan <code>SOLAR</code> passes through
SNPs variants under the study one by one according to the equation given
above. The p-value of association between trait <span class="math inline">\(y\)</span> and SNP <span class="math inline">\(snp\)</span> is again based on LRT test with the
alternative model outlined in the table below. This alternative model is
the same as the polygenic model.</p>
<table>
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th>Alternative model</th>
<th>Free parameters</th>
<th>Restricted parameters</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>No association</td>
<td><span class="math inline">\(\beta, \sigma_g^2, \sigma_c^2\)</span>,
<span class="math inline">\(\sigma_e^2\)</span></td>
<td><span class="math inline">\(\beta_{snp} = 0\)</span></td>
</tr>
</tbody>
</table>
<p>More details on implementation of the association analysis in
<code>SOLAR</code> can be found in the <code>SOLAR</code> manual, for
example, <span class="citation">[@SOLARDOC, Apendix 1,
[mga](http://helix.nih.gov/Documentation/solar-6.6.2-doc/91.appendix_1_text.html#mga)
and
[snp](http://helix.nih.gov/Documentation/solar-6.6.2-doc/91.appendix_1_text.html#snp)
commands]</span>.</p>
</div>
</div>
<div id="software" class="section level1">
<h1>Software</h1>
<p>The idea of the proposed software (the R package
<code>solareclipser</code>) is to perform the calculus in
<code>SOLAR</code> at the back-end and to offer the user interface in R
at the front-end. The table given below shows a basic description of the
software via implementation of the main three models: polygenic,
association and linkage.</p>
<table>
<colgroup>
<col width="20%" />
<col width="20%" />
<col width="20%" />
<col width="20%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th>Model</th>
<th><code>SOLAR</code> command</th>
<th><code>solareclipser</code> function</th>
<th><code>solareclipser</code> S3 classes of returned object</th>
<th>Parallel computation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Polygenic</td>
<td><code>polygenic</code></td>
<td><code>solarPolygenic</code></td>
<td><code>solarPolygenic</code></td>
<td>None</td>
</tr>
<tr class="even">
<td>Association</td>
<td><code>mga</code></td>
<td><code>solarAssoc</code></td>
<td><code>solarAssoc</code>, <code>solarPolygenic</code></td>
<td>Automatic or custom (by SNP file)</td>
</tr>
<tr class="odd">
<td>Linkage</td>
<td><code>multipoint</code></td>
<td><code>solarMultipoint</code></td>
<td><code>solarMultipoint</code>, <code>solarPolygenic</code></td>
<td>Custom (by chromosome)</td>
</tr>
</tbody>
</table>
<p>One can see how the software benefits from both <code>SOLAR</code>
and R sides. On the side of <code>SOLAR</code>, the <code>SOLAR</code>
commands <code>polygenic</code>, <code>mga</code> and
<code>multipoint</code> do all the hard job on estimating the variance
component models with control via a set of parameters and settings. On
the side of <code>R</code>, the <code>solareclipser</code> functions
<code>solarPolygenic</code>, <code>solarAssoc</code> and
<code>solarMultipoint</code> prepare the input data for the analysis,
pass the arguments to the commands and set up other necessary settings,
and finally read the output <code>SOLAR</code> files and store the
results into R objects of S3 classes in R.</p>
<p>The main advantage of running the <code>SOLAR</code> models in
<code>solareclipser</code> instead of <code>SOLAR</code> is that the
analysis is conduced by only one function call, while the same analysis
in <code>SOLAR</code> could require typing more commands with
appropriate options and settings (to be learned from the
<code>SOLAR</code> manual). The <code>solareclipser</code> functions
just automate the analysis flow and keep configuration options via the R
function arguments, if such configuration is required. Another advantage
of using the R package <code>solareclipser</code> is that the results
are stored in objects of especially designed S3 classes with associated
methods like print, summary, plot and others.</p>
<p>The last column of the table given above contains the information
related to parallel computation available for the models. The
implementation of parallel calculations is straightforward in
<code>solareclipser</code> package, since the association and linkage
analyses are <em>embarrassingly parallel</em> problems (easily divisible
into different identical independent task) and the R environment offers
a number of packages with parallel interfaces.</p>
<p>More information about the <code>solareclipser</code> functions is
available in the software manual of the package, and the practical guide
of using the functions is presented in the next section dedicated to
examples. Meantime, some points should to be stressed to describe some
other features and drawbacks of the <code>solareclipser</code>
package.</p>
<ul>
<li>The <code>solareclipser</code> functions, by default, do not create
any directories visible to the user, though the analysis flow in
<code>SOLAR</code> is heavily based on storing the data in directories
and files. However, the input phenotype and genotype data might be
stored in CSV data format and passed to the functions. The user can
specify a special directory (<code>dir</code> argument for the
<code>solarPolygenic</code>, <code>solarAssoc</code> and
<code>solarMultipoint</code> functions), where the <code>SOLAR</code>
output will be saved in addition to the output R object.</li>
<li>The <code>tcl</code> scripts used by <code>SOLAR</code> for
automation are not visible to the user. The routine of executing either
<code>SOLAR</code> separate commands or a block of commands in a
<code>tcl</code> script is internally organized in the functions of
<code>solareclipser</code>.</li>
<li>The work on parsing the output of <code>SOLAR</code> analysis is
implemented in the functions in <code>solareclipser</code>, and the user
does not need to manually look for the analysis results.</li>
<li>The <code>SOLAR</code> commands used to compute the analysis and
some <code>SOLAR</code> output files are saved in the returned objects.
This feature might be necessary for those users who are familiar with
<code>SOLAR</code>. It is also helps to go back to <code>SOLAR</code>
and repeat or extend the analysis in the <code>SOLAR</code> environment,
if it is necessary.</li>
<li>The package <code>solareclipser</code> offers only three analyses
via the most powerful <code>SOLAR</code> commands
<code>polygenic</code>, <code>mga</code> and <code>multipoint</code>. If
a more specific genetic analysis is required, the user needs to go back
to <code>SOLAR</code>.</li>
</ul>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>The installation process has three steps: install <code>SOLAR</code>,
register <code>SOLAR</code> and install <code>solareclipser</code>.</p>
<div id="solar-installation" class="section level3">
<h3><code>SOLAR</code> installation</h3>
<p>The <code>SOLAR</code> distribution is available on the official web
page <a href="http://solar.txbiomedgenetics.org/">http://solar.txbiomedgenetics.org/</a>
for the most popular operating systems (Linux, Mac OS, Windows).</p>
<p>In the case of <a href="http://solar.txbiomedgenetics.org/solarlinux.html">Linux</a>, the
installation files are compressed in <code>solar_linux.tar.gz</code>
archive. Once the archive is downloaded to a local machine of the user
and the installation files are unpacked, one can find the
<code>README</code> file and <code>install_solar</code> script. In
particular, the script copies the necessary library, binary and
documentation files into an installation directory defined by the user.
A special bash script called <code>solar</code> is also created and
copied to the system path defined by the user,
e.g.Â <code>/usr/local/bin/solar</code>, so that the program can be
executed by simply typing <code>solar</code> in the terminal.</p>
<p>An example of the installation command might be the following.</p>
<pre><code>./install_solar /opt/appl/solar/7.6.6 /usr/local/bin solar</code></pre>
<p>As a result of execution of the command, the library, binary and
documentation files of <code>SOLAR</code> version <code>7.6.6</code>
will be copied to <code>/opt/appl/solar/7.6.6</code> directory, and
<code>solar</code> execution script will be copied to
<code>/usr/local/bin</code> system directory.</p>
<p>After the installation process the new user is required to obtain a
registration key, as it is explained by the message that appears when
<code>SOLAR</code> program starts. The registration key has to be stored
in <code>.solar_reg</code> file in the home directory of the user.</p>
<p>Once both installation and registration processes are finished,
something similar to the following message will be displayed when
<code>SOLAR</code> is executed.</p>
<pre><code>SOLAR Eclipse version 7.6.6 (Experimental), last updated on December 11, 2014
Copyright (c) 1995-2014 Texas Biomedical Research Institute
Enter help for help, exit to exit, doc to browse documentation.

solar&gt;</code></pre>
</div>
<div id="installation-problems-with-solar" class="section level3">
<h3>Installation problems with SOLAR</h3>
<p><strong>SOLAR version for Windows</strong>. Currently, SOLAR does not
have an appropriated version for Windows. As stated in the official web
page of SOLAR <a href="http://solar.txbiomedgenetics.org/solarwindows.html">http://solar.txbiomedgenetics.org/solarwindows.html</a>:</p>
<blockquote>
<p>Because SOLAR requires a Unix OS or equivalent, we have prepared a
VMware Virtual Machine containing SOLAR plus the Xubuntu edition of
linux which runs on Windows XP or later using the free VMware Player.
Beware this requires a very large multi 1.4 Gb download, and will
require more than 10 gigabytes free disc space during installation.</p>
</blockquote>
<p>This is a clear limitation of <code>solareclipser</code> package that
comes from the dependency on SOLAR. Starting from version 3.*,
<code>solareclipser</code> is not supported for Windows. <a href="https://github.com/ugcd/solarius/blob/master/DESCRIPTION">DESCRIPTION</a>
file has a line <code>OS_type: unix</code>.</p>
</div>
<div id="solareclipser-installation" class="section level3">
<h3><code>solareclipser</code> installation</h3>
<p><code>solareclipser</code> package is available in the CRAN
repository for R packages <a href="http://cran.r-project.org/web/packages/solareclipser/">http://cran.r-project.org/web/packages/solareclipser/</a>,
so it can be install by calling the usual R function
<code>install.packages</code>.</p>
<pre><code>install.packages(solareclipser)</code></pre>
<p>The code repository of <code>solareclipser</code> package is hosted
in the public GitHub repository of the UGCD group <a href="https://github.com/ugcd/solarius">https://github.com/ugcd/solarius</a>.
The R package <code>devtools</code> allows to install the development
version of the package hosted at GitHub. The installation commands are
the following.</p>
<pre><code>library(devtools)
install_github(&quot;ugcd/solareclipser&quot;)</code></pre>
</div>
<div id="installation-problems-with-solareclipser" class="section level3">
<h3>Installation problems with solareclipser</h3>
<p><code>solareclipser</code> has several R packages as dependencies
(that can be seen in <a href="https://github.com/ugcd/solareclipser/blob/master/DESCRIPTION">DESCRIPTION</a>
file). Installation of these packages should be straightforward using
the standard R function <code>install.packages</code>.</p>
<p>In the case of problems in the installation process, web pages of the
packages are the main source of information.</p>
<p>Imported packages:</p>
<ul>
<li><a href="https://github.com/hadley/plyr">https://github.com/hadley/plyr</a>
for <code>plyr</code>.</li>
<li><a href="https://github.com/hadley/ggplot2">https://github.com/hadley/ggplot2</a>
for <code>ggplot2</code>.</li>
<li><a href="https://github.com/Rdatatable/data.table">https://github.com/Rdatatable/data.table</a>
for <code>data.table</code>.
<ul>
<li>See also <a href="https://github.com/Rdatatable/data.table/blob/master/CRAN_Release.cmd">CRAN_Release.cmd</a>
document.</li>
</ul></li>
</ul>
<p><strong>Installation of data.table R package</strong>. If the user
experiences problems installing <code>data.table</code> package, we
recommend to upgrade the R environment to the latest stable version.</p>
</div>
</div>
<div id="solareclipser-r-package-structure" class="section level2">
<h2><code>solareclipser</code> R package structure</h2>
<p><code>solareclipser</code> is organized as a standard R package, and
its structure can be seen in the code repository web site <a href="https://github.com/ugcd/solarius">https://github.com/ugcd/solarius</a>.</p>
<ul>
<li><a href="https://github.com/ugcd/solarius/tree/master/R">R</a>
directory contains <code>*.R</code> files with functions, classes and
methods.</li>
<li><a href="https://github.com/ugcd/solarius/tree/master/data">data</a>
directory has data sets distributed with the package and stored in
<code>*.RData</code> files.</li>
<li><code>man</code> directory includes documentation organized in
<code>Rd</code> files (<strong>R</strong> <strong>d</strong>ocumentation
files).</li>
<li><a href="https://github.com/ugcd/solarius/tree/master/inst">inst</a>
directory includes directories, where some extra data in any format can
be stored for the R package. Sub-directories in <code>inst</code>
directory will be copied to the installation directory of the package
and, thus, will be available within the package via
<code>system.file</code> function.
<ul>
<li><a href="https://github.com/ugcd/solarius/tree/master/inst/examples">inst/examples</a>
has <code>*.R</code> scripts with code examples.</li>
<li><a href="https://github.com/ugcd/solarius/tree/master/inst/extdata">inst/extdata</a>
mostly contains <code>SOLAR</code> data files for data sets of the
package.</li>
<li><a href="https://github.com/ugcd/solarius/tree/master/inst/tests">inst/tests</a>
has <code>*.R</code> scripts for testing units during development of the
package (R package <code>testthat</code>).</li>
</ul></li>
<li><a href="https://github.com/ugcd/solarius/tree/master/vignettes">vignettes</a>
directory includes vignettes, which are known as a long-form
documentation for an R package. Typically, vignettes has a form of a
technical report or an academic paper, that describes the problem that
your package is designed to solve. Files with <code>Rmd</code> extension
are the source files of the vignettes.</li>
</ul>
</div>
<div id="documentation" class="section level2">
<h2>Documentation</h2>
<p>The source files of documentation are stored in <code>man</code>
directory as <code>Rd</code> files. R package <code>staticdocs</code>
converts this documentation into html pages. The documentation for
<code>solarius</code> is published on <a href="http://ugcd.github.io/solarius/doc/">http://ugcd.github.io/solarius/doc/</a>.</p>
<p>In the R environment the help for a function of the package, for
example <code>solarPolygenic</code> function, can be obtained by the
standard <code>?solarPolygenic</code> R comand. The content of the
documentation for <code>solarPolygenic</code> function is the same as on
<a href="http://ugcd.github.io/solarius/doc/solarPolygenic.html">http://ugcd.github.io/solarius/doc/solarPolygenic.html</a>.</p>
</div>
<div id="vignettes" class="section level2">
<h2>Vignettes</h2>
<p>The vignetts are created by <code>rmarkdown</code> R package and
published as html pages on the web.</p>
<ul>
<li><a href="http://ugcd.github.io/solarius/vignettes/tutorial.html">http://ugcd.github.io/solarius/vignettes/tutorial.html</a>:
The vignette presents a tutotial of the package.</li>
<li><a href="http://ugcd.github.io/solarius/vignettes/minimal.html">http://ugcd.github.io/solarius/vignettes/minimal.html</a>:
The vignette gives a minial example of the package. It includes three
analyses, polygenic, association and linkage, applied to
<code>dat30</code> data set.</li>
<li><a href="http://ugcd.github.io/solarius/vignettes/modelsGAIT1.html">http://ugcd.github.io/solarius/vignettes/modelsGAIT1.html</a>:
The vignette reports results on the real family-based data set GAIT1.
The traits under study are the FXI levels in blood, the BMI and the
Thrombosis affection. Although the original calculus were conducted with
<code>SOLAR</code>, the code presented in the vignette is based on
<code>solareclipser</code>. The produced results are the same as those
reported in the published articles.</li>
</ul>
</div>
<div id="load-solareclipser" class="section level2">
<h2>Load <code>solareclipser</code></h2>
<p>To start using <code>solareclipser</code> package, the user needs to
load the package with <code>library</code> function.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">library</span>(solareclipser)</span></code></pre></div>
<p>The version of <code>solareclipser</code> loaded here is:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">packageVersion</span>(<span class="st">&quot;solareclipser&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="co">#&gt; [1] &#39;0.3.3&#39;</span></span></code></pre></div>
<p>Other R packages will be needed for the tutorial.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">library</span>(plyr)</span></code></pre></div>
</div>
<div id="minimal-example" class="section level2">
<h2>Minimal example</h2>
<p>The following code shows how the main functions
(<code>solarPolygenic</code>, <code>solarMultipoint</code> and
<code>solarAssoc</code>) of the package work.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="co"># load library</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="fu">library</span>(solareclipser)</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="co"># load data set</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="fu">data</span>(dat30)</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="co"># univariate polygenic model</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>mod1 <span class="ot">&lt;-</span> <span class="fu">solarPolygenic</span>(trait1 <span class="sc">~</span> <span class="dv">1</span>, dat30)</span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="co"># bivariate polygenic model</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>mod2 <span class="ot">&lt;-</span> <span class="fu">solarPolygenic</span>(trait1 <span class="sc">+</span> trait2 <span class="sc">~</span> <span class="dv">1</span>, dat30,</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>  <span class="at">polygenic.options =</span> <span class="st">&#39;-testrhoe -testrhog&#39;</span>)</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a><span class="co"># specify directory with IBD matrices and run linkage model</span></span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a>mibddir <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&#39;extdata&#39;</span>, <span class="st">&#39;solarOutput&#39;</span>,</span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>  <span class="st">&#39;solarMibdsCsv&#39;</span>, <span class="at">package =</span> <span class="st">&#39;solareclipser&#39;</span>) </span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="fu">solarMultipoint</span>(trait1 <span class="sc">~</span> <span class="dv">1</span>, dat30,</span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a>  <span class="at">mibddir =</span> mibddir, <span class="at">chr =</span> <span class="dv">5</span>)</span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a></span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a><span class="co"># run association model in parallel</span></span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a>assoc <span class="ot">&lt;-</span> <span class="fu">solarAssoc</span>(trait1 <span class="sc">~</span> <span class="dv">1</span>, dat30, <span class="at">cores =</span> <span class="dv">2</span>,</span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a>  <span class="at">snpcovdata =</span> genocovdat30, <span class="at">snpmap =</span> mapdat30)</span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a><span class="co">#&gt; [1] &quot;snp.geno-list&quot;</span></span></code></pre></div>
<p>The user can think of this block of code as a quick start in using
<code>solareclipser</code>. The material given further will provide more
details and all necesseary information on the usage of
<code>solareclipser</code>.</p>
</div>
<div id="data-sets" class="section level2">
<h2>Data sets</h2>
<p>The genome-wide QTL analysis typically involves large-scale data
sets. Data sets distributed within <code>solareclipser</code> package
contain relatively small simulated data, which are used for reproducible
demonstrations of how functions of the package work. There are three
data sets available in <code>solareclipser</code>:</p>
<ul>
<li>An adapted version of the simulated data from the R package
<code>multic</code> <span class="citation">[@Andrade2006]</span>
(polygenic and linkage models);
<ul>
<li>A smaller subset of these data is given in <code>dat30</code> data
set (polygenic, linkage and association models);</li>
</ul></li>
<li>An adapted version of the simulated data <code>dat50</code> from the
R package <a href="http://mga.bionet.nsc.ru/soft/FFBSKAT/">FFBSKAT</a>
(polygenic model and association model, both with custom kinship
matrix);</li>
<li>GAW10 data distributed within <code>SOLAR</code> (polygenic and
linkage models).</li>
</ul>
<p>The simulated data set (in <code>SOLAR</code> format) from
<code>multic</code> R package is placed in <a href="https://github.com/ugcd/solarius/tree/master/inst/extdata/solarOutput">inst/extdata/solarOutput</a>
directory of <code>solareclipser</code> package. 1200 individuals in 200
families are included in the study. The simulated phenotypes include two
traits <code>trait1</code> and <code>trait2</code> and two covariates
<code>age</code> and <code>sex</code>. The two traits possess a high
genetic correlation. For the linkage analysis 6 IBD matrices for
chromosome 5 6 IBD matrices for chromosome 2 are available. The six
matrices of chromosome 5 are the original data from <code>multic</code>
package, while the six matrices of chromosome 2 are the exact copies of
those of chromosome 5. Data for the two chromosomes are needed to test
parallel calculation (separated by chromosome) of the linkage model.</p>
<p>A smaller subset of the simulated data from <code>multic</code>
package is provided in <code>dat30</code> data set
(<code>dat30.RData</code> file in <a href="https://github.com/ugcd/solarius/tree/master/data">data</a>
directory). This smaller data set contains only 174 individuals in 29
families.</p>
<ul>
<li>The phenotypes (<code>dat30</code> data frame) are the same as in
the complete data described above.</li>
<li>A hundred of synthetic SNPs (<code>genocovdat30</code> matrix) were
randomly generated to run the association analysis on these data (see
the script <a href="https://github.com/ugcd/solarius/blob/master/inst/scripts/generate-genocovdat30.R">inst/scripts/generate-genocovdat30.R</a>).</li>
<li>Annotation information (mapdat30 data frame) also was generated,
mainly in order to plot the association results with Manhattan
plot.</li>
</ul>
<p>Documentation of <code>dat30</code> data set is available on the page
<a href="http://ugcd.github.io/solarius/doc/dat30.html">dat30</a>.</p>
<p>The simulated data from the R package <code>FFBSKAT</code> are stored
in <code>dat50</code> data set (<code>dat50.RData</code> file in <a href="https://github.com/ugcd/solarius/tree/master/data">data</a>
directory). This data set is adapted from <code>example.data</code> data
set in the R package <code>FFBSKAT</code>. The following variables are
included.</p>
<ul>
<li><code>phenodata</code>: A data.frame with the phenotypes
<code>trait</code>, <code>sex</code> and <code>age</code> given for 66
individuals (related and unrelated).</li>
<li><code>genodata</code>: A matrix of 50 genetic variants (columns)
given for 66 individuals (rows). The genotypes are coded in the format
such as <code>1/1</code>, <code>1/2</code> and <code>2/2</code>.</li>
<li><code>genocovdata</code>: A matrix of covariates derived from the
genotype data (the additive model). The covariates are coded as integers
0, 1 and 2.</li>
<li><code>snpdata</code>: A data.frame of annotation for genetic
variants. The variables name of the variant <code>name</code>,
chromosome <code>chrom</code>, gene name <code>gene</code> and position
in bp <code>position</code>.</li>
<li><code>kin</code>: A kinship matrix for 66 individuals.</li>
</ul>
<p>Documentation of <code>dat50</code> data set is available on the page
<a href="http://ugcd.github.io/solarius/doc/dat50.html">dat50</a>.</p>
<p>The simulated data from the workshop GAW10 are described elsewhere
<span class="citation">[@SOLARDOC, Chapter 3
[Tutorial](http://helix.nih.gov/Documentation/solar-6.6.2-doc/03.chapter.html)]</span>.</p>
</div>
</div>
<div id="examples" class="section level1">
<h1>Examples</h1>
<div id="loading-data" class="section level2">
<h2>Loading data</h2>
<p>To load both data sets <code>dat30</code> and <code>dat50</code>, the
user needs to use the standard R function <code>data</code> A quick view
to the data tables might be performed by means of <code>str</code>
function.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">data</span>(dat30)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="fu">str</span>(dat30)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="co">#&gt; &#39;data.frame&#39;:    174 obs. of  10 variables:</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="co">#&gt;  $ famid : int  1 1 1 1 1 1 2 2 2 2 ...</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="co">#&gt;  $ id    : int  11 12 13 14 15 16 21 22 23 24 ...</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a><span class="co">#&gt;  $ fa    : int  0 0 11 11 11 11 0 0 21 21 ...</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="co">#&gt;  $ mo    : int  0 0 12 12 12 12 0 0 22 22 ...</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="co">#&gt;  $ sex   : int  1 2 1 2 1 1 1 2 2 1 ...</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="co">#&gt;  $ affect: int  2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="co">#&gt;  $ class : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a><span class="co">#&gt;  $ trait1: num  11.96 7.1 10.32 9.76 9.46 ...</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="co">#&gt;  $ trait2: num  13.58 5.37 6.4 8.98 9.21 ...</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="co">#&gt;  $ age   : int  50 25 35 49 51 45 37 29 39 41 ...</span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a><span class="fu">str</span>(genocovdat30)</span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a><span class="co">#&gt;  num [1:174, 1:100] 1.978 0.795 0.231 0.139 0.487 ...</span></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a><span class="co">#&gt;  - attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a><span class="co">#&gt;   ..$ : chr [1:174] &quot;11&quot; &quot;12&quot; &quot;13&quot; &quot;14&quot; ...</span></span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a><span class="co">#&gt;   ..$ : chr [1:100] &quot;snp_1&quot; &quot;snp_2&quot; &quot;snp_3&quot; &quot;snp_4&quot; ...</span></span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a><span class="fu">str</span>(mapdat30)</span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a><span class="co">#&gt; &#39;data.frame&#39;:    100 obs. of  4 variables:</span></span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a><span class="co">#&gt;  $ SNP : Factor w/ 100 levels &quot;snp_1&quot;,&quot;snp_10&quot;,..: 1 12 23 34 45 47 48 49 50 2 ...</span></span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a><span class="co">#&gt;  $ chr : int  1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb10-23"><a href="#cb10-23" tabindex="-1"></a><span class="co">#&gt;  $ pos : num  2105324 2105467 2106094 2108138 2109262 ...</span></span>
<span id="cb10-24"><a href="#cb10-24" tabindex="-1"></a><span class="co">#&gt;  $ gene: Factor w/ 12 levels &quot;gene1&quot;,&quot;gene2&quot;,..: 1 1 1 1 1 1 1 1 1 2 ...</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">data</span>(dat50)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="fu">str</span>(phenodata)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="co">#&gt; &#39;data.frame&#39;:    66 obs. of  4 variables:</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co">#&gt;  $ id   : num  1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#&gt;  $ sex  : int  0 1 0 0 0 1 0 1 1 0 ...</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="co">#&gt;  $ age  : int  80 77 56 44 75 79 75 82 77 76 ...</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co">#&gt;  $ trait: num  -1.763 -1.423 -0.805 0.268 -1.334 ...</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a><span class="fu">str</span>(genodata)</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="co">#&gt;  chr [1:66, 1:50] &quot;1/1&quot; &quot;1/1&quot; &quot;1/1&quot; &quot;1/2&quot; &quot;1/2&quot; &quot;1/1&quot; &quot;1/1&quot; &quot;1/1&quot; &quot;1/1&quot; ...</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co">#&gt;  - attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="co">#&gt;   ..$ : chr [1:66] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...</span></span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a><span class="co">#&gt;   ..$ : chr [1:50] &quot;s1&quot; &quot;s2&quot; &quot;s3&quot; &quot;s4&quot; ...</span></span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a><span class="fu">str</span>(genocovdata)</span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="co">#&gt;  int [1:66, 1:50] 0 0 0 1 1 0 0 0 0 0 ...</span></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a><span class="co">#&gt;  - attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a><span class="co">#&gt;   ..$ : chr [1:66] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...</span></span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a><span class="co">#&gt;   ..$ : chr [1:50] &quot;s1&quot; &quot;s2&quot; &quot;s3&quot; &quot;s4&quot; ...</span></span></code></pre></div>
<p><code>loadMulticPhen</code> function in <code>solareclipser</code>
package loads the complete version of the data from the R package
<code>multic</code> (1200 individuals).</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>mdat <span class="ot">&lt;-</span> <span class="fu">loadMulticPhen</span>()</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="fu">str</span>(mdat)</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="co">#&gt; &#39;data.frame&#39;:    1200 obs. of  10 variables:</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="co">#&gt;  $ ID    : chr  &quot;1001&quot; &quot;1002&quot; &quot;1003&quot; &quot;1004&quot; ...</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="co">#&gt;  $ affect: int  2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="co">#&gt;  $ class : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="co">#&gt;  $ trait1: num  9.61 11.5 11.05 12.37 7.97 ...</span></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="co">#&gt;  $ trait2: num  9.61 11.66 13.62 13.88 7.29 ...</span></span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a><span class="co">#&gt;  $ age   : int  45 37 45 48 31 41 50 38 43 54 ...</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a><span class="co">#&gt;  $ FAMID : chr  &quot;100&quot; &quot;100&quot; &quot;100&quot; &quot;100&quot; ...</span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a><span class="co">#&gt;  $ FA    : chr  &quot;0&quot; &quot;0&quot; &quot;1001&quot; &quot;1001&quot; ...</span></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a><span class="co">#&gt;  $ MO    : chr  &quot;0&quot; &quot;0&quot; &quot;1002&quot; &quot;1002&quot; ...</span></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a><span class="co">#&gt;  $ SEX   : chr  &quot;1&quot; &quot;2&quot; &quot;1&quot; &quot;2&quot; ...</span></span></code></pre></div>
<p><code>loadExamplesPhen</code> function returns the GAW10 data sets
available in <code>SOLAR</code>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="co">#gawdat &lt;- loadExamplesPhen()</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a><span class="co">#str(gawdat)</span></span></code></pre></div>
<p>Both <code>loadMulticPhen</code> and <code>loadExamplesPhen</code>
functions are based on a more general function <code>readPhen</code>,
which reads the phenotype data from text files of a table format. The
data from the R package <code>multic</code> can be also loaded by means
of <code>readPhen</code> function.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>dat.dir <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;solarOutput&quot;</span>, <span class="at">package =</span> <span class="st">&quot;solareclipser&quot;</span>)</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>mdat <span class="ot">&lt;-</span> <span class="fu">readPhen</span>(<span class="at">phen.file =</span> <span class="fu">file.path</span>(dat.dir, <span class="st">&quot;simulated.phen&quot;</span>), <span class="at">sep.phen =</span> <span class="st">&quot;,&quot;</span>, </span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  <span class="at">ped.file =</span> <span class="fu">file.path</span>(dat.dir, <span class="st">&quot;simulated.ped&quot;</span>), <span class="at">sep.ped =</span> <span class="st">&quot;,&quot;</span>)</span></code></pre></div>
</div>
<div id="data-exploration" class="section level2">
<h2>Data exploration</h2>
<div id="plot-kinship-matrix" class="section level3">
<h3>Plot kinship matrix</h3>
<p>The two times kinship coefficients describe the relatedness among
individuals in the family-based sample. Two functions
<code>solarKinship2</code> and <code>plotKinship2</code> allows to
compute and plot the kinship matirx of a given data set.
<code>solarKinship2</code> function runs <code>SOLAR</code> for
estimation of the kinship coefficients based on the ID fields in the
data. The returned value of the function is two times kinship
coefficients in a matrix form. <code>plotKinship2</code> plots the two
times kinship matrix by using the R package <code>Matrix</code>.</p>
<p>For example, the kinship matrix for <code>dat50</code> data sets has
the following image.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="fu">plotKinship2</span>(<span class="dv">2</span><span class="sc">*</span>kin)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAC3FBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxubm5vb29wcHBxcXFzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyOjo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmbm5udnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6ens7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///9YL9+2AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAQQUlEQVR4nO2d+2PV1B3Av+2KXPqgRSiDUsZE6MobRDuK4yFjYxuU4iqie2hx4JgOlSnzAVMQy7MiTBhDFKFIXUvLABERGVKsFHzwEAXEUQSxtFRG2/MPLK+bm+Se5HtzH7lp7/fzQ5r7zbnJyaf3JCc5JyfACEsg2hlwOyQIgQQhkCAEEoRAghBIEAIJQiBBCCQIgQQhkCAEEoRAghBIEAIJQiBBCCQIgQQhkCAEEoRAghBIEAIJQiBBCCQIgQQhkCAEEoRAghBIEAIJQiBBCCQIgQQhkCAEEoTgoKD917nh6jpu+Nh5bvjsKW744lFuuPEgni0EBwVlneSGJ+zihmet4YYXz+WGt0zjhg/lBJAva5wUdIIbnrCTG571MjdcxBdUwhdURYIUSJAXEuSFBHkhQV5IkMrM4Tw8A7nh1CxuuGsvbjizOzd8843ccL9k/ecH7e9LRAQNAqj2Zw8nJoYPVVfnyWjD+97npj6wX/ozR0adUT9zt7lIYfZg+/sSKUH2vlAoE/gXimXUGfUzlxKFJa4RdJTZU0SCEGJQkLBmu+UsLFhqPkKCSBBGKxIkrjwqt+PMFZEgiVYkKBRFhT6K/VESWZ3ANEvceRaTV0+CeAzS3CG2c6y2WRnyR/GkN1ZS4p1zYREjQVxIEMIgfSNMwIpIEEKsCrJ3MrN94aqDc1qLnqAL50wWkCCZmWImvhifOvaUYYGfoCAqRK2/HrS1IE7M1fD5FxcOMSwiQSKVqwYL267udJ01px/QL+IICvx834aK2BQhE5tHCTNjN8iBfb1lPPt4W4q+oHfby/nL4uXPhFAFrZwozOQvlQPXP5fJdusvKFvJIL8jDpdQBW0cLczcsU6/gFvEmAO3PxCv0ShiVV2aWUs3w0+WBMmIgtiAlc2rs1r0C8wERVJRIOUyKoI+H9Nl1GeGBSQIwVyQg/fy1WqUL0SCtJAghNYqiDne2OFTRIK4tD5BdhUVckpLUJAghNYjKNhjdYimSBACCUJoTYJYaCez4DyRIIRWJigMFSKbikgQgrOCakal9H0t4GYfs807Uqv2FkdHBV1OW/7160lVgTb7mG2+7Qqq6CNM8osCb/bhb77tCqo7x1h95h5ds8+h8TId37OTA+duf+SnyPn75QeBfy2U3NXcMqNJ1+xTt1umd42NtTgpqLeSQf6D1lyCz13D7Ix1Npt9zPLg1O0PR4tY89iCb5nNZh+zPLRJQdsyztXW1tbzm33ybdZyhWO1vYqx2nqq7+Fgnbr4CScFzQORIn6zDwmyhgQh2BYkJBcVBf4FVZDaV8Hq4tV9HclJkDXBCBIV2fyCuA1uxzMjJKjVCdLUg+zc2JIO1aFsWPm2W7rgmUKCEIIVJBazkDKkaNFv0+eKBJEghCgIOtyCJODhd7EqZjdQT3KlOthDUaHa71wNRfYYBF1/9yp/PDoLYklQ+eO5N8CQx3c32llpSILkSnUI/aSN12URP4td3bvwduhgZ6WxJejCm4/dltBuhF+8NDtx2J4Am33sCQrlWM25LovsQfqBLEgc97e3G/wWnE4pv7gw7WpgzT5tWBDAT8u/4S3YUSAUvrhTgTX7tGFB+4omp8cNfHCD8aahwHdVT+Tqn/b55EGZzqowNZ+2DyrKyUyP8arU6jpVs0S9WO2sZPATG/kIIM3J9WN4/9EzYz0rWnTNPufXymR+6E3jMkGZSgZtVFxQQS0fr57WDTKn85ZdSn4HafYJXpByMtNpKVZNKYnsCopAEbuzM6TkrTzmv2D5bGEybA3S7NP2BY2c/x7/6bM9iZWX1rc/jjztE4og6VitF2TU4V/ofDgkSODaGW41esPgpGHbsad9YkBQ2dB4iBtcZmulsSSoHCa+unvjZKiws9JwCSqUD0TmgvyPSj40ab1zkWgXy3lA+jPD/1LDglgSlFwp/dmZYmelYRQkKrISZF7ENOoiWcSyi6Q/S/vbWWksCXrOs7K2pXaVZ4GdlfoEaXZFmyAAXZqSBWqNyLun3hsYJSXGiA/NkgqFlyIgqPnhBIiHhIea7Kw0lgQx9uW2NdvO2ltp7Ai6smvLGeFy7KtP9k60s9IwCypULzvcJujTnsI/709HfiBM4+ysNGYE5XV+7ejm9B65JW9V1dtZafgFKYrcJih9sTBZDKftrjRmBMEWYfKm/bufkRAkXZm5TlCpMCkjQeYLrQU1/KgMa/aJbUF/hDJskDcTQQG0cagNgMXFPgtqhUhJFHVB7TweTzvwiPgvLRvTrwwb5K2NC3pGg9/C872/GFimb/Y5WyTTXX2aJsyC1AqRksiuoO5KBm1cGQTdHadlwnomCNI1+3w2V6brQd9uGmcUoiOoq5JBTiufGUELeimfiYKQZp+wC1LuwyqJol3ErLhHelbjHqTZJ4YFiQi/IKTZhwQhzT4REFSo6XLudkFmkCCEQXN5N4mN6HqoSBXmwLegnsxKdJikJkEkyP+7SkdGEmT2XRKEfFfpp0eCzL5LgpDvtm5BeD1IwkqJcqdMXw8ymvC/hebD1fUgEmSNr4hZC1JRTGhD6u1WriAVfaOrdgUaQVKCYjc9N0+CrHFQkNpPz0qQ9+4/CdKuIOqCpoo3zGr5zT4kSGBoyenTp5v5zT5OClKapd0nKFl6eRa/2YcEMXY+7udJA0r1zT4X35Dp5bAgUREqSEiV30vJ4MXA9zNoQUdz37pS6jmsa/Y5PFUmza2C0pQMHg58P0M7i+Uv4Df7mF6LmdSdFROBb1jdd1WD9l6+kajUpKu2C5OCZfxmHxIkvqtr+9XKtM/5zT4kSODNIck575g0+0RBkLaxw4iLL1aNkCAJ7VPPxZrTE3ZLTDqXhbJhRZH+jFahPqnkytFfSBAHSVCh7paqZCqgL1sPiYjh66enhkiQFhKE0IoEMc0xx44gpddCkJuWakQkyJzWJUhEuegMSJDajoF2jjFBrVVrV+CbJUEkCEGtVWvH1/FtlwSRIITWJ8j/jmKg++69j1yseUrMh5LIP+JD89hCdC5W6+5P773Z/tM+sSNo3NxLOzqcsP20j2OCNI8tREXQoe9fZ+zCVdtP+8SMoI25M9KzN+mbfRqrZfoe8aaKqiC1I6MqqK+SQRvjigYt6EVYUber3Ye6Zp//DJdJUm/iu0xQopy/EQf8UpsStKBXs4TJaJNmH3cUMbWfXlSK2IHewmTKUttP+8SMIDZgSUNlyknbT/vEjqAvxqUM3ok97aNvf1BCpopKSorlhuZgb3coIvT3S3yVahfWpEkQl0FLvDY0hYGbUilwihftEkWsuKDCAvWZKemTdgVqEqVSXeKmF4+QIGvcJsg7agMJ0q6ABMWEICaf04IXpFaWrARJaxIORPkkSAMJsilILGeuEfTIokWLpPsKyh9pZk81l7y86uo5c+YsWqSmVb4gIS5YYcEcmbw86ZN2BWoSZV15ee4RNGM4D89Abjg1ixvu2osbzuzODd98IzfcL1n/eab9fYmIID5ZJ7jhCTu54Vkvc8NFc7nhkmnccFUOni0EEoRAghBIEAIJQiBBCE4KOskNT9jFDc9aww0v5gvawhd0qFUJ2s9/g0l1HTd8jP/+nbOnuOGLxvdRyTQexLOF4KCg1gkJQiBBCCQIgQQhkCAEpwRdOOfQhsKNU4Jmii2dxu56NaNS+r7mHzZ9gyt3hG+TESAsegjawhlBWwviREGG7nqX05Z//XpSlTFs/gZX7gjfJiNAWPQQtIUzgipXDS72H6Whoo8wyS8yhk3f4Mof4Zs/AoRVD0FbOFXEpgiCdN31BOqEXavP3GMM897gKsIZ4ZvxR4Bg/B6CQeGkIF13PZmaW2Y0+Yf93+DK+CN8M/4IEIzfQzAonBTk112vYXbGOk6Ycd/gajXCt/8IEFY9BG3hpCBjd73msQXfMv+wyRtcTUb4NhkBwqqHoC2cFGTsrrct41xtbW29MWzxBlfOCN9mI0BY9BC0haOCDN315km/iSK/Xnzmb3DljfBtMgKERQ9BW9ClBgIJQiBBCCQIgQQhkCAEEoRAghBIEAIJQiBBCCQIgQQhkCAEEoRAghBcI6h8Qo+Ow//RpA11WRWtzGhwi6A5ULDq5bvjZmtjJMjHe7BW/LMctF3pLAXx+/OFH5cIGjlC+nPljk2MNS/sn/Jj8cEdUZBnozBTOImxzBW5ib3//t9Jab1eFz6sHg+pU20MvB487hDU1H6578PDnhcqp0O5UVC7+W9Pistc8e8czxWW2em3e5ckPOxE1twh6CT8S53/MmGlMJ04zChoKmPH4RHGKuEjljm0hbG8253ImjsEHdMI2ga1wnRD/DWDoMXCDw02iZZqWOaTQvQvI5zImjsEXb9hmTyzajVbEy+28+2GMwZBy0RBpYogsbU9lgSxW2+T/nyX+qjwC7ogzG2Ma/QJmuInSPQZU4LeAmmsjSfhXXY2QTy7Tx4sH4MSn2WsoSsJYrNg8kurp8AsYfahDst2zhT7komCRnTfsHdCPAlirGRMeqcR68XDT9Nz/ZJztjFZ0KejO8CtfyZBFrRciOrm3S8oypAgBBKEQIIQSBACCUIgQQgkCIEEIZAgBBKEQIIQSBACCUIgQQjhETQCAOJ6jN0qzqeJNwMDxkbqtyd26j+vSTsTTBK7hE+QyLMscoJ2tIeMePi9ZiaYJLYJl6D7mv738S/A8xVjTU12Hs8KPPUgWMn2/7DPNd9MMElsEy5B4pi19R3hGek3UQ+wJisp98Pdw5NzDjJ27an+icPEB2sBtk/rkvlUM2NfTe/ZIXtBo/ILKr09tdddp/gJLpWWXhI3cRi6yirVGZFXAPawJZBQY54kRMIpiN0Fv/YKgnYAqe2ESUYT+xV4BooPzgn7nx7/PYB1rGUIpPaPhz/IgpYIqZIg6Tg3QTVAtbjuzXBr2dS7N2lmJMZBbl0XeIJZJAmNsAp6RBwLVBb0m7qtAPfWbQY4uRs61LL14Lko7P8dZ89nw73sOMAptqvbTU1i6tpEWMDqRkA+4yXwCnoRPPJRTp2ROOmB8XBTA7NIEhqRErSXXRF//d8A1MyHPgsXzouD7cL+C+XpMRjP6jzQfVaZODaXkLoS0q4zVg6dGC+Bl+cB1tY9D+2/UWfk+ALBxA7rJCER5iJ2p1dQDZMml4XJdOUU94qw//sZe0bYf/bPPkIgaamUejUMFL77EUAdL4GXNZDRwhoSYb86I8cv3wADkCQhEU5BDanwNEfQ03Cfd2Pq/jN2dKFwXDotpq6QfkEVkMpN4KUc+jDWmAKH1Rk5/ldBZKl1kpAIn6CWE5PAc44jqAIyv2aHb8up9e3/+uz8JlYbD++Lqc93gBfYlVyYxHgJvGex+hSoFA4uyVfVGWnLRxLgTugmlyWTJKERkYqiXhAbDV1GtodHNft/PAk6Du8IA65JZ7EigF4dIfE4N4H3IM2WASQCLNPMCDTnwM8ae4AyDDI3SYiEUVDGGO+lhkFQw6N9E4eKnTR9Jagqv0f7nvd/qdSD3hiZ2rPgFD+BKoit/0nK0HW6GfHEBR+wtRC31zxJiNDFKgIJQiBBCCQIgQQhkCAEEoRAghD+D3x9LRYtpjceAAAAAElFTkSuQmCC" /><!-- --></p>
<p>One can see that both unrelated and related individuals appear in
<code>dat50</code> data set.</p>
<p>The relationships among individuals in <code>dat30</code> data set is
examined in the following two lines of code.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>kin2.dat30 <span class="ot">&lt;-</span> <span class="fu">solarKinship2</span>(dat30)</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="fu">plotKinship2</span>(kin2.dat30)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAACu1BMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERETExMUFBQVFRUWFhYYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8hISEiIiIkJCQlJSUmJiYoKCgpKSkqKiorKyssLCwuLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dZWVlaWlpbW1tcXFxeXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxubm5vb29wcHBxcXFzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICCgoKDg4OEhISGhoaHh4eIiIiJiYmKioqLi4uMjIyOjo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmbm5udnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6ens7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///8Q9npJAAAACXBIWXMAAA7DAAAOwwHHb6hkAAANFElEQVR4nO2d+2MUxR3AvwkJXB4QMBEhiAUp4SGRmAQCoaAtyEOk2mLFFkul9d2ilAqpqAilpGgtoggooUBCq6AGqW2pqRGoRoqNDySIaAJEIMn8Gd3du+Que3v7nd2d3Z1lv58f9o7br7MzH2e/czvZuQVGmAJ+V0B2SBACCUIgQQgkCIEEIZAgBBKEQIIQSBACCUIgQQgkCIEEIZAgBBKEQIIQSBACCUIgQQgkCIEEIZAgBBKEQIIQSBACCUIgQQgkCIEEIZAgBBKEQIIQSBACCUIgQQgkCEEGQQc6+WPr+UMvvm29KsnIIGjI59yh7Vn8xR4dY6MuSUgh6AR36HkLgo6QIHNIEAIJQiBBCCQIQW5B7xTz03ccd+iEPvzFFmUbftxgrSFuCarJbVi7dm0DF2/whYmJrdxprSGuCaqohdpalwp3wi3yCFIMuVS4E2QSVCvDCKBHKkHUg1JTMz9aunSa5BLEgASlICpITdQkyJBuQdKNZfIJkmwsk0tQ7BgyJSIZBUk1lMkoSKpMLaUgmS7L5BQk0VAmo6DYgeQ4z6QVxCRJRNIKkiURSSyIelAiRoKkyNQyC5LiokNaQbGj+X6eyS1IgkwtuyDfE5H0gvxORPILoh6kYSwodkg/NQVAkL8TaAEQ5O9MfiAE+TnYB0OQj2NZAAR1H9if8ywwgvyaHwqMIL++EgVIkD+ZOkCC/LksC5IgX4aywAiKHd7zRBQsQT4kIk8FLQCFFva/6XlTj+t28QryPBF5Kmj89ubm5k5WvOJ01TjdLm5Bl3UPyv1M3TYMusQ6C/7eexe/II8ztZeCTqbdmFNUw7ZNVt5P3Rz7cO+zGj+dZqEKXvajyiXRCu7ljHci6HDZvraayLvrZyrv5zwV+3D13RozbuAvx9P5oZIZ0Qqu5ox33MHnrNpSobxUPtf7Y85hXsXb6Q8vT7FDajedv+ZQfifrGvxW733WBHk4lnkp6EC/vefrBn7EitZ3Vl/b1XufRUHepWpPR7Gd43JL3mTsoyn5k/+r22VBUAyPElGwvkkn4FWmDqwgr2bygyvIo6EswIK8ydSBFaThwUx+sAV5MD8UbEEeXN0HXZDrmTrogly/6Ai8ILeHsmALiuJqpr4MBLmbiIQKerfLbK8pjgS5mYiECoIrF75w0l49nAkKSg/a/VBZXxj30P526/VwKMi9TC06B52vr5oEFn7VqBsnSVrFtZl8wYJO7bx/YkZmqfV6OBXk2vyQUEF3XQvZ037z+jkb9XAqyLWZfLFJGr6z+4y9eggQ5M5YJlTQW6tnF6SNuXuzfsKZAxGCXEnVwr8oHts0BWxU1HEOiiI+EYkV1HW0+rbBULjIej3ECHIhUwsVNO8K6H/z+g/s1EOMIBdm8oUKKl9x8JLNeggSJH4oE52DLnxs42s0EydIeKYWK2jX+HRIG7vLRj0EJWkVsTP5Yq/FYOYL+7fMtvN9RKAgsfNDQgWV3KW9/MSPS404Yq/uhQrKrdNeXu1vvR5CBYnM1EIFjYrehvXUaOv1ENuDBF50CBW0MrK+patlQ2SV9XoIPsXEDWVCBXUuzYB0yLinw3o9RCZpFWFDmeDvQZ/ueWbPJ3bqIVqQsEQkUFDbay9/rFyOnfhP/Uzr9RAvSFAiEifo/WEA8LP3rla2acl7T31mXq4LgsQkInGCbr7ixcPbCoaWbd936Gzy3sXrlE1skYb9tRr8CPsyJE5QwRPK5gloNtq3Y36aKii2SMP+Wg2LiJjJFycIXlY2O433120Yu65nkYaDtRpWETA/JFBQjbLZlWr/XEVQbJFG77UaG3+pMbfcWj24EPGbDeXzohXcyBnvRFBskUbvtRpbV2ssrOCsgBVEzORPXhit4FbOeCeCYos0nK7V4EfETL7AUywzEolkQkQlea8qKLZIw+laDas4S0TiBC1PIHmvKqh7kYbDtRqWkUSQOZqg2CINMWs1uHGWqS+DG6gwnF2WhUKQk6EsBII0bE9/hEWQ7Zn8sAiynYjCI4h6kCm2M3WIBNm76AiLIA07Q1moBNnJ1CETZD0RhU2Q5UQUOkHUgzAszuSHT5DFCbTwCbI4kx9GQZbGsnAKsjCWhU9QFO5EFFZB3ENZWAVxZ+rQCuK9LAuvIM6hLKyCNHimP0ItiGcmP9SCeBJRyAVRDzKFJ1OHXRB60RFqQRrI/BAJQqY/SBDymw0kCJkfIkHI/JBXgrxeq2GZVJnaK0HaWg3Hz9Vwj5QTaN4Iiq3VcPxcDfdIOZR5Iyi6VsP5czXcI+VQ5uVtwIbP1Vi2QKOs2GbBgkiZqSeURyu4jLMgJ4IMn6tx8BWNB2bYLFgkRvND0x+IVvAgdxn20G4kV3H+XA0XMUhEXp5igp6r4R5G0x9eChL0XA33MMrUnq7VEPpcDRcwuuigS40EjIYyEpRE70xNgvToEhEJ0qNLRCRIj26sJ0F6dJmaBBkSn8knQcYACTIlPpNPggyJz+STIEPiM/kkyAzFUgUJSg2QIHPURE2CTFATtTSCpjVw8wZ/qLPYtQq5kgj6VzE/fcdxh07ow19sUbbhx+9Ya4gXD6bEGHKCO/S8hYfEHBljoy5JkCAEEoRAghBIEAIJQpBC0Ofcoe0WBB29bAQd6OSPrecPvfi29aokI4MgqSFBCCQIgQQhkCAEEoQQGEHYnet2YzF8FmRyK7oO7CkDSbH8RZvisyCTW9F7gT9lICmWt2gEnwWZ3IreC/wpA/pY7qIR/BVkeCu6MSmfMpAi1kLRpvgryPBWdGNSPmUgRayFok2RYBRLcSu6DuwpA/pYC0Wb4q8gs1vRdWBPGUiK5S/aFH8Fmd2KrgN7yoA+1kLRpvh8ipnciq4De8pAUix/0aZIkIPkhgQhkCAEEoRAghBIEAIJQiBBCCQIgQQhkCAEEoRAghBIEAIJQiBBCFIJ2n3T0AHFf+xI/Ch/g1+ViSGToAdg/oaN30/7ReJnJCjOQXhWfXkaDid8aCrokrsV0pBIUHmp9tJWuZWxzqrR/W9Ql1CqgiJblDd3zmKscG1Z9rf+8PmsgcNfUv5RPR3yFpx2u1byCOro93T8H0sjj9ctgt16QZkrXp+VVrj2LyWRNlY46If1T2Ysdbta8gg6Bn/uef9pxnplO/M6vaAFjDXBvYzVwRFWOL6LsZsnuV0teQR9kCBoD7Qo283pF3SCnlA6GmxVLTWyQvVn2h4sdbta8gi61HdN9M2GavZMuvqnvv3wsU7QGlVQTUyQ+gf3MAli10/UXr7Ju0/pQaeUd1vS2uOC5iYJUn2GStA+0O45WAYH2CcZ6ug+e2w0B2U/xti5K0kQWwKzf189F5Yob+/JWvPqYtgVFVR61eb6m9JJEGPbpxQMKt2kpp+Old/OLdnDooLer8iC639OgkzoOuXboYMhyEdIEAIJQiBBCCQIgQQhkCAEEoRAghBIEAIJQiBBCCQIgQQhkCAER4JKASBt6NQd6vuB6vwfN5aiG7dfYuxBiLJOXCwXjgWpPMbcFHRmJJyNN1r7U3RlJPKh41g+HAq6o+Pi0e9C5ARjHR1WFmXxR9c8NhzURnd2KLyaPvxr5bMXlcY3OYzlxaGgO5Xt2QGwXOsTZwGeuTan7N/7i3NL/snYhUdGZ1+nLqcF2HtbfuEjnYydWDQsa9Sq9lgPqpmUN/zW48YBX9bUfKkdI1/tCmejx/u6MO1N9WXwDd2Nfh7gDfYkZDRyxNpDgCB2K9zSLQgyAfIylc2QDvY9iIwBWK22vyC9D8BzrGsc5I1Ohx9HBT2pROVATpNhQANAg3aMI40v9TT6flisvUQO9TR6GpS15sPDjCfWFiIE3QtjuwX9oHUHwO2t2wCO7YesFrYJIqeV9ld+cnIU3M6aAI6z1wZf06FGt2TDKtZaCnOYUUCPIM1VtNHNmZnNysvRjJVf9DT6WASmwzXnGE+sLQQLqmdtaqc/A9C4AkZUVf06DfYq7d+l/h+dzlojcNWSXa1Mi66DgcqAsxsGMaOABHoa/bDatRibOvKbhEavUrrtXzlj7SDmFJvXLaiRaZuvlM2i2FDyvNL+vzG2XGk/+9MI5YOcp7ToalB/hO0IQKtRQALdje66WlHP2DZ4jSU0+qu+UMQbawcBgs7lwaMGgh6FO7qP0dN+xg5XKXmpWY2u1XpQLeQZBiTQ3ei3tf+A3RkTvz2691fK2xrOWDs4FtT14SyIfGYgqBYKv2DvTixpibd/06g5HawlHf6hRp/MgsdZWxnMYkYBPaNYvNG/VUOVk2fEiBHXAFxdp+18LwPmweAzjCfWFiK/KPYWxCogv7wf3JfQ/qYcGFA8AIouaKPYaoDhAyC7yTDAIEkvUG+ditJz2nSWwIz2odEBC4u1h3NBQ6Z0X2roBJ27b2T2ePW+zPgZdGjO0H7DfvRp7HvQK+V5w+YfNw4wEHRd/NKhp9G/A3iHPQtp9Ryx9qCLVQQShECCEEgQAglCIEEIJAiBBCH8Hx9GPyI0748hAAAAAElFTkSuQmCC" /><!-- --></p>
<p>The structure of patterns in the kinship matrix can be explored more
closely if the first 30 individuals are plotted.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="fu">plotKinship2</span>(kin2.dat30[<span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>])</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAC+lBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///9090knAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAUFUlEQVR4nO2dfXwURZ6Ha97f8koyIa8CISQhrxMG8kISCBoBBY1IcrucdwuymNNFV7PnIYq4DueCeHDLgqBxYd1FQYEVSNxFoizyoqJxxReU5cgu+HLEWwUugJwQUp/P/ao6r0NXV2Z6ZjJh6vnjS01Xd0/PQ1dPZ7qrGmGBImigNyDYEYI4CEEchCAOQhAHIYiDEMRBCOIgBHEQgjgIQRyEIA5CEAchiIMQxEEI4iAEcRCCOAhBHIQgDkIQByGIgxDEQQjiIARxEII4CEEchCAOQhAHIYiDEMRBCOIgBHEQgjgIQRyEIA5CEIdACtrbwag4+VfmIqyKz1oZFZcOshZpbmPVKBFIQTHfMiqWLGJUXLCx1lX7LKPi0yzWIlNeY9UoEVBB3zAqXCxB59mCnmFUHGEL2sWqUUII4iAEcRCCOAhBHIQgDn4RlGklZDudziRSMJogTEarxmo2ktcJTjeSEt2ndFKgY1Q47cMZFdkW1iKR6c5azz+LXwSZEeHp5ubmm26uq6uLq4AYXgiRkwMxo6zZjYMH3Kd0sYdVse9tjxfZ8+6mPM8/ix8FNUHpjp81NDSMnAcxphqi8kaIxdP88Z794UMhSBkhiEOQCRpTWVkZaQ4PD9eaIHRGCL0BwsL8nvE3QSMo/dmmpqYx1FKNy+UaehNE6gSIfCfEP0/yx3v2h6ARlPsRRCUVJJqYHEIQByGIQ/zkmpqaCCooPDY2VmeDMFggjGaICC820zcEjSDzjAULFph+CBFGLd0ApeuKIbLyIKrL/fGe/SF4BC2ElhTughhKBYkm5oYQxEEI4mBIzMjI0CZD6KkgO5QsQyDCIyGSnf54z/4QPIImzZ492zAZwjIBIopayoXSiDSIG4r98Z79IWgE9Wpi9DyIChJNrAchiIMQ1Isa8uHdLpQbRjqdTu0oCMMICOnXj3goRcdApBWpfU9vGRBBjs0nTpy40neaoaS6ulpfDmEqhAjLg4ihloZDqTS0vsXCvrx6mnsTo+dBVFDoNbFTmsm27K2dL/ZES2iCVNBefecG7un/MmoFfVSy+9xW8wfSi47TEqYgFfRhducGsm5UksEn32JVrr6vjTkVFRX6fAjjaAhbGoR0tpgApbxSX7ynNwxEEzv0KsTMp/pO1BdMnTpVNw7CmAdhzYSITIWIo5byVb6n1wyEoDdNr37XGNXSd6LsQZqeB1FBIXWQxtvywwr/5DZNCOIgBHEwz3K5XJY5EFHkulgiuS42ilwXk77KipqAT2C+jj+RUj0JciWt6bndEK9f8scmUYJGkL24srIyvRQiqwgi1wkxJh+iMBuigFoqhPm+MDocjlwNhAPlQmhGQ9gO+GOTKEEjiF72YdJIBY2F0kk7NLuXDBANms0Q+nUQ2fv8sUkUIYiDEMRhUAnKam5uboyoq6u7VwdRh+6F0M2FSGJ2P1DNIBH08Vin0zmaWtKnpaWlaiHSUCqJ6yDMW5UWVsUgEUR5V7rcIZoYCyGIgxDEpuNvLS0tr1BBUU888cRjeognNI9B6Ooghoe8oCZTfHx8hAFiiHSkhlI8iiMRC2F82R+bRBkkghrHQUO6fxTEcipINDE3QkfQ1zIXNChCkMRd/wnxt8rIiuNuFUIQYctMDRHkXPztEvcfmZUFraVaNCR06Go018pBunF1Hghqjr6Mr9jf6lvV7z2IngdJXq69PQjj20DQpnIoVDwvTTixVCLhXaXFBk7QoYTODTzR/2XUClpFfmCuWi5NaFkgMfSQ0mIDJ+idOGn7Hmnhz9uFWkEbJ0Bh0q/7VogmJkEEHYq9gjvi3T6TsqCnkQboFdKxufvltXKQlgTh7FVX1qS7Xevm7EEFmzZtuicV4vFYiPUGiE2SphVQyry29iDcMjG2/L/cKjxrYtJ5EBV0jTUxJkIQByGIA+dMWqPT6bQkdIiEVKLQl9fOQZqJsqCGvPr6+rnDIRYOgfiVHqJeswpC0hTye5B8E6PnQUIQQQgKKUEfeHC/YzfKgv4Y4XA4hlkh0undHdqeuzuooHGVALktdD0p5OVBOLMgitLJ60e92J4u/CEIxf3Tb095ulJlQZf2NDU17X6B3BC00S3I/UG51BK5vHrPLS6Xa/gkiNxCiLJsiHllnm5ML/whaMeDJUaU/2DTRU9W2o/rYmzKegTdA80uZxbEhJshbh8PsSzYBAHf7V1SiiyerDS0BH297YFxeoNHneBCSdDcdGS9/rE3Lni0UlWCyqmg8TU1NUPDYmNj9VbS6Z70tzebICJHqFi1Xw7S6IYdZzxdqSpB778MFFFLkxcsWJBUBpFeADE2A+IOh4pV+0PQvqW32jU5tc+7/6KhiCpBlJlU0GBoYoRjGyYiufm8vHDYHwaRoI5P1vwgHiXfKVNFLxzK9TgMJUEzYlD4Las+k6npvHAo1+NQlaD6WkDq9RKdkpJiiIAwh0HYLBBxaSpW7Q9B4xfvvyxfI104lO1xqO5r/vb58+dbpkJESn95QGlEDkTeSIjbc1Ws2k/HoO9PMk6jyW/SfXoc4jNSf7VsVYKWQUuKIb09rqOCfNnEcqTt+18PluEK2u7QIk3edrkqIqhPj8OuLpn6Nz3YAnf8KMgfXTJ3oGm/bdp4K2qQqaNXNQjuPQ7VNTE/7kF+aGKFc+k/8+T+1KAXDuV6HKoSlJXhdDr1pOe9iQoi47xG2iHioiEy1Iwx6A9BYY30n13hMnVEkGyPQ1WCMiuqq6uNxRC2Agipr2sqlFKTICoyVazaH4Iyl9J/lsv9x9EmJtfj0FdNjJ4HUUHB28QeN69q7WhdbXZx5utDKAm6Ml+PtEj/k3ZPVhpKgjD+Yue6nZ97tlJVgnILSH/7bAhLOoQ02mAilJLtEAXBdaJ47rWXTsKfY18d2etRR1xVgtKKpk6danBAWLIgItIgYqmloVAqCqo/NT5NIX8xfkjORzSerNSnTYyeB0k/ogVdE7sl5ncfbbInlWzefei8JysNGUH2ZRDLkAe3PEqEjCD0EsQ2zy++qhLkKK+qqjIVQoTlQwzJhEiUjtRQKg+qn1zptYXtARY0Iq+srEyfCWEaCWFLgYhKgIinloLqR/sBESTbxOh5EBUUXE1MCOIIMpjNZgMyEzxZacgIWtQLT1aqTtA8l8sV8UOIOLebF6igXHKXA/2N/BApvdAdL5I7H/YojowySO4P4vAoucklnQyKkuW8+vaXcdQS+c3jnI7eNpNHbpvJgtCmQ0T8QWnV14YgZfZTQaOgdNZGbrxCv4fQrocwr4AYJ/fLZzdCkBA0mAQdLg8f9TtvumSqQhI0oqWl5bCZ9LdHLgjtAgjjfIjMnUoLB1TQ2agV37xgO+RFl0xVHEtLTU2VLnfoaH/7oSTsJGIgTGuUFg6ooAbyw0zVUi+6ZKrnL1RQkDexti8xPp+8p0+XzCO1EjFvKS2pHm8FvRXTuYFH+v9eag7Sh8fMa+/TJbP1GYnk91WstR94K6g5uXMDWQ+KlsF7QRfuS/y1V10y1XH29OnT71FB1ucAtBZCuwzCtAgiP3ia2JWKmeQeAC+6ZKriz7qwsDALCRu1hKAUhmwkLBD61UoLB1TQzsQvW1tbz3vRJVMV+7OgIT2eBLFeEhS0B+lH6eYt9aJLpioGjyAlhCAOQhAHPwragpQJnoO0EoHZg6TzIIrYg7oRgjgIQRyEIA5bNN1d7d3620ud7oPn5w4l/LkHpdXX1z8wFGKZhfS3R2shtNTSI1DKC54fzJQIZBOj50FUkGhiBCGIgxDEYZvWZDIZNBAm1BMUI5S0a5UWDgVBb6auXLnyHjvEY2aIlWg5hHYxhIFaku1U0kUoCJJvYvQ8iAoK+SY2iAR1Xjj0T5dMJoNHUNeFQ593yVTmoM3hcIw0QWS7391BBTkqKytvJAOBbic3iThzyV0h5NaQUnJrSHGc52+o9sKh77tkKnPpdTIySs/zkqQgtwY9twtC6m9PLrI8dKPL5Ro9HqIoD2LySIhZHt0GJqH2wmGfLpmXWiQy/Hxllc20HkGzodkVV0FMq4CY44B4yNS5gR48gErthcM+XTL3kQvngElNl0xVcARppO1L82AMJ7UXDik+7ZKpCo6gQDaxrguHPu+SqQpJ0MTa2tqRkSkpKaZwCKsVItwMEWvwfI1qLxz6vkumKt4hV95LqaWS+fPnX+eAGJ0OUZQCMdXo+RpVXzj0eZdM9cylmzbQTUwJIYiDEMRh4AS9QkZkT6WC7BkZGdZoiIgIiBgrRGIgD9JKDOC32A2zZ8+2lUJEU0tZUBqZCuFIgJgkBE1bDC0p6T6ITCpINDE3hCAOQhCH4jSn02kcBmGhguKgFDMEIiECIjWQJ4pKDJwgZ2l1dbXFCRGRA2GnllKglBkHUSIO0r2aGD0PooJEE+tBCOIgBHEoL66qqrIWQERmQQylgpKhNNoOMU4cpPOyysrKjGkQluEQkUkQ0pE6CkqZes/XeI0Jcm9i9DxIutwhmhhBCOIgBHGYVDlnzpywUogYJ0RyNkQ6FZQIpUkmz9d4jQnKSnU4HPoUCFMChM0OERUNIY2eKw7Ssk2MngdRQQPdxN6tlIjY78u1egJHkFbavpve6/8afSqorUki9bAv1+oJHEHGzg1s6/8a/dPENjfLsUd2KnDwAKuGuci+t2Unl82oq6uLngyRWA6RNgbCMRpiQlA0sS7mOWUxOOSnO5MSGRUFOkaF0z5cdnKC1aK1Gs1Wq9VogjAZIcwGCIvBqjNbvRgI1i+CGMR8w6hwsQbfOW9jrav2GUbFEeZwwVN2sWqUEII4CEEchCAOQhAHIYhDQAV9y6hYwhJ0gS3oWUbFp2xBr7FqlAikoL2s5yWe/CtzEVbFZ6zxSS4dZC3S7MH5cw+BFDQoEYI4CEEchCAOQhAHIYjDwAtiPgiQWRFQAidIrmMZgT4IUG64StYTApkjXCr0YPOewAmS61jW/SDAq4erZD4hkDnCpVIPNu8JnCC5jmVdDwKUGa6S+YRA5giXSj3YvCdggvo+668X5BlTfYar7F0hs5TcCJe9K5hv5B0BE9SnY1lviIc+w1X2rpBf6qoRLntXMN/IOwL7LebesYxAPMgOV8l6QiBzhEuFHmzeEzBBsh3LCPRBgHLDVTKeEMgc4VKpB5v3BEyQbMcyAt1R5IarZDwhkDnCpVIPNu8JXBOT61hGoILkhqtkPCGQOcKlUg827xn4M+kgRwjiIARxEII4CEEchCAOQhAHIYiDEMRBCOIgBHEQgjgIQRyEIA5CEAchiEPQCNoxJSnCWd/ee1Ks4jMgAkSwCPoZmrl67T9o7us9TQjqYT+iN2WuQL07UikKuuzfDeomSASNL6b/nJv0IsZXlmSFF5ExWYkg80YozJmOcfLKEmvq0/89PWrYC/BiTSWKrGHdE+pTgkNQu2lFz4v55l803ol2uAsyLH5juiZ55R8KzedwcvQde5/Uzw/EpgWHoGPole7yF/pVkNMK3AXVYHwU/RTjRvQxTnZ0YHxLaSA2LTgEfdZL0E5668rz2u/dBC2DHQ29SCwdxskLYeq/Fgdi04JD0GVj55XQ1WvwOi25GtiETroJeooI2topiFyTDyVBeOw4+s//Rd4Pe9DXUNqoudgj6LarBBGfISVoN6J3KixEb+LP6YMcb82TjkHWn2N8IU4IwnejW3+15jZ0NxR/Ynlq113k8QZEUHHC83unaIUgjDdPtEcXbyCHn/bHR4cVkmf0EEGfTrCgsfcKQQp0fD2gbx/8ggYYIYiDEMRBCOIgBHEQgjgIQRyEIA5CEAchiIMQxEEI4iAEcRCCOAhBHHwjiAxsrUmq2ELKUcrPOnWj/3P//cfxtoINpLRqTNSUN7ycxWN8J4jwc+w3QZfzkdmO0HOkV48mEZmavJrFc3wl6Eftlz6ZisxfYdzezhoFR45+z70b2U+116BS3BaGXscLUIVXs3iOrwTNgTwfgRbRfeI8QuvSbSXvNznDCt/B+PuHs6wFpPstQq/+IDb54SsYf3VniiXTdbFzD9paGjms+rj8DKe3bj1N3uL3Of+C8VJUhF9CIzE+jrRnydTfILQHP4n0h9mzqMSXgnA1ur1LEDIgFGmASGzHNyNzDunpBp/frtWRRxB25KPILC36sSToSZjLhmxHZWdoRqi5620+H43W4X9HZRhfQOgYnXQ9KmmLRf+GFWZRh08F/RTldQn6x7YtCM1q2wRb2YQsrXgDMn8Ln3/S56cy0Sx8FKHj+LX4Ee1k7lYrcuG2YlSF5WboJehhhG7uwA+hKVDWorfptGNmVIlGXMAKs6jDX4L24nNk7z+D0OHFKG3Jkkc16FVMH+b9AKrEbWaUcPd2MmQWzN2Ioi5jvANFY7kZetE4x4hq4QBMPr0OdQ7F5YK99Y/Ks6jCx01sRpegw5jGWYg7O7/ifgOf/wDGi+Dz4/VpMMG2nM69BuXAsh8j1CY3Qx8akPH8L9F4jC+SPYxy1oiyObOowpeCLkSiR2QEPYJ+1PVm3Z8f44+WwHHpBJm7ge5BDShSdoYuNk+HA83/IPTnl1ECxkcQOidNfwhEblWeRRW+E9Txl+nI/KWMoAaU/A3+YFxha8/n35BZ1Y5byUEC5j5lQb/A50rQdCw3Q9e32BakP/Ddg0h3/owRbbtUi66X3vlDPZqB4s9ghVnU4ZcTxb6C8AQUO96E7u/1+Y/aUIQzAmV/T7/FliI0LAJZj8rO0HWQvpRHvxkXYjjBQSakl0aNvFKIbryYhO7C7FlU4kNBiRO7/tRwE3Th/lFWB7lJs6cFHapKMqXM/qLzPOjl8ZEpM4/Lz9D9Lfb3u4bbHGvJsDf/4YyY8Lr0xr9E6D38DNLsZc+iEvHHKgchiIMQxEEI4iAEcRCCOAhBHIQgDv8PCI+6+e/pp68AAAAASUVORK5CYII=" /><!-- --></p>
<p>It seems that the simulated data include only sib-pair relationships.
Such an observation can be visually confirmed by
<code>histKinship2</code> function.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">histKinship2</span>(kin2.dat30)</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAACN1BMVEUAAAACAgIDAwMEBAQICAgLCwsODg4TExMUFBQWFhYXFxcYGBgaGhobGxseHh4hISEiIiIkJCQnJycpKSkqKiovLy8zMzM0NDQ1NTU3Nzc4ODg6Ojo8PDw+Pj5AQEBHR0dISEhLS0tNTU1OTk5QUFBSUlJTU1NUVFRWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19hYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t9fX1+fn5/f3+BgYGCgoKDg4OFhYWGhoaHh4eIiIiJiYmKioqMjIyNjY2Pj4+QkJCRkZGSkpKTk5OUlJSVlZWYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGkpKSlpaWnp6eoqKiqqqqrq6usrKytra2urq6vr6+wsLCxsbG0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vNzc3Q0NDR0dHT09PU1NTV1dXW1tbX19fY2NjZ2dnc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///8sZZMLAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAIyUlEQVR4nO2d/XsU5RWGU6FVsbaglY9KtQa/qCEVxBblWwSkYjChVsRSoEIktdoW1NoqKokRgkGotBhRhCgmk81udpcsLBvOH9fZ3WEnuOd9nxkco+F97h8yk+c6593hJptMMnOdaRBipeG7PoDvOxQEoCAABQEoCEBBgCQEda0PWbdm3XqNtYZ4jRqvf9wQxyw35Xq8Zu24T/4SQdDrixY8WxA5vuzBXWOGTYWOo2FP0Supa527qMajnv7S2UtqnEvr5SN6nMnGKh86F+6XVgc7FkGfLvJyv3tZSov+U1jXqW+quCqox/8qe+15ObpW5P0WfVPFVUH+CxxbfUT+vU3k1FJ9U8VdQb1LHz0rf9slMtCsb0QG582bt6xruEbKG1ZJpdR4yFA+ZIhNeSLl3rjcWx5FkMirLfKvP/lfLI/oG/+/etu2bS2H8jWyXi6vkc6qccZT4/ywHqdTem6KDcsYyr10uD+yEgva947Iid/I0Sf8b0ct+qaKq2+x7lVf5p/bKqWFJ8eeelffVHFVkLyyuPkP/kucWLGk/ZJhU8FZQVGhIEAo6IEqSg0FVaAgFQoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgwO7uTEAgKFNPKq2EfuypcSZliIdilQ+ZltFjb1yeWpGgIH4FASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEcFbQ4WVNGwdjTeJ0S9DA/L7C9k2xJnG6JaizTeRMU6xJnG4JKuREujaCSZwjbW1tTx7MBQSCcvWkR5TQjz01zg0b4pSem2LDMoZyLx3uZyLMUfTpXngi1iTOQJAy5fKanMSZbV15SsAkzjKuvsWKqyqTEuNM4nRL0HsbqqUxJnG6Jai90WdBrEmcbgmKDAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCCAK4J+WfmYvlfvtuCEoNyWLQ1byiy9Xu+24ISgwblzG+aW+cVuvduCE4J8ZultGFcESeGTCnq3BVcE/X1qQwW924Irgqb/qm+ojN5twRVBU/+r90FcEfSzvXofxBVB+27e/FaXj95twRVBUwL0bguuCLpqXBF0eUKe3m3BFUENDTwPqqEJ+tjno5d+8me920J7lJs4s+p9kwndxGm6WdO0jB5Hu4nznSkG7Wb2fFgKCASV6sldUMJSKe+pcWnkohpnh/XyjB6nR2KVD+XC/QurTIKO/+B8XEGuvMWOlzl41616twVXBFW/Rd/4pt5twRVB+Qp6rxVXBMmlrp3bO/Ujs+KKIG/WddNnTJnNP3eU0QQ1zflc5PTPf613W3BF0LTK7/Hv36R3W3BM0DS924IrgprmnC6/xRbo3RZcEeTNvm7GjCmzDIdswRVBcqlzx44D/DFfQRN08bGHRGY/XtC7LbgiaMO0DpEXbl4tcXFF0E//Wv748o/1bguuCLqhp/zxCO/uKKMJuv9O/+Wzjffo3RZcETR42w/n3P6jW77Quy24IkjG3nimda/+D7HijKCrhYIAFASgIAAFASgIQEEACgJQEICCABQEoCCAw4IOFYWDJi30z/dbOGjSyDP3NfotHDRp4V6/xT5o8kJvb+/W3mJAIKhYT66ghH7sqXExc0GNR4YN5XqcNuSGeCgb7heMt+DVCbIPmjw7c+bMxQe8gECQN/n5Ksqgyaog+6DJUn9//06nbuKsE8RBkxbKgjho0kJZEAdNfmMoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgwIsRbuLMXps3cUYU1DMaEAgarSeTV8LR0RFPjUfT5/Q4pZcPG+J0rHIvE+7noj1OPRp8iwEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggATIqi68uQRNOGDJieZoIkfNDnJBE38oMlJJqg2aLLMtSHoykP/poImftDkt7ey+gIRB00aqQ2azHd0dLT2jLv50TPcrfmt3sRpulnTkBviJG/iNAyaFCl6JbXBtfMgw6BJCqqhD5qkIBUKAlAQgIIAFASgIEDHS701Du0/3KvR3aPH+9W4t/MDNe5611CuxwcMuSF+uzvc70lS0EcdIc8v3tGh0b5HjTc/rMam8ieX6+W79Xj1E7HKl2wa98nlk7wkBI3nyMyTccpfmRVr9bb5scoXb4hVfsduJaSgEAoCTIig8/3FOOX5/lirpwdilQ+mYpV/of3QS1rQNQcFAZIRFF7jUJ6uZSs/vKxp46DImsbGxp0RyoO6iKv/o7FMBqwePCTMdOyJCAqvcWhP17KUD8zvK2zfJNKcLRb1M/AryoO6qKuPFYvFo0+D1YOHhBmPPRFB4TUO7elalvLONpEzTVJojrZ6UBd1dZ/SYwNg9eAhYcZjT0RQeI3j60/XAuWFnEjXRvmseXlT6zAuD+qiru7zzz0CVpfg+TOmY09EUO0aR93TtUC5T/fCE/Lx5lTpud/j8qAu+uoXl6QFrC41QfqxJyKodo2j7ulaoFyyrStPVcM+4znglYv5dZFXl67WsMtMIEg/9mS+B9WucWhP17KUF1dV/rDd1ydy0vglEZYHdVFXF2l5A64uNUH6sSfzUyy4xtE3qj1dy1L+XvV3gYOLBsb+uB2XB3VRV5fzd5+Fq0sgyHTsyZwHBdc47v6f9nQtS3l7+URlgcjehxduPYfLL9dFXF2O/bYS2FcPBJmOnWfSAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBPg/OV7a8yuQ7AYAAAAASUVORK5CYII=" /><!-- --></p>
<p>One see that all the related individuals have the twice kinship
coefficient equal to <code>0.5</code>. The unique values of
<code>kin2.dat30</code> matrix can also be extracted by the following
code.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">unique</span>(<span class="fu">as.vector</span>(kin2.dat30))</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="co">#&gt; [1] 1.0 0.0 0.5</span></span></code></pre></div>
</div>
<div id="plot-pedigrees" class="section level3">
<h3>Plot pedigrees</h3>
<p>The family trees are plotted by <code>plotPed</code>, which reuses
some routines from the R package <code>kinship2</code>. The family with
index <code>2</code> from <code>dat30</code> data set is plotted in the
following code.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">plotPed</span>(dat30, <span class="dv">2</span>)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAC9FBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrK0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///+xnk/2AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAL8ElEQVR4nO2deXgU5R3HpxoEIVg5QjiSEsIVypkgtxwSi1CwpYhAuCIgKSIE5CqooMhdLkWOYggUSARECglKoFAoIJWiApZDoASQQ0RoAkjO/ae7CTt7zPvudyadMbt9vp8/9tn5zXdm3/08uzs777z7rmIjPlFKuwH+DgUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKApSuoIKuMTIW+twwf/9bfWM79Zmy40erm1i6gnKDjkqY3c/HZtlzqreYkpKxd/OMzlXG37S2iaUsqIxsTaoPQenhcSec9zPHVlltbpu8CEBBs8P+6r74ddNh+SY2yZvAEzSz6beehXvdBhSa1yRvAk7Qpshr3qUH7d8yrUUaAk3Q1WpHtcXrof80q0UaAk3QyImianJ7c9ojIMAEXa30g6hcUP+ASS3SEGCC5r4sji96yZT2CAgwQR0/FcczqxaY0iAtgSUot/xdSb72N6Y0SEtgCTpfW5bvuc2M9gjwV0HJnTOc7Fir3l3cQpaPT7Kkgf4raH5orJPmVdW7LSNk+ZErrWmh3wpye4tt76Xe/bKJLP9iillt8iKwBN0Olp11xRw2rVGeBJYgW90TgqydH6WHt/+VABOUMF8c397FtDZ5EWCCDvxS/B7rs9y8RnkSYIIKm38kSp8KuWdiqzwIMEG2T+oJPmwKn1lsZqs8CDRBtvhh2vDc1tZ1ugacoOzo6d7ZjTUvm9oqDwJOkO1Gk1G57suFC2p9bXa73AgAQZs6ea7K+m2T/a6lk8+2vWh6u9wIAEGrNGeoqZEdVhZZuZbas9qSPEua5qR0BeUFya481xmohrzeYg7yt8ZVf6JOoypP9lxr2fH9IaU8eOG07NLz0dtqRiDIwa3zJ7/7CVoYAKM7JIJ+IigIQEEACgJQEICCABQEoCCAXwh6t3WFBvOLzxgy/qJZqxWk5t02tAp/EDRTGb9jStAbjrsFrSZoVmsEqXm3DS3DDwTlVBxjv32tXL7t8rKOChak5l0bWogfCDqv7LLfblEu2NI7dCiLBal514YW4geCHpx7YL8dV65oTHhdraCN7SV5jw2twg8EFbE+qHhsnUBQUksfedcdi/APQd8NVuKLD0YCQYLDvJp3bWgVfiEoLSTCObxHlyA177ahVfiDoLRHX1E/RvQIUvPuG1qFHwjKqzHYtaBDkJr32NAq/EDQHmVSsgPZUcxbkJr32NAq/EDQCqWY644FHYLUvMeGVuEHghA8WQVQEICCAP+/ghJmzzHCzCHi+tAocX3iaEO7nzPyi5I8CSsFlU2cbIRhVcX10X3E9WejDe1+ctSykjwJKwU9ccdQ/Kumxnb/7qvG8v1SjeWLoSAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAVgnK+mha/8f6TvjQ968FCvvGumgT7LYQCyYKOLE0ITry5XmHdc5pkr9vztDw1q+uOK2z+S6sEXR8YMXuMzYsXD/3N5WeP+QjlxuU4eLTD9wWpvmaJvDBygYRCe+vWbVybLPwmdm4OTdfD4mZkPTe+qXxYTHJBqeJsULQvcTQec53190/hcfdliZLNk3gwaju6g/Gjg+uvhW1Jynk9865YQp2dWh+HOU9sEDQlZgB37st3h8TcUoWLZGgFaEeSj6LfM3nJHi5I5p+6b68tqqhzyLzBV2vN9ursrb6vyTZkghaWu+cZ+F2uwQfzSno92uvN+GJmut85L0xXVBuK+2cfcn1hDOPlUhQepjmF5h3WyySt2dqlwfepVPVfH0uemG6oDd6CF7woyVTjBkXdLPGfm0xs6r0Z70HagqOo9sj78vyGswWdKnyt4JqVthnwrRxQYmjRdVlsZLdFLb8UFTu97bscTWYLWjMFGF5WW9h2bCgm0/eEJXzan8u3k1atPAD/Kz+qSxMFvSgaqawfl/8xAwLWjxEHJ8r+Zz+nWSq4J7rZQ/sjcmC9rSTrOgvbKlhQc+kiePnQoWvlJxgyXew5BdkD+yNyYJm/EGyYvkIUdWooLwK/5Hkw4TD7Q9HS+IXaske2BuTBcX9WbLiQFtR1aigf4fL8j22i6rrBoqqdgqD9Z5Imyyoyx7JijP1RVWjgg4KNTsYukZUnTtJlo84L1vjhcmCOu2TrLj4C1HVqKADHWT5hBWi6izZO94WJfty743JgnrskKz4SjjBn1FBx5rJ8v03iqpLpF1GNa7I1nhhsqBR70lWbOshqhoVdOsJWb7V34WP2lMSv/d4rmSNNyYLWiUb+/7mVFHV8GG+tqTHK6dClqh8qZrkRP+Q7PCmwWRB52pIfv4Xs1dUNSxo2BJxfKdkSvK6knGJ08Rf+AWYfarRJl1YPikWZ1jQbslst3GSWd6mjxGW8yJ0T4JvtqB1TwvLg737iIoxLKig0U5R+UIVyRfIy5WFv1PY0ElUFWK2oPwmmwTVgzXFT8D42fzHjTXdO3Z6zZLtZ9xwQTErTNBnIsH0/qAjoZc0tR/qbRGHS9Bh9sJYbW1F0xzZfrIiBTNTxo2UxbWY3+W6uJH33xHd7Txeki2BoFsNNR83O6r5mK/98xDNF4DXW+rvL7Oi0/7N+p7H4itth8t+2V6SPunMyGmeV24+CD3iqzm7QjxfQzmjmwu7XiRYcdknqcofXS/5gjWh70ivOpToqsbNLu3drtxc7NsYnDUcqx3vZuRQdG/5ZSgBllw4/KZHrZnFEz+fWdiwwz/kwZJdFyt4P/T5zUWf+nd3Dak8Hb5fssdXSthT9M35Zsqvwo1c0rBZdun5i7GRwc3aNa8SNtznX6bkPRorockAX9vdX/1c+YiY9vXLt1/8va+ck+uznioX9VTbsJ/3ShEdBH1h3eCFO8cOHoM/ljySIeMq2LLg3NEDZww825yTRw5mluDfSQLg92KlCwUBKAhAQQAKAlAQgIIAFASwRpBz+r7sxIjyMZt1xx2kwRF1rvzWoqk7hNdsxfs/1btynaV4/+5YIkidvm9Q8KL0l5QMvXE7px8fpH/3C0JW2JFditPmz4Z03zJF0T1uoQgrBKnT9935mf3MsLBBvM64/TY3Rhmke/e2V7oaao5tVOMcm+1p2fgKMVYIUqfvO9vZMZyw44s64/bbya1iBuneve05XT2Daj630jz7neulPsrVc96/wvRysgEN2vi+CmfaDNK/+/rdoss3g39Np+YvKHvzThrpLHNg2VHs4fR9S8oqibrjt8OX27AgNV/wWOUl20YoC/TmDyvvBCtKN2PzMVkkSJ2+78LHE8vgZ/Aw3q+7TZ+g4nxOqmOIxpCKuBOjOJ+mVN+V/beassvRYqwR5DF937hInfGUytf0CfLY/VblnI+oe/6QkmS/Xa6Ufperc/q+zUVDglcrYMSkM574cE4yNDWiM3/jqGP32+EcZs68/TPIfrtTOYva744VgtTp+9IVR3/08Fo642d224nquhv8n4iaz1A22G8TpKPOvPO2Ro7hQpOCDfUrWiFInb4vt12d5E8mPCIc2ySIFy3ht5iaz28dMjNtzCPom7pr/6llpu6cGuRjWL4AKwS5pu/LGtGgQssNuuMOsCBX/n5iw+B2wov1kv2ntKrQPFnXc1DhySqAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCAP8FcbgA4vg0rC4AAAAASUVORK5CYII=" /><!-- --></p>
</div>
<div id="transform-traits" class="section level3">
<h3>Transform traits</h3>
<p>To show an example of trait transformation, a trait with an anomalous
distribution <code>exp_trait1</code> is created in <code>dat30</code>
data set.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>dat30 <span class="ot">&lt;-</span> <span class="fu">mutate</span>(dat30,</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>  <span class="at">exp_trait1 =</span> <span class="fu">exp</span>(trait1))</span></code></pre></div>
<p>The list of available transforms in <code>solareclipser</code> is
returned by <code>availableTransforms</code> function.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">availableTransforms</span>()</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;none&quot;    &quot;inormal&quot; &quot;log&quot;     &quot;out&quot;</span></span></code></pre></div>
<p>The following code plots the distribution of the original values of
<code>exp_trait1</code> and its transformed values.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a><span class="co">#&gt; Attaching package: &#39;ggplot2&#39;</span></span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a><span class="co">#&gt; The following object is masked from &#39;package:solareclipser&#39;:</span></span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a><span class="co">#&gt;     annotate</span></span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span>
<span id="cb23-10"><a href="#cb23-10" tabindex="-1"></a></span>
<span id="cb23-11"><a href="#cb23-11" tabindex="-1"></a><span class="fu">with</span>(dat30, {</span>
<span id="cb23-12"><a href="#cb23-12" tabindex="-1"></a>  <span class="fu">grid.arrange</span>(</span>
<span id="cb23-13"><a href="#cb23-13" tabindex="-1"></a>    <span class="fu">qplot</span>(exp_trait1),</span>
<span id="cb23-14"><a href="#cb23-14" tabindex="-1"></a>    <span class="fu">qplot</span>(<span class="fu">transformTrait</span>(exp_trait1, <span class="st">&quot;out&quot;</span>)),</span>
<span id="cb23-15"><a href="#cb23-15" tabindex="-1"></a>    <span class="fu">qplot</span>(<span class="fu">transformTrait</span>(exp_trait1, <span class="st">&quot;log&quot;</span>)),</span>
<span id="cb23-16"><a href="#cb23-16" tabindex="-1"></a>    <span class="fu">qplot</span>(<span class="fu">transformTrait</span>(exp_trait1, <span class="st">&quot;inormal&quot;</span>))</span>
<span id="cb23-17"><a href="#cb23-17" tabindex="-1"></a>  )</span>
<span id="cb23-18"><a href="#cb23-18" tabindex="-1"></a>})</span>
<span id="cb23-19"><a href="#cb23-19" tabindex="-1"></a><span class="co">#&gt; Warning: `qplot()` was deprecated in ggplot2 3.4.0.</span></span>
<span id="cb23-20"><a href="#cb23-20" tabindex="-1"></a><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span>
<span id="cb23-21"><a href="#cb23-21" tabindex="-1"></a><span class="co">#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span>
<span id="cb23-22"><a href="#cb23-22" tabindex="-1"></a><span class="co">#&gt; generated.</span></span>
<span id="cb23-23"><a href="#cb23-23" tabindex="-1"></a><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></span>
<span id="cb23-24"><a href="#cb23-24" tabindex="-1"></a><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></span>
<span id="cb23-25"><a href="#cb23-25" tabindex="-1"></a><span class="co">#&gt; Warning: Removed 18 rows containing non-finite values (`stat_bin()`).</span></span>
<span id="cb23-26"><a href="#cb23-26" tabindex="-1"></a><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></span>
<span id="cb23-27"><a href="#cb23-27" tabindex="-1"></a><span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAC2VBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAiIiIjIyMkJCQmJiYoKCgpKSkqKiorKystLS0uLi4wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk8PDw+Pj4/Pz9AQEBBQUFCQkJERERFRUVGRkZHR0dISEhJSUlLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dnb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///9khLolAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAYiklEQVR4nO2d+2MU1dnHR0mhUGmrpUCqVvFesVyUAkotLZdALLAQUQQJIBLlKkUaK8EWo8hFRSUkUnx9RVBRVJCQEJKwQRGseRHEGiHVwt4v2d0km33+gnfOmduZy+6Zze4yG3K+P+TMPDNnzjefndmdy3POcMCUUJzVBrJdDBBFDBBFDBBFDBBFDBBFnQNUvWLFiuVLlq8gtFQ1s4ScW6ZaMU611zttZMVSou3ly4h2SRfkOnHjhLVtKQGqOAYQdbSSIR85E/CQcy3txEyHI2JULfp4p42Aj2i7LUS061KmY35lOuLo0BsA9X8ULRInGCBlkgESdZEA2Ww2MmQhINJIlgAqqwsGeUBBQm5yxuNSzfmJmYDDZ1TNV0Rv1UDZCqj8aHs7D6idkJec8bnJuUCEmGl1tBhVizzWKSPZCogdYiZ8MUAUXwwQxRcDRPHFAFF8MUAUXwwQxVdWACqrDQQCNltAls+jTHucynTArUx6HX7DuN/hVdZ5VGyhiwMq/zQajdpsUVmRoDIddCnT7T5lOuRoU2a8ymSrI6xMLxJb6OKA2CFmwoiVgOra+D+NC2e+1iEXsi8GCKB5agu/6qwvwsurpELxxQDB3+8bzwM6tgzgcLFUKL4YIF75PKCPNgI0LZAKvtndu3c/cyQc5gGFCXnIGa+LnPO1EDMhR8CoWrBr3g9CgHa+BnC+QCoAfhgyZMjsgw4HD8iRPv330YRG4ikbAH24id915kmF4osdYiAAOvYEQH2xVCi+GCAQAEUf+Kbjr5VSofiyAlD4A9Cfb1gNCL5aVFgWkwvZlxWANs4F/fkGO5OWdaRoLujPNxggSe5FjXNBOd9oO3ny5Kaj7ej5ivKUJBwknqa4lOk2nzLd4mg1fBrT6gjJ05EueLEaKz7+3VzQn2840nq+IZ12LBRb7UKAPngVECD5fIPtQRqtnzZtyoRpXt35BvsOUoT2IN35BgOkCAHSnW8wQBQjDBDFCANEMcIAUYxkIyCWQEURehxlIx9HqZ4vRaN+NzkXaCVm2hwho2qtLIFKVPc4xBggE74YIIqvrABULlysKgml3fli1Ujo5zSIAYm/hn6v8gvpdRr/yvocAcM4+QPru0SyO8RDTPHCDjG9EQaIYoQBohhhgChGGCCKEcsBLRk/fvyW7E6gshZQgb+tLZrdCVSWAgoX4CKrE6gsBfRtwaPT17izIYHqyMLpqy8YJi9YCujUWlf0uWeyIIHq/NTT4c1/M0xesPxX7PTULEigqioBODfdMHnBUkCnTwN8U5AFCVThAED1auVYl4xYDah21vmODZuzI4Gq5oGvlGPdWVBQUHTI4/FgQB5BbpdHlsuhTHucZNxtGHcTcWcSyQu75zywviUbEqj8a4qaiOQFX0lJySqhM4tN7s+SdGcW16XTmaXtMfwBGSUvWP4lrZUlgA6tFNY0SF5ggJDK+Eue8TMMkxcYoERGGCCKEQaIYoQBohhhgChGGCCKEQaIYoQBohjJRkCoP7+N7M+vuqbhL3xUcx7i0kd1wUNcInlTSqDKOkBiAhWZQmVpAlXWARIPMfIgs+YQ21rj9XqxE68gj9sry+1Qpr0uMu4xjHscSmVXWq7mrQek2oPQX7YH6Y0wQBQjDBDFCAnIZmOAdEYEJ5IYIJ0RBohihAGiGDECJM3JKzJAhEAFSDAZ8ytuGSBrABkkUJGMLh4g4+wOY0Cyw8wDMkqgsgRQnOyOBEIrioDwHx0gcZ0ocUgkD8gogYpEJP2neD6TgOJkd1gOSE6qCFZUVKypD4U0LjwhQfxkKOR1hQj5gsRMi8NPLpOqBUzfD9IZ0TrRCTfrIQKOFsWrR/IcCgWJGoGkr+b1I0KkVeZHoMqwEclP0kNT6BKorHorgs5IlrwVQZdUYRUgfXZHdgBSJ1XsaGg4sr+ugVAVOVNdSc7V1BMz9v21RtXqzf+KaY00VBFt19UQ7R4gmiXWqd1vN/RN/kf1yZ8HkUkVX/Lfjy/Pfr6CUBk5s6pItaicmHl19jpy2VZpojJh44mMVCxdqmyvfJsyXTyfiBP+np39qoEBzX+0P2lAWp0bUhd32dPT4y5yDdkfd1kntWiRcfzFscbxT4a4DONnjf4jBkgRAyTqIgFqbw7HXeY+H3dRR3Mo7rJOyuk0jvu+N46HmjsM44b/UecBdRMxQBQlAYi8zaANvjV54kP/jbNM7G9mdlHSIn0ZjBDOF08vnL66duHMv2vjK2fNWPe5QZwcX9w8INVtBk3w+wl7PTO1X5VShYLnHm6LxlmEu6KlJtKXwQjhfHF24pvhTflfBCdvV8e/mrwnsHKqPq4aX9w8IOk2w/GFsze0aYLl/K/We5OMl4WnovEzjRcVpAiH9AXGI4TzRdXiYtiXB8cKi9Xx+pJi2Fygj6vGFzcPSLzN4LU1dbzwMgrUKME9zwHs+rPxsi/yHp64xm24SOyKlppUnVv0I4TzRfjkAtjwIHy0doE6Dk2FJx5abRQnxhc3D0i8zbBvLcDXM1CgCMjRuGvun2a4LLZ0xb/mPveMYTWxK1pqkm9/SIDUI4QLRc2Uf8DODQXa+NQFBS8YxYnxxc0DEm8z/M/MIl7SfyqNxv38mqJ642V4/MzTUw2rIZ2emiIg1e0P/QjhqPi/Pxft2AQfPjNPE+eLZ2YZx+WtJvEdJNxm+PAVgMgZOLRwYf7ChafEYMOUslicZeunTJkyYcoMw2piV7TUpLr9oR8hnC/aHp4d44tjC4rV8Xe2FMN7+fq4anzxJH7FhNsM52c2tb+0AQWKlGDVpPjLamcdn7thc5xFuCtaalLd/tCPEM4X1TZctE3+X3X80NTdwdL79HHV+OJJnAeJtxnq5hWs8Uv/qRiU+pkYLYPdD05c3xJnkdAVLTWRtz8MRgj/atE05G7aosJnNfHC5bML/nHMIE6OL87OpCligChigChigChigChigChigCjKOKALAfV8hDshxZ4IJFpRq8+HjYq7bBwnaJxhS62/dXTeZcYB/a5UPd++9D9irJZzJlpRq1l5F+IuO2W3v8Ftt9tPGbZUUpiCy4sFqEUb25/fyxBQ3BPrsU+pZrXrNfKfuXFLkQFfpeAydUC+wl//dNw5+LDPN9B++1/s/V7L7XvPSXnpUI4bC5z93gnw9cT+fUedQDsvjtWWFqkA6VaEnHdv633DLoA9g3tfvx5Gc9xd4CwYmDuDP17wejkvXtPn7uaiAb/El3gCIKOWdo5MxWXqgO4Zfejo9Fs9cP/vYe2tEfvluW/X/KmfkhKAPgduxI5zcMvdlQdHDsNHt/B5fabfg8gVIefqtxun/Tj8Xc8nPl3H1aM9KDZseFXV8GHiejmDDlf/qvfqU3N6eBRABi3NWZWKy5QB2X/E24sO2Aqu/qt/chTsHP+Zh/pvVDe9FiBW+m+A7f0ogIgVIYefOc2dOXDZGYAPmhCgqh7fATT1OITXg5x/AjxyM8C3XKMCyKClO95IxWXKgMq5HF7c0wA7uWW8E/xf582Tl+OmP+Ynwm+tyr+SBohYEXL2ATi5M6ERvfI2nsXfQS/diFYctEVYL6ca4Mk/AHhJQAYt5X6cisuUAb0zUJraxE2Sms6fo266FiAweOi69ytogIgVMQAeEMQOP3lHz3cRoBcxoBs34/UEQGM1gAxaunZvKi5TBnT6Mv67rvnOL+DMFW/23cU3vZv/GHKf1zW9p6cbYIcJQNKKEqBKdDhNnIQAHcw5B3A2pyoxIG1Ld1ak4jL1L+nJN31UOeq2aGxkIWwc6LFz1753ZOJVSobZyEc8uOk6buvZ3df1caCmUUwHSLuiBOjg5Zsad15Zgr+kh46oPTxiaCwxIG1Ljz2eisvUAbXMv+bneU2wIdcLHcMfsnN7b+87ulFZvPUXk4X/p2TgVfedumUYahrFdIC0K8qH2AuDel23sh2fBzmmDxiIf+YTAdK2tG9wKi7TfaJo51J9Upp2ddygy2pJwmU3AARl+dqIhYCOX4+brhkjakeat98ZxaZpL1aTcMlud1DEAFHEAFHEAFHEAFHEAFHEAFHEAFHEAFHEAFHEAFHEAFHUtQDhRHrx7RGQpkR9iroUIJxIL749Aikdifo0dSVAQiK9+PYIgPQk6tOUGBDepckduXqFqKUrNFq6RBfRBpYs0wSWE4HXzbhFOZri2yMAlER91/z589cuWbJk8RKsx4VysVg8LkQTLZTnFsurbjUDCO/Sqh0Zd8NG8sc06/o9moC6oziSK6gJRJWAuW7h+cLDc/T2CAAlUR8BWlLj9TqFMRCdwnCIbrGQoqqxEl3CUIpOMepSDbDIB11S15xEgIRdWrUjZwMg4e0RouREfeRMbNElYIwIHfSl3vPCQqnbflD4yH1eoXUh+6XVEZVWlf0kPsSQIXlHbjt58uSmo+J7SH1t7Wr53JoA+QZTQS6/JtCqBCKmhipFfsS3RwCoE/WtAyTvyGbGyrCJb0SyJftmJHNjdyA/4tsj4HRIlahPB6Qa8CSNgLAZtCOb2YNs4itzUZmhPUjK6s87qUrUtw6QbkfGivMdJI3Ag8oMfAclkHWAdDsyFgOEhQ8x7Y6MxQDFt4GVdYDQIOni8OZO4Q1RXq/qHVFuFSC3MEq6Sxg63e9WvTbKTYyQfskAKv8sFvPFsFwtuAiHcRF0CVHNHtQuBL24aA/gIuIQo3zkktuD2CFmwhkDRHHGAJEjT4B+TM5uD4gceQIMxuTs7oBUI0+AwZic3R2QauQJIKZaGxoa1je0tXnbsFwBXLS04CLgEqJeFSB/BAc9HlxEfEINhxDltxOWrg2TBFR22C/I5dfI5UR/UdNSqVvF6dYEfErAE2eYLT0gZSgK82Ny2oxl5u5C19uDlKEo5CnpPoN4/8AVwEVLCy4CLnxzwUh8DY9wW8GHi5CjVbqTFenkHpQFgJShKHRjcsb9DooDKGu+g8RoegBJg0acDuneuMoAkSNP5J3UvXGVAaI4Y4AozhggijMGKIHKu9DPvPjxXFxAZXXBoDuI5fTiwufDhdcZDMYBxNdw4XX8Qk2fI4BLfs5n7o6i5uo5mwFZcohprp6BAVJLe/UMDJBWqqtn6ZpZuOBtbVPL60Z/+VbxH1R65WU42tbm9GvqRJRA2NSTVaRd6MnqePyPkYk51gFK6j0WNnSJLLROPKNX5lN8No/UweM89jSe1CbmxAEUBw9SOgDJ18zR5ubmLZ+KL471tUfV8uF3z9rQu2mFtvE7ZvG8EIhGXQFNnVa/MpnMi2iji/GA3rrEHGsA6d9jgWTyO8imvKspjd9BH5TjQjUUrHWAdNfMWFYCai8UkrXkxBxnQUFB0SGPx+nBcrhw4XbjwuVMAMgpVHGLNRxCFT7oNJNhpr56zh5A1WuUaZyY4yspKVlVK7+QW/voOQEg40fPrgw/es40oOK9Qmk2wyyDh5hOCQGJYBRA4nyaAUXyfkCF+Qyz7gboxFxcmMwwS0BH8nupAYrvjAGiOGOAKM4YIIqzrAZENKgzIE1mEtDWGq/XZdAVgQrIoCuCy2RXBANlL6As2YPQ5yRBVov4RHSfkDSpq6QE5E8svYCoeARE3eEQY4BMOMtKQDblQY8xIB0uRZ0DpNxHpKfgdUtA8n1EEyl43RGQch/RRApedwSk3EeUHyeEP/nkk2ftkYgnguX04yIQiERMAYpEQl5cI+gI4ZLfTkva7gdddEDKK6VMPE4wBSizKXgXHRCS0FNVfpzQ4fP5XjPsq2EKUGb7augAJVbqgJT7iCZS8Ex6Su07SPt4Dss6QNJ9RFMpeBcFUOJ+8xcdkHQf0VQK3sUApHs8h2UhIGNZB0j+WZV+TIVfwrA4YUM/kxGTP6jiupJCPnlS/lnteoAo/eZtylN4M4rXpd78s3mzgEwaUiilcqKIf1alH1Msf4c4YUM/kzGTP6hoXakC6gkZjMmTqR5i27R9NZIE1Pm+GpR+87Zkv4NsyhdROr+DttWHQu4QltOH/iYJCFUJOIJ4A/x2AubPpCn95rMFkIWHWOJ+8wyQzgYWAxTfBhYJKDkfMiCbrXOAjAd5Y4AkxRnkTQ0oSVeit0sDUJxB3hggSQaDvDFAGmkHebswYcKEhdVut9ON5XC5qT6M5HbxNZH47TjSecPsIgPSD/Lm37hxY3FdMOiS+2p0CpDcV8Nluq9GhgDZxKdlnQJkapC3TgFKxyFWbnQtloykCrhmp67FTA3yZhkgfb/5JE1IFXDNpPvNYwv0Qd46hQcrS76kbakcYgmcMUAUZ1kFSPonUxEDxABlDyAyBa/zjtKYgpdtgNgeZMJZam4IMUAUpQgIbSDZE0RD8YB0hrogIF2/eYsBxR8FLx2etH7ogPT95q0FlGAUvHR40vqhA9L3m7cWUIJR8NLhSeuHDkhOdAtWVFSsqceP5EIhTzp8BL3i4z3iQZ15P3pn6fCk9UMHlFS/+U7L/LP5zoyCl4IfOiBl1Dmg91kllYFB3tR+dM4SDzSJ/6b5rQhISfWbJ5UhQJ0ZBU+xkwFA6kS3HQ2CquwNalVVNmgj2sCBQ5rAESVQb/5XLM4oeDvkFg/U4KK2Fhc1B0g79v1C9NARIXhQMFKNi7r9R6RV65M4DyIT3b6sEFVWXqHWysc0gYqt2sD8Yk2gvEyZrkzowcCPJgXvS6XFBU/hYts2XPxtHmnn1dmlpP/lS0gjz89+WV5V8pO2t0P9dSZ1lT9upq6SFt37Ejn3yhhyzjFE9UksXkDO1Q05p9sYA0SIAbISkOs8dZUfdL9rmdH3qnZ835NzHc0hctapehN9uLldt7G0AbpUxQBRlC5A1BdW6kYczJyUDCuybSydAWWRQT2kdAGivbBSP+JgxkRkWBFtY+kMKIsM6mGlCZAql8lABiMOZkxKhhXZNpbWALFIX09QmgCpcpkMpR6vP5NSMqzItrH0BvJbEtRDShMgpRtgPKnH68+wxAwrsm0svQECkL4eUhp/xeQXVhpKPV5/pvTx/PmfExlWeJagoDdAAlJlZklKEyBVLpOhdCMOZkxKhhXZNpbeAAFIXw8pTYBUuUyG0o04mDFJGVbqtrH0BghA+npI6TrEyFwmQ+lGHMyYpAwrTdtYOgMEIH09JHYmTREDRBEDRBEDRBEDRBEDRBEDRFF8QJ8PGxV32ThO0DgyGOFOwAX0BK71t4449WRdCKin5LpW2dDUBXhCXBQf0Kw83c0jWafs9je47Xb7KTLYvvQ/8LtSfqKkkGZMWI+YkutaZUNTF2o5pxDi4p7+jn1KNatdr5GHrV+Ath4ZoLskjuesRTu1P98iG+rY/vxeMqAJ8PXE/n1H8c3kvHtb7xt2AewZ3Pv69TCa4+4CZ8HA3Bn8jsrZ750AOS9e0+fu5qIBv9ygOMMLxA3w++dQjhsLO0fyC3yFv/7puHPwYZ9voP32v9j7vZbb956Tsgm8Hq77G47LGcXZx3An8FRFqTU2NHVrS4tkQOfglrsrD44cxgO6+u3GaT8Of9fziU/XcfXoo4sNG15VNZxfxI3YcQ5yBh2u/lXv1afm9PAoztACcQPoAEb456ziF9wz+tDR6bd64P7fw9pbI/bLc9+u+VM/5XEMWk+o+7N5B0dyI8q5E3jKMhvqugCfyYAgVvpvgO39eGdrAU5zZw5cdgbggybkrKrHdwBNPQ4Bxy+CnH8CPHIzwLdco+KMXyBtQNr6HW8A2H/Eu48O2Aqu/qt/chTsHL9LhPpvVDvDdYeWwnZubYQ7jqcss6GuqwIE4bdW5V+JnO0DcHJnQiN65W08iw/+l25EqwzaAtzHyFk1wJN/APCSztACcQPS1nP5WDmXw4t7GmAnt4w3ipvLm6d2huvePCb/Su5jvi6essyGuq4KUGDw0HXvV/QTWuadQezwk3f0fBc5exE7u3EzcLWSs7EaZ/wCaQPS1q/dC/DOQMnDJm6S5Cx/jtoZrnvFuPcruNoIV4+nLLOhrqsCtKenG2CH4qwS7ccTJyFnB3POAZzNqUrsTNqAtPU7K/hj5DL+q7D5zi/gzBVv9t3FO9vNf0q5z+uc7el5VynsQIA24CnLbKjrqgDVcVvP7r6uj0NydvDyTY07ryzB345DR9QeHjE0ltiZtAG09ZGPeOAxlHs0+aaPKkfdFo2NLISNAz127tr3jky8SklCQ+sJdW96YPt13N4I9zqessyGuq4EqHwz+g4qGXjVfaduGSbv2y8M6nXdynZ8AuKYPmAg/n1N5EzaANr61l9Mhn2D+VjL/Gt+ntcEG3K90DH8ITu39/a+oxtlY3g9oe7PLrv6FHcLXxdPWWZDXVcCNGZwJq7FOm6o04bsXKKnrplRWmw8mJGL1TLd6bAVgNJhY8srGQEUm6a9Sjx+PXZWM0bUjgy0mgkbTna7gyoGiCIGiCIGiCIGiCIGiCIGiCIGiKL/B1UoFjCfAoI8AAAAAElFTkSuQmCC" /><!-- --></p>
</div>
</div>
<div id="polygenic-model-univariate" class="section level2">
<h2>Polygenic model (univariate)</h2>
<p><code>solarPolygenic</code> function runs the polygenic analysis in
<code>SOLAR</code> and returns back the results in R. The basic call
looks like the standard call of <code>lm</code> function with the
formula interface.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>M1 <span class="ot">&lt;-</span> <span class="fu">solarPolygenic</span>(trait1 <span class="sc">~</span> age <span class="sc">+</span> sex, dat30)</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>M1</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a><span class="co">#&gt; Call: solarPolygenic(formula = trait1 ~ age + sex, data = dat30)</span></span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a><span class="co">#&gt; File polygenic.out:</span></span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a><span class="co">#&gt;  Pedigree:    dat.ped </span></span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a><span class="co">#&gt;  Phenotypes:  dat.phe </span></span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a><span class="co">#&gt;  Trait:       trait1                Individuals:  174 </span></span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb24-11"><a href="#cb24-11" tabindex="-1"></a><span class="co">#&gt;           H2r is 0.8061621  p = 6.1167535e-10  (Significant) </span></span>
<span id="cb24-12"><a href="#cb24-12" tabindex="-1"></a><span class="co">#&gt;         H2r Std. Error:  0.1100465 </span></span>
<span id="cb24-13"><a href="#cb24-13" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb24-14"><a href="#cb24-14" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb24-15"><a href="#cb24-15" tabindex="-1"></a><span class="co">#&gt;  Proportion of Variance Due to All Final Covariates Is </span></span>
<span id="cb24-16"><a href="#cb24-16" tabindex="-1"></a><span class="co">#&gt;                0.0330070 </span></span>
<span id="cb24-17"><a href="#cb24-17" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb24-18"><a href="#cb24-18" tabindex="-1"></a><span class="co">#&gt;  Loglikelihoods and chi&#39;s are in trait1/polygenic.logs.out </span></span>
<span id="cb24-19"><a href="#cb24-19" tabindex="-1"></a><span class="co">#&gt;  Best model is named poly and null0 </span></span>
<span id="cb24-20"><a href="#cb24-20" tabindex="-1"></a><span class="co">#&gt;  Final models are named poly, spor, nocovar </span></span>
<span id="cb24-21"><a href="#cb24-21" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb24-22"><a href="#cb24-22" tabindex="-1"></a><span class="co">#&gt;  Residual Kurtosis is -0.3603, within normal range</span></span></code></pre></div>
<p>The print method for the returned object of class
<code>solarPolygenic</code> shows the output produced by
<code>SOLAR</code>. The tables of the results are printed by
<code>summary</code> method.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="fu">summary</span>(M1)</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a><span class="co">#&gt; Call: solarPolygenic(formula = trait1 ~ age + sex, data = dat30)</span></span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a><span class="co">#&gt; File polygenic.out:</span></span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a><span class="co">#&gt;  Pedigree:    dat.ped </span></span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a><span class="co">#&gt;  Phenotypes:  dat.phe </span></span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a><span class="co">#&gt;  Trait:       trait1                Individuals:  174 </span></span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a><span class="co">#&gt;           H2r is 0.8061621  p = 6.1167535e-10  (Significant) </span></span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a><span class="co">#&gt;         H2r Std. Error:  0.1100465 </span></span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a><span class="co">#&gt;  Proportion of Variance Due to All Final Covariates Is </span></span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a><span class="co">#&gt;                0.0330070 </span></span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a><span class="co">#&gt;  Loglikelihoods and chi&#39;s are in trait1/polygenic.logs.out </span></span>
<span id="cb25-18"><a href="#cb25-18" tabindex="-1"></a><span class="co">#&gt;  Best model is named poly and null0 </span></span>
<span id="cb25-19"><a href="#cb25-19" tabindex="-1"></a><span class="co">#&gt;  Final models are named poly, spor, nocovar </span></span>
<span id="cb25-20"><a href="#cb25-20" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb25-21"><a href="#cb25-21" tabindex="-1"></a><span class="co">#&gt;  Residual Kurtosis is -0.3603, within normal range </span></span>
<span id="cb25-22"><a href="#cb25-22" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-23"><a href="#cb25-23" tabindex="-1"></a><span class="co">#&gt;  Loglikelihood Table:</span></span>
<span id="cb25-24"><a href="#cb25-24" tabindex="-1"></a><span class="co">#&gt;       model    loglik</span></span>
<span id="cb25-25"><a href="#cb25-25" tabindex="-1"></a><span class="co">#&gt; 1  sporadic -229.3348</span></span>
<span id="cb25-26"><a href="#cb25-26" tabindex="-1"></a><span class="co">#&gt; 2 polygenic -210.8689</span></span>
<span id="cb25-27"><a href="#cb25-27" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-28"><a href="#cb25-28" tabindex="-1"></a><span class="co">#&gt;  Covariates Table:</span></span>
<span id="cb25-29"><a href="#cb25-29" tabindex="-1"></a><span class="co">#&gt;   covariate      Estimate         SE Chi pval</span></span>
<span id="cb25-30"><a href="#cb25-30" tabindex="-1"></a><span class="co">#&gt; 1      mean  7.7823346631 0.33112422  NA   NA</span></span>
<span id="cb25-31"><a href="#cb25-31" tabindex="-1"></a><span class="co">#&gt; 2       age  0.0004591117 0.01483783  NA   NA</span></span>
<span id="cb25-32"><a href="#cb25-32" tabindex="-1"></a><span class="co">#&gt; 3       sex -0.4559509166 0.31250447  NA   NA</span></span>
<span id="cb25-33"><a href="#cb25-33" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-34"><a href="#cb25-34" tabindex="-1"></a><span class="co">#&gt;  Variance Components Table:</span></span>
<span id="cb25-35"><a href="#cb25-35" tabindex="-1"></a><span class="co">#&gt;   varcomp       Var        SE         pval</span></span>
<span id="cb25-36"><a href="#cb25-36" tabindex="-1"></a><span class="co">#&gt; 1     h2r 0.8061621 0.1100465 6.116753e-10</span></span>
<span id="cb25-37"><a href="#cb25-37" tabindex="-1"></a><span class="co">#&gt; 2      e2 0.1938379 0.1100465           NA</span></span></code></pre></div>
<div id="testing-covariates" class="section level3">
<h3>Testing covariates</h3>
<p>By default <code>SOLAR</code> does not perform the LRT tests for the
covariates. <code>covtest</code> argument of <code>solarPolygenic</code>
function tells to <code>SOLAR</code> to make this job.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>M2 <span class="ot">&lt;-</span> <span class="fu">solarPolygenic</span>(trait1 <span class="sc">~</span> age <span class="sc">+</span> sex, dat30, <span class="at">covtest =</span> <span class="cn">TRUE</span>)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>M2</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a><span class="co">#&gt; Call: solarPolygenic(formula = trait1 ~ age + sex, data = dat30, covtest = TRUE)</span></span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a><span class="co">#&gt; File polygenic.out:</span></span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a><span class="co">#&gt;  Pedigree:    dat.ped </span></span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a><span class="co">#&gt;  Phenotypes:  dat.phe </span></span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a><span class="co">#&gt;  Trait:       trait1                Individuals:  174 </span></span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a><span class="co">#&gt;           H2r is 0.8061621  p = 6.1167535e-10  (Significant) </span></span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a><span class="co">#&gt;         H2r Std. Error:  0.1100465 </span></span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a><span class="co">#&gt;                                       age  p = 0.9753082  (Not Sig., but fixed) </span></span>
<span id="cb26-15"><a href="#cb26-15" tabindex="-1"></a><span class="co">#&gt;                                       sex  p = 0.1455341  (Not Sig., but fixed) </span></span>
<span id="cb26-16"><a href="#cb26-16" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb26-17"><a href="#cb26-17" tabindex="-1"></a><span class="co">#&gt;  Proportion of Variance Due to All Final Covariates Is </span></span>
<span id="cb26-18"><a href="#cb26-18" tabindex="-1"></a><span class="co">#&gt;                0.0330070 </span></span>
<span id="cb26-19"><a href="#cb26-19" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb26-20"><a href="#cb26-20" tabindex="-1"></a><span class="co">#&gt;  Loglikelihoods and chi&#39;s are in trait1/polygenic.logs.out </span></span>
<span id="cb26-21"><a href="#cb26-21" tabindex="-1"></a><span class="co">#&gt;  Best model is named poly and null0 </span></span>
<span id="cb26-22"><a href="#cb26-22" tabindex="-1"></a><span class="co">#&gt;  Final models are named poly, spor, nocovar </span></span>
<span id="cb26-23"><a href="#cb26-23" tabindex="-1"></a><span class="co">#&gt;  Initial sporadic and polygenic models are s0 and p0 </span></span>
<span id="cb26-24"><a href="#cb26-24" tabindex="-1"></a><span class="co">#&gt;  Constrained covariate models are named no&lt;covariate name&gt; </span></span>
<span id="cb26-25"><a href="#cb26-25" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb26-26"><a href="#cb26-26" tabindex="-1"></a><span class="co">#&gt;  Residual Kurtosis is -0.3603, within normal range</span></span></code></pre></div>
<p>The test statistics and p-values are stored in <code>cf</code> slot
of the returned object <code>M2</code>.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>M2<span class="sc">$</span>cf</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a><span class="co">#&gt;   covariate      Estimate         SE    Chi      pval</span></span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a><span class="co">#&gt; 1      mean  7.7823346631 0.33112424     NA        NA</span></span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a><span class="co">#&gt; 2       age  0.0004591117 0.01483782 0.0010 0.9753082</span></span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a><span class="co">#&gt; 3       sex -0.4559509166 0.31250456 2.1184 0.1455341</span></span></code></pre></div>
</div>
<div id="custom-kinship-matrix" class="section level3">
<h3>Custom kinship matrix</h3>
<p>The custom kinship matrix is passed by <code>kinship</code> argument.
The following code shows such an example of <code>dat50</code> data set,
which phenotype data are stored in <code>phenodata</code>
data.frame.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>M3 <span class="ot">&lt;-</span> <span class="fu">solarPolygenic</span>(trait <span class="sc">~</span> <span class="dv">1</span>, phenodata, <span class="at">kinship =</span> kin)</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>M3</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a><span class="co">#&gt; Call: solarPolygenic(formula = trait ~ 1, data = phenodata, kinship = kin)</span></span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb28-6"><a href="#cb28-6" tabindex="-1"></a><span class="co">#&gt; File polygenic.out:</span></span>
<span id="cb28-7"><a href="#cb28-7" tabindex="-1"></a><span class="co">#&gt;  Pedigree:    dat.ped </span></span>
<span id="cb28-8"><a href="#cb28-8" tabindex="-1"></a><span class="co">#&gt;  Phenotypes:  dat.phe </span></span>
<span id="cb28-9"><a href="#cb28-9" tabindex="-1"></a><span class="co">#&gt;  Trait:       trait                 Individuals:  66 </span></span>
<span id="cb28-10"><a href="#cb28-10" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb28-11"><a href="#cb28-11" tabindex="-1"></a><span class="co">#&gt;           H2r is 0.1000000  p = 0.5000000  (Not Significant) </span></span>
<span id="cb28-12"><a href="#cb28-12" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb28-13"><a href="#cb28-13" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb28-14"><a href="#cb28-14" tabindex="-1"></a><span class="co">#&gt;  Loglikelihoods and chi&#39;s are in trait/polygenic.logs.out </span></span>
<span id="cb28-15"><a href="#cb28-15" tabindex="-1"></a><span class="co">#&gt;  Best model is named poly and null0 </span></span>
<span id="cb28-16"><a href="#cb28-16" tabindex="-1"></a><span class="co">#&gt;  Final models are named poly, spor </span></span>
<span id="cb28-17"><a href="#cb28-17" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb28-18"><a href="#cb28-18" tabindex="-1"></a><span class="co">#&gt;  Residual Kurtosis is -0.6742, within normal range</span></span></code></pre></div>
<p>The polygenic and sporadic models have the same LRT statistics, that
means <code>SOLAR</code> was not able to make the polygenic analysis in
a proper way.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a>M3<span class="sc">$</span>lf</span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a><span class="co">#&gt;       model    loglik</span></span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a><span class="co">#&gt; 1  sporadic -27.77137</span></span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a><span class="co">#&gt; 2 polygenic -27.77137</span></span></code></pre></div>
<p>This bug seems the case of unrelated individuals or the individuals
withount such ID fields as <code>fa</code> and <code>mo</code>.</p>
</div>
</div>
<div id="polygenic-model-bivariate" class="section level2">
<h2>Polygenic model (bivariate)</h2>
<p>The bivariate polygenic analysis is run in the same way as the
univariate one.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>B1 <span class="ot">&lt;-</span> <span class="fu">solarPolygenic</span>(trait1 <span class="sc">+</span> trait2 <span class="sc">~</span> <span class="dv">1</span>, dat30)</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>B1</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a><span class="co">#&gt; Call: solarPolygenic(formula = trait1 + trait2 ~ 1, data = dat30)</span></span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a><span class="co">#&gt; File polygenic.out:</span></span>
<span id="cb30-7"><a href="#cb30-7" tabindex="-1"></a><span class="co">#&gt;  Pedigree:    dat.ped </span></span>
<span id="cb30-8"><a href="#cb30-8" tabindex="-1"></a><span class="co">#&gt;  Phenotypes:  dat.phe </span></span>
<span id="cb30-9"><a href="#cb30-9" tabindex="-1"></a><span class="co">#&gt;  Trait:       trait1 trait2         Individuals:  174 </span></span>
<span id="cb30-10"><a href="#cb30-10" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb30-11"><a href="#cb30-11" tabindex="-1"></a><span class="co">#&gt;           H2r(trait1) is 0.8218823   </span></span>
<span id="cb30-12"><a href="#cb30-12" tabindex="-1"></a><span class="co">#&gt;         H2r(trait1) Std. Error:  0.1053258 </span></span>
<span id="cb30-13"><a href="#cb30-13" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb30-14"><a href="#cb30-14" tabindex="-1"></a><span class="co">#&gt;           H2r(trait2) is 0.6270026   </span></span>
<span id="cb30-15"><a href="#cb30-15" tabindex="-1"></a><span class="co">#&gt;         H2r(trait2) Std. Error:  0.1158107 </span></span>
<span id="cb30-16"><a href="#cb30-16" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb30-17"><a href="#cb30-17" tabindex="-1"></a><span class="co">#&gt;           RhoE is 0.4120487   </span></span>
<span id="cb30-18"><a href="#cb30-18" tabindex="-1"></a><span class="co">#&gt;         RhoE Std. Error:  0.1969379 </span></span>
<span id="cb30-19"><a href="#cb30-19" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb30-20"><a href="#cb30-20" tabindex="-1"></a><span class="co">#&gt;           RhoG is 0.9728759 </span></span>
<span id="cb30-21"><a href="#cb30-21" tabindex="-1"></a><span class="co">#&gt;         RhoG Std. Error:  0.0374442 </span></span>
<span id="cb30-22"><a href="#cb30-22" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb30-23"><a href="#cb30-23" tabindex="-1"></a><span class="co">#&gt;         Derived Estimate of RhoP is 0.8045957 </span></span>
<span id="cb30-24"><a href="#cb30-24" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb30-25"><a href="#cb30-25" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb30-26"><a href="#cb30-26" tabindex="-1"></a><span class="co">#&gt;  Loglikelihoods and chi&#39;s are in trait1.trait2/polygenic.logs.out </span></span>
<span id="cb30-27"><a href="#cb30-27" tabindex="-1"></a><span class="co">#&gt;  Best model is named poly and null0 </span></span>
<span id="cb30-28"><a href="#cb30-28" tabindex="-1"></a><span class="co">#&gt;  Final models are named poly, spor</span></span></code></pre></div>
<div id="testing-correlations" class="section level3">
<h3>Testing correlations</h3>
<p>The test of significance of environmental and genetic correlation
coefficients is indicated by <code>polygenic.options</code> argument of
<code>solarPolygenic</code> function.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>B2 <span class="ot">&lt;-</span> <span class="fu">solarPolygenic</span>(trait1 <span class="sc">+</span> trait2 <span class="sc">~</span> <span class="dv">1</span>, dat30, <span class="at">polygenic.options =</span> <span class="st">&quot;-testrhoe -testrhog&quot;</span>)</span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>B2</span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a><span class="co">#&gt; Call: solarPolygenic(formula = trait1 + trait2 ~ 1, data = dat30, polygenic.options = &quot;-testrhoe -testrhog&quot;)</span></span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a><span class="co">#&gt; File polygenic.out:</span></span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a><span class="co">#&gt;  Pedigree:    dat.ped </span></span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a><span class="co">#&gt;  Phenotypes:  dat.phe </span></span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a><span class="co">#&gt;  Trait:       trait1 trait2         Individuals:  174 </span></span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a><span class="co">#&gt;           H2r(trait1) is 0.8218823   </span></span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a><span class="co">#&gt;         H2r(trait1) Std. Error:  0.1053258 </span></span>
<span id="cb31-13"><a href="#cb31-13" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb31-14"><a href="#cb31-14" tabindex="-1"></a><span class="co">#&gt;           H2r(trait2) is 0.6270026   </span></span>
<span id="cb31-15"><a href="#cb31-15" tabindex="-1"></a><span class="co">#&gt;         H2r(trait2) Std. Error:  0.1158107 </span></span>
<span id="cb31-16"><a href="#cb31-16" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb31-17"><a href="#cb31-17" tabindex="-1"></a><span class="co">#&gt;           RhoE is 0.4120487  p = 0.1424023 </span></span>
<span id="cb31-18"><a href="#cb31-18" tabindex="-1"></a><span class="co">#&gt;         RhoE Std. Error:  0.1969379 </span></span>
<span id="cb31-19"><a href="#cb31-19" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb31-20"><a href="#cb31-20" tabindex="-1"></a><span class="co">#&gt;           RhoG is 0.9728759 </span></span>
<span id="cb31-21"><a href="#cb31-21" tabindex="-1"></a><span class="co">#&gt;         RhoG Std. Error:  0.0374442 </span></span>
<span id="cb31-22"><a href="#cb31-22" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb31-23"><a href="#cb31-23" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb31-24"><a href="#cb31-24" tabindex="-1"></a><span class="co">#&gt;         RhoG different from zero  p = 2.2463156e-09 </span></span>
<span id="cb31-25"><a href="#cb31-25" tabindex="-1"></a><span class="co">#&gt;         RhoG different from 1.0   p = 0.2098041 </span></span>
<span id="cb31-26"><a href="#cb31-26" tabindex="-1"></a><span class="co">#&gt;         Derived Estimate of RhoP is 0.8045957 </span></span>
<span id="cb31-27"><a href="#cb31-27" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb31-28"><a href="#cb31-28" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb31-29"><a href="#cb31-29" tabindex="-1"></a><span class="co">#&gt;  Loglikelihoods and chi&#39;s are in trait1.trait2/polygenic.logs.out </span></span>
<span id="cb31-30"><a href="#cb31-30" tabindex="-1"></a><span class="co">#&gt;  Best model is named poly and null0 </span></span>
<span id="cb31-31"><a href="#cb31-31" tabindex="-1"></a><span class="co">#&gt;  Final models are named poly, spor</span></span></code></pre></div>
<p>The results are stored in <code>vcf</code> slot of the returned
object.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>B2<span class="sc">$</span>vcf</span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a><span class="co">#&gt;       varcomp  Estimate         SE         Z        pvalZ</span></span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a><span class="co">#&gt; 1 h2r(trait1) 0.8218823 0.10532575  7.803242 5.215258e-03</span></span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a><span class="co">#&gt; 2  e2(trait1) 0.1781177 0.10532575  1.691112 1.934544e-01</span></span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a><span class="co">#&gt; 3 h2r(trait2) 0.6270026 0.11581068  5.414031 1.997553e-02</span></span>
<span id="cb32-6"><a href="#cb32-6" tabindex="-1"></a><span class="co">#&gt; 4  e2(trait2) 0.3729974 0.11581068  3.220751 7.271026e-02</span></span>
<span id="cb32-7"><a href="#cb32-7" tabindex="-1"></a><span class="co">#&gt; 5        rhog 0.9728759 0.03744417 25.982039 3.446085e-07</span></span>
<span id="cb32-8"><a href="#cb32-8" tabindex="-1"></a><span class="co">#&gt; 6        rhoe 0.4120487 0.19693793  2.092277 1.480453e-01</span></span></code></pre></div>
</div>
<div id="trait-specific-covariates" class="section level3">
<h3>Trait-specific covariates</h3>
<p>It might happen that traits in the multivariate polygenic analysis
have different covariates. Trait-specific covariates are passed to
<code>solarPolygenic</code> function according the specification of
<code>SOLAR</code>.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>B3 <span class="ot">&lt;-</span> <span class="fu">solarPolygenic</span>(trait1 <span class="sc">+</span> trait2 <span class="sc">~</span> sex <span class="sc">+</span> <span class="fu">age</span>(trait2), dat30)</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>B3</span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a><span class="co">#&gt; Call: solarPolygenic(formula = trait1 + trait2 ~ sex + age(trait2), </span></span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a><span class="co">#&gt;     data = dat30)</span></span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb33-7"><a href="#cb33-7" tabindex="-1"></a><span class="co">#&gt; File polygenic.out:</span></span>
<span id="cb33-8"><a href="#cb33-8" tabindex="-1"></a><span class="co">#&gt;  Pedigree:    dat.ped </span></span>
<span id="cb33-9"><a href="#cb33-9" tabindex="-1"></a><span class="co">#&gt;  Phenotypes:  dat.phe </span></span>
<span id="cb33-10"><a href="#cb33-10" tabindex="-1"></a><span class="co">#&gt;  Trait:       trait1 trait2         Individuals:  174 </span></span>
<span id="cb33-11"><a href="#cb33-11" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb33-12"><a href="#cb33-12" tabindex="-1"></a><span class="co">#&gt;           H2r(trait1) is 0.7956974   </span></span>
<span id="cb33-13"><a href="#cb33-13" tabindex="-1"></a><span class="co">#&gt;         H2r(trait1) Std. Error:  0.1129518 </span></span>
<span id="cb33-14"><a href="#cb33-14" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb33-15"><a href="#cb33-15" tabindex="-1"></a><span class="co">#&gt;           H2r(trait2) is 0.6029008   </span></span>
<span id="cb33-16"><a href="#cb33-16" tabindex="-1"></a><span class="co">#&gt;         H2r(trait2) Std. Error:  0.1197632 </span></span>
<span id="cb33-17"><a href="#cb33-17" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb33-18"><a href="#cb33-18" tabindex="-1"></a><span class="co">#&gt;           RhoE is 0.4655421   </span></span>
<span id="cb33-19"><a href="#cb33-19" tabindex="-1"></a><span class="co">#&gt;         RhoE Std. Error:  0.1736882 </span></span>
<span id="cb33-20"><a href="#cb33-20" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb33-21"><a href="#cb33-21" tabindex="-1"></a><span class="co">#&gt;           RhoG is 0.9676250 </span></span>
<span id="cb33-22"><a href="#cb33-22" tabindex="-1"></a><span class="co">#&gt;         RhoG Std. Error:  0.0397582 </span></span>
<span id="cb33-23"><a href="#cb33-23" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb33-24"><a href="#cb33-24" tabindex="-1"></a><span class="co">#&gt;         Derived Estimate of RhoP is 0.8027999 </span></span>
<span id="cb33-25"><a href="#cb33-25" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb33-26"><a href="#cb33-26" tabindex="-1"></a><span class="co">#&gt;  </span></span>
<span id="cb33-27"><a href="#cb33-27" tabindex="-1"></a><span class="co">#&gt;  Loglikelihoods and chi&#39;s are in trait1.trait2/polygenic.logs.out </span></span>
<span id="cb33-28"><a href="#cb33-28" tabindex="-1"></a><span class="co">#&gt;  Best model is named poly and null0 </span></span>
<span id="cb33-29"><a href="#cb33-29" tabindex="-1"></a><span class="co">#&gt;  Final models are named poly, spor</span></span></code></pre></div>
<p>In the results of this analysis, <code>sex</code> covariate is common
for both traits, while <code>age</code> covariate is applied only to
<code>trait2</code>. The given model specification can be ckecked by
looking at the model files from <code>SOLAR</code>.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="fu">tail</span>(B3<span class="sc">$</span>solar<span class="sc">$</span>files<span class="sc">$</span>model<span class="sc">$</span>poly.mod, <span class="dv">3</span>)</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;# mu = \\{t1*(&lt;Mean(trait1)&gt;+&lt;bsex(trait1)&gt;*Female) + t2*(&lt;Mean(trait2)&gt;&quot;</span></span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a><span class="co">#&gt; [2] &quot;# +&lt;bsex(trait2)&gt;*Female+&lt;bage(trait2)&gt;*(age-39.87931034))\\}&quot;           </span></span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a><span class="co">#&gt; [3] &quot;loglike set -361.499294&quot;</span></span></code></pre></div>
</div>
</div>
<div id="association-model" class="section level2">
<h2>Association model</h2>
<p><code>solarAssoc</code> function runs the association analysis in
<code>SOLAR</code>, in particular via <a href="http://helix.nih.gov/Documentation/solar-6.6.2-doc/91.appendix_1_text.html#mga">mga</a>
command. The genotype data are passed to <code>solarAssoc</code>
function in the form of SNP genotypes (<code>snpdata</code> argument) or
covariates (<code>snpcovdata</code> or <code>genocov.files</code>
arguments). The later case assumes that the user has converted the
genotypes to covariates by some method (outside
<code>solareclipser</code> package) before computing the association
model.</p>
<p>The function returns an object of class <code>solarAssoc</code>. This
class has additional methods such as <code>annotate</code>,
<code>plotQQ</code> and <code>plotManh</code> for exploration of the
association results. The main table of the results is <code>snpf</code>
slot of the returned object, which class is <code>data.table</code> from
the R package <code>data.table</code>.</p>
<p>The association analysis is shown with <code>dat50</code> data set
with 66 individuals and 50 SNP variants. The kinship matrix
<code>kin</code> of this data set is also used.</p>
<div id="snp-data-by-snpdata-argument" class="section level3">
<h3>SNP data by <code>snpdata</code> argument</h3>
<p>The option of using <code>snpdata</code> is recommended in the case
when the small number of SNPs is to be tested.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>A1 <span class="ot">&lt;-</span> <span class="fu">solarAssoc</span>(trait <span class="sc">~</span> <span class="dv">1</span>, phenodata, <span class="at">snpdata =</span> genodata, <span class="at">kinship =</span> kin)</span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;snp.geno-list&quot;</span></span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>A1</span>
<span id="cb35-4"><a href="#cb35-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb35-5"><a href="#cb35-5" tabindex="-1"></a><span class="co">#&gt; Call: solarAssoc(formula = trait ~ 1, data = phenodata, kinship = kin, </span></span>
<span id="cb35-6"><a href="#cb35-6" tabindex="-1"></a><span class="co">#&gt;     snpdata = genodata)</span></span>
<span id="cb35-7"><a href="#cb35-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb35-8"><a href="#cb35-8" tabindex="-1"></a><span class="co">#&gt;  Input SNP data:</span></span>
<span id="cb35-9"><a href="#cb35-9" tabindex="-1"></a><span class="co">#&gt;   *  50 SNP genotypes passed by `snpdata` argument</span></span>
<span id="cb35-10"><a href="#cb35-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb35-11"><a href="#cb35-11" tabindex="-1"></a><span class="co">#&gt;  Output results of association:</span></span>
<span id="cb35-12"><a href="#cb35-12" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb35-13"><a href="#cb35-13" tabindex="-1"></a><span class="co">#&gt;   *  Table of association results (first 5 out of 50 rows):</span></span>
<span id="cb35-14"><a href="#cb35-14" tabindex="-1"></a><span class="co">#&gt;    SNP NAv      chi     pSNP      bSNP   bSNPse   Varexp  est_maf   est_mac</span></span>
<span id="cb35-15"><a href="#cb35-15" tabindex="-1"></a><span class="co">#&gt; 1:  s1  66 0.189994 0.662922  0.160860 0.369045 0.002875 0.053031  7.000026</span></span>
<span id="cb35-16"><a href="#cb35-16" tabindex="-1"></a><span class="co">#&gt; 2: s10  66 0.050972 0.821380 -0.105065 0.465364 0.000772 0.015152  1.999998</span></span>
<span id="cb35-17"><a href="#cb35-17" tabindex="-1"></a><span class="co">#&gt; 3: s11  66 0.050972 0.821380 -0.105065 0.465364 0.000772 0.015152  1.999998</span></span>
<span id="cb35-18"><a href="#cb35-18" tabindex="-1"></a><span class="co">#&gt; 4: s12  66 0.909920 0.340136 -0.196598 0.206100 0.013692 0.204545 27.000006</span></span>
<span id="cb35-19"><a href="#cb35-19" tabindex="-1"></a><span class="co">#&gt; 5: s13  66 0.182890 0.668901  0.118876 0.277971 0.002767 0.106061 13.999986</span></span>
<span id="cb35-20"><a href="#cb35-20" tabindex="-1"></a><span class="co">#&gt;    dosage_sd</span></span>
<span id="cb35-21"><a href="#cb35-21" tabindex="-1"></a><span class="co">#&gt; 1:  0.307915</span></span>
<span id="cb35-22"><a href="#cb35-22" tabindex="-1"></a><span class="co">#&gt; 2:  0.244311</span></span>
<span id="cb35-23"><a href="#cb35-23" tabindex="-1"></a><span class="co">#&gt; 3:  0.244311</span></span>
<span id="cb35-24"><a href="#cb35-24" tabindex="-1"></a><span class="co">#&gt; 4:  0.549856</span></span>
<span id="cb35-25"><a href="#cb35-25" tabindex="-1"></a><span class="co">#&gt; 5:  0.408810</span></span>
<span id="cb35-26"><a href="#cb35-26" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb35-27"><a href="#cb35-27" tabindex="-1"></a><span class="co">#&gt;  CPU time on 1 core(s): 00:00:01</span></span></code></pre></div>
<p><code>summary</code> method reports the number of significant SNPs
based on Bonferroni correction.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="fu">summary</span>(A1)</span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a><span class="co">#&gt; Call: solarAssoc(formula = trait ~ 1, data = phenodata, kinship = kin, </span></span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a><span class="co">#&gt;     snpdata = genodata)</span></span>
<span id="cb36-5"><a href="#cb36-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb36-6"><a href="#cb36-6" tabindex="-1"></a><span class="co">#&gt; Association model</span></span>
<span id="cb36-7"><a href="#cb36-7" tabindex="-1"></a><span class="co">#&gt;  * Number of SNPs: 50 </span></span>
<span id="cb36-8"><a href="#cb36-8" tabindex="-1"></a><span class="co">#&gt;  * Input format: snpdata </span></span>
<span id="cb36-9"><a href="#cb36-9" tabindex="-1"></a><span class="co">#&gt;  * Number of significal SNPs: 0 (Bonferroni correction with alpha 0.05)</span></span></code></pre></div>
<p>The significance threshold can be passed as <code>alpha</code>
parameter of <code>summary</code> method.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="fu">summary</span>(A1, <span class="at">alpha =</span> <span class="dv">1</span>)</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a><span class="co">#&gt; Call: solarAssoc(formula = trait ~ 1, data = phenodata, kinship = kin, </span></span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a><span class="co">#&gt;     snpdata = genodata)</span></span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a><span class="co">#&gt; Association model</span></span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a><span class="co">#&gt;  * Number of SNPs: 50 </span></span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a><span class="co">#&gt;  * Input format: snpdata </span></span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a><span class="co">#&gt;  * Number of significal SNPs: 1 (Bonferroni correction with alpha 1)</span></span>
<span id="cb37-10"><a href="#cb37-10" tabindex="-1"></a><span class="co">#&gt;    SNP NAv      chi    pSNP      bSNP   bSNPse   Varexp  est_maf  est_mac</span></span>
<span id="cb37-11"><a href="#cb37-11" tabindex="-1"></a><span class="co">#&gt; 1: s43  66 6.504508 0.01076 -1.186139 0.465081 0.093852 0.030303 3.999996</span></span>
<span id="cb37-12"><a href="#cb37-12" tabindex="-1"></a><span class="co">#&gt;    dosage_sd</span></span>
<span id="cb37-13"><a href="#cb37-13" tabindex="-1"></a><span class="co">#&gt; 1:  0.238606</span></span></code></pre></div>
</div>
<div id="snp-data-by-snpcovdata-argument" class="section level3">
<h3>SNP data by <code>snpcovdata</code> argument</h3>
<p>SNP covariates are passed to the association model by
<code>snpcovdata</code> parameter.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>A2 <span class="ot">&lt;-</span> <span class="fu">solarAssoc</span>(trait <span class="sc">~</span> <span class="dv">1</span>, phenodata, <span class="at">snpcovdata =</span> genocovdata, <span class="at">kinship =</span> kin)</span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;snp.geno-list&quot;</span></span>
<span id="cb38-3"><a href="#cb38-3" tabindex="-1"></a>A2</span>
<span id="cb38-4"><a href="#cb38-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb38-5"><a href="#cb38-5" tabindex="-1"></a><span class="co">#&gt; Call: solarAssoc(formula = trait ~ 1, data = phenodata, kinship = kin, </span></span>
<span id="cb38-6"><a href="#cb38-6" tabindex="-1"></a><span class="co">#&gt;     snpcovdata = genocovdata)</span></span>
<span id="cb38-7"><a href="#cb38-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb38-8"><a href="#cb38-8" tabindex="-1"></a><span class="co">#&gt;  Input SNP data:</span></span>
<span id="cb38-9"><a href="#cb38-9" tabindex="-1"></a><span class="co">#&gt;   *  50 SNP covariates passed by `snpcovdata` argument</span></span>
<span id="cb38-10"><a href="#cb38-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb38-11"><a href="#cb38-11" tabindex="-1"></a><span class="co">#&gt;  Output results of association:</span></span>
<span id="cb38-12"><a href="#cb38-12" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb38-13"><a href="#cb38-13" tabindex="-1"></a><span class="co">#&gt;   *  Table of association results (first 5 out of 50 rows):</span></span>
<span id="cb38-14"><a href="#cb38-14" tabindex="-1"></a><span class="co">#&gt;    SNP NAv      chi     pSNP      bSNP   bSNPse   Varexp  est_maf   est_mac</span></span>
<span id="cb38-15"><a href="#cb38-15" tabindex="-1"></a><span class="co">#&gt; 1:  s1  66 0.189994 0.662922  0.160860 0.369045 0.002875 0.053031  7.000026</span></span>
<span id="cb38-16"><a href="#cb38-16" tabindex="-1"></a><span class="co">#&gt; 2: s10  66 0.050972 0.821380 -0.105065 0.465364 0.000772 0.015152  1.999998</span></span>
<span id="cb38-17"><a href="#cb38-17" tabindex="-1"></a><span class="co">#&gt; 3: s11  66 0.050972 0.821380 -0.105065 0.465364 0.000772 0.015152  1.999998</span></span>
<span id="cb38-18"><a href="#cb38-18" tabindex="-1"></a><span class="co">#&gt; 4: s12  66 0.909920 0.340136 -0.196598 0.206100 0.013692 0.204545 27.000006</span></span>
<span id="cb38-19"><a href="#cb38-19" tabindex="-1"></a><span class="co">#&gt; 5: s13  66 0.182890 0.668901  0.118876 0.277971 0.002767 0.106061 13.999986</span></span>
<span id="cb38-20"><a href="#cb38-20" tabindex="-1"></a><span class="co">#&gt;    dosage_sd</span></span>
<span id="cb38-21"><a href="#cb38-21" tabindex="-1"></a><span class="co">#&gt; 1:  0.307915</span></span>
<span id="cb38-22"><a href="#cb38-22" tabindex="-1"></a><span class="co">#&gt; 2:  0.244311</span></span>
<span id="cb38-23"><a href="#cb38-23" tabindex="-1"></a><span class="co">#&gt; 3:  0.244311</span></span>
<span id="cb38-24"><a href="#cb38-24" tabindex="-1"></a><span class="co">#&gt; 4:  0.549856</span></span>
<span id="cb38-25"><a href="#cb38-25" tabindex="-1"></a><span class="co">#&gt; 5:  0.408810</span></span>
<span id="cb38-26"><a href="#cb38-26" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb38-27"><a href="#cb38-27" tabindex="-1"></a><span class="co">#&gt;  CPU time on 1 core(s): 00:00:01</span></span></code></pre></div>
<p>The values of the SNP covariates can take any numerical values. That
means dosage format of imputed SNP data is allowed.</p>
<p>The previous example is repeated for the SNP covariate data, where
values of <code>2</code> are replaced by <code>1.9</code>.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a>genocovdata2 <span class="ot">&lt;-</span> genocovdata</span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a>genocovdata2[genocovdata2 <span class="sc">==</span> <span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fl">1.9</span></span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a></span>
<span id="cb39-4"><a href="#cb39-4" tabindex="-1"></a>A2 <span class="ot">&lt;-</span> <span class="fu">solarAssoc</span>(trait <span class="sc">~</span> <span class="dv">1</span>, phenodata, <span class="at">snpcovdata =</span> genocovdata2, <span class="at">kinship =</span> kin)</span>
<span id="cb39-5"><a href="#cb39-5" tabindex="-1"></a><span class="co">#&gt; [1] &quot;snp.geno-list&quot;</span></span>
<span id="cb39-6"><a href="#cb39-6" tabindex="-1"></a>A2</span>
<span id="cb39-7"><a href="#cb39-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb39-8"><a href="#cb39-8" tabindex="-1"></a><span class="co">#&gt; Call: solarAssoc(formula = trait ~ 1, data = phenodata, kinship = kin, </span></span>
<span id="cb39-9"><a href="#cb39-9" tabindex="-1"></a><span class="co">#&gt;     snpcovdata = genocovdata2)</span></span>
<span id="cb39-10"><a href="#cb39-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb39-11"><a href="#cb39-11" tabindex="-1"></a><span class="co">#&gt;  Input SNP data:</span></span>
<span id="cb39-12"><a href="#cb39-12" tabindex="-1"></a><span class="co">#&gt;   *  50 SNP covariates passed by `snpcovdata` argument</span></span>
<span id="cb39-13"><a href="#cb39-13" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb39-14"><a href="#cb39-14" tabindex="-1"></a><span class="co">#&gt;  Output results of association:</span></span>
<span id="cb39-15"><a href="#cb39-15" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb39-16"><a href="#cb39-16" tabindex="-1"></a><span class="co">#&gt;   *  Table of association results (first 5 out of 50 rows):</span></span>
<span id="cb39-17"><a href="#cb39-17" tabindex="-1"></a><span class="co">#&gt;    SNP NAv      chi     pSNP      bSNP   bSNPse   Varexp  est_maf   est_mac</span></span>
<span id="cb39-18"><a href="#cb39-18" tabindex="-1"></a><span class="co">#&gt; 1:  s1  66 0.189994 0.662922  0.160860 0.369045 0.002875 0.053031  7.000026</span></span>
<span id="cb39-19"><a href="#cb39-19" tabindex="-1"></a><span class="co">#&gt; 2: s10  66 0.050972 0.821380 -0.110595 0.489857 0.000772 0.014394  1.900008</span></span>
<span id="cb39-20"><a href="#cb39-20" tabindex="-1"></a><span class="co">#&gt; 3: s11  66 0.050972 0.821380 -0.110595 0.489857 0.000772 0.014394  1.900008</span></span>
<span id="cb39-21"><a href="#cb39-21" tabindex="-1"></a><span class="co">#&gt; 4: s12  66 0.927276 0.335572 -0.201592 0.209348 0.013951 0.203031 26.800026</span></span>
<span id="cb39-22"><a href="#cb39-22" tabindex="-1"></a><span class="co">#&gt; 5: s13  66 0.182890 0.668901  0.118876 0.277971 0.002767 0.106061 13.999986</span></span>
<span id="cb39-23"><a href="#cb39-23" tabindex="-1"></a><span class="co">#&gt;    dosage_sd</span></span>
<span id="cb39-24"><a href="#cb39-24" tabindex="-1"></a><span class="co">#&gt; 1:  0.307915</span></span>
<span id="cb39-25"><a href="#cb39-25" tabindex="-1"></a><span class="co">#&gt; 2:  0.232095</span></span>
<span id="cb39-26"><a href="#cb39-26" tabindex="-1"></a><span class="co">#&gt; 3:  0.232095</span></span>
<span id="cb39-27"><a href="#cb39-27" tabindex="-1"></a><span class="co">#&gt; 4:  0.541289</span></span>
<span id="cb39-28"><a href="#cb39-28" tabindex="-1"></a><span class="co">#&gt; 5:  0.408810</span></span>
<span id="cb39-29"><a href="#cb39-29" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb39-30"><a href="#cb39-30" tabindex="-1"></a><span class="co">#&gt;  CPU time on 1 core(s): 00:00:01</span></span></code></pre></div>
</div>
<div id="snp-data-by-genocov.files-single-value" class="section level3">
<h3>SNP data by <code>genocov.files</code> (single value)</h3>
<p><code>genocov.files</code> argument is supported to pass the SNP
covariate data in files of the appropriate <code>SOLAR</code> format.
This option suits for large-scale calculations, when the real
genome-wide scan is run. Both arguments <code>genocov.files</code> and
<code>snplists.files</code> are required, and <code>snpmap.files</code>
argument is optional.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a>dir <span class="ot">&lt;-</span> <span class="fu">package.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;solarAssoc&quot;</span>, <span class="at">package =</span> <span class="st">&quot;solareclipser&quot;</span>)</span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a>genocov.files <span class="ot">&lt;-</span> <span class="fu">file.path</span>(dir, <span class="st">&quot;snp.genocov&quot;</span>)</span>
<span id="cb40-3"><a href="#cb40-3" tabindex="-1"></a>snplists.files <span class="ot">&lt;-</span> <span class="fu">file.path</span>(dir, <span class="fu">c</span>(<span class="st">&quot;snp.geno-list1&quot;</span>, <span class="st">&quot;snp.geno-list2&quot;</span>))</span>
<span id="cb40-4"><a href="#cb40-4" tabindex="-1"></a></span>
<span id="cb40-5"><a href="#cb40-5" tabindex="-1"></a>A3 <span class="ot">&lt;-</span> <span class="fu">solarAssoc</span>(trait <span class="sc">~</span> <span class="dv">1</span>, phenodata, </span>
<span id="cb40-6"><a href="#cb40-6" tabindex="-1"></a>  <span class="at">genocov.files =</span> genocov.files, <span class="at">snplists.files =</span> snplists.files)</span>
<span id="cb40-7"><a href="#cb40-7" tabindex="-1"></a>A3</span>
<span id="cb40-8"><a href="#cb40-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-9"><a href="#cb40-9" tabindex="-1"></a><span class="co">#&gt; Call: solarAssoc(formula = trait ~ 1, data = phenodata, genocov.files = genocov.files, </span></span>
<span id="cb40-10"><a href="#cb40-10" tabindex="-1"></a><span class="co">#&gt;     snplists.files = snplists.files)</span></span>
<span id="cb40-11"><a href="#cb40-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-12"><a href="#cb40-12" tabindex="-1"></a><span class="co">#&gt;  Input SNP data:</span></span>
<span id="cb40-13"><a href="#cb40-13" tabindex="-1"></a><span class="co">#&gt;   *  SNP covariates passed in 1 file(s) by `genocov.files` argument</span></span>
<span id="cb40-14"><a href="#cb40-14" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-15"><a href="#cb40-15" tabindex="-1"></a><span class="co">#&gt;  Output results of association:</span></span>
<span id="cb40-16"><a href="#cb40-16" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-17"><a href="#cb40-17" tabindex="-1"></a><span class="co">#&gt;   *  Table of association results (first 5 out of 25 rows):</span></span>
<span id="cb40-18"><a href="#cb40-18" tabindex="-1"></a><span class="co">#&gt;    SNP NAv      chi     pSNP      bSNP   bSNPse   Varexp  est_maf   est_mac</span></span>
<span id="cb40-19"><a href="#cb40-19" tabindex="-1"></a><span class="co">#&gt; 1:  s1  66 0.189994 0.662922  0.160860 0.369045 0.002875 0.053031  7.000026</span></span>
<span id="cb40-20"><a href="#cb40-20" tabindex="-1"></a><span class="co">#&gt; 2: s10  66 0.050972 0.821380 -0.105065 0.465364 0.000772 0.015152  1.999998</span></span>
<span id="cb40-21"><a href="#cb40-21" tabindex="-1"></a><span class="co">#&gt; 3: s11  66 0.050972 0.821380 -0.105065 0.465364 0.000772 0.015152  1.999998</span></span>
<span id="cb40-22"><a href="#cb40-22" tabindex="-1"></a><span class="co">#&gt; 4: s12  66 0.909920 0.340136 -0.196598 0.206100 0.013692 0.204545 27.000006</span></span>
<span id="cb40-23"><a href="#cb40-23" tabindex="-1"></a><span class="co">#&gt; 5: s13  66 0.182890 0.668901  0.118876 0.277971 0.002767 0.106061 13.999986</span></span>
<span id="cb40-24"><a href="#cb40-24" tabindex="-1"></a><span class="co">#&gt;    dosage_sd</span></span>
<span id="cb40-25"><a href="#cb40-25" tabindex="-1"></a><span class="co">#&gt; 1:  0.307915</span></span>
<span id="cb40-26"><a href="#cb40-26" tabindex="-1"></a><span class="co">#&gt; 2:  0.244311</span></span>
<span id="cb40-27"><a href="#cb40-27" tabindex="-1"></a><span class="co">#&gt; 3:  0.244311</span></span>
<span id="cb40-28"><a href="#cb40-28" tabindex="-1"></a><span class="co">#&gt; 4:  0.549856</span></span>
<span id="cb40-29"><a href="#cb40-29" tabindex="-1"></a><span class="co">#&gt; 5:  0.408810</span></span>
<span id="cb40-30"><a href="#cb40-30" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-31"><a href="#cb40-31" tabindex="-1"></a><span class="co">#&gt;  CPU time on 1 core(s): 00:00:00</span></span></code></pre></div>
</div>
<div id="snp-data-by-genocov.files-many-values" class="section level3">
<h3>SNP data by <code>genocov.files</code> (many values)</h3>
<p>Many values can be passed to <code>genocov.files</code> parameter,
but this case is thought for parallel calculations.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a>dir <span class="ot">&lt;-</span> <span class="fu">package.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;solarAssoc&quot;</span>, <span class="at">package =</span> <span class="st">&quot;solareclipser&quot;</span>)</span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>genocov.files <span class="ot">&lt;-</span> <span class="fu">file.path</span>(dir, <span class="fu">c</span>(<span class="st">&quot;snp.genocov1&quot;</span>, <span class="st">&quot;snp.genocov2&quot;</span>))</span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a>snplists.files <span class="ot">&lt;-</span> <span class="fu">file.path</span>(dir, <span class="fu">c</span>(<span class="st">&quot;snp.geno-list1&quot;</span>, <span class="st">&quot;snp.geno-list2&quot;</span>))</span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a>A4 <span class="ot">&lt;-</span> <span class="fu">solarAssoc</span>(trait <span class="sc">~</span> <span class="dv">1</span>, phenodata, </span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a>  <span class="at">genocov.files =</span> genocov.files, <span class="at">snplists.files =</span> snplists.files)</span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a>A4</span>
<span id="cb41-8"><a href="#cb41-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb41-9"><a href="#cb41-9" tabindex="-1"></a><span class="co">#&gt; Call: solarAssoc(formula = trait ~ 1, data = phenodata, genocov.files = genocov.files, </span></span>
<span id="cb41-10"><a href="#cb41-10" tabindex="-1"></a><span class="co">#&gt;     snplists.files = snplists.files)</span></span>
<span id="cb41-11"><a href="#cb41-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb41-12"><a href="#cb41-12" tabindex="-1"></a><span class="co">#&gt;  Input SNP data:</span></span>
<span id="cb41-13"><a href="#cb41-13" tabindex="-1"></a><span class="co">#&gt;   *  SNP covariates passed in 2 file(s) by `genocov.files` argument and 2 files(s) by `snplists.files` argument</span></span>
<span id="cb41-14"><a href="#cb41-14" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb41-15"><a href="#cb41-15" tabindex="-1"></a><span class="co">#&gt;  Output results of association:</span></span>
<span id="cb41-16"><a href="#cb41-16" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb41-17"><a href="#cb41-17" tabindex="-1"></a><span class="co">#&gt;   *  Table of association results (first 5 out of 25 rows):</span></span>
<span id="cb41-18"><a href="#cb41-18" tabindex="-1"></a><span class="co">#&gt;    SNP NAv      chi     pSNP      bSNP   bSNPse   Varexp  est_maf   est_mac</span></span>
<span id="cb41-19"><a href="#cb41-19" tabindex="-1"></a><span class="co">#&gt; 1:  s1  66 0.189994 0.662922  0.160860 0.369045 0.002875 0.053031  7.000026</span></span>
<span id="cb41-20"><a href="#cb41-20" tabindex="-1"></a><span class="co">#&gt; 2: s10  66 0.050972 0.821380 -0.105065 0.465364 0.000772 0.015152  1.999998</span></span>
<span id="cb41-21"><a href="#cb41-21" tabindex="-1"></a><span class="co">#&gt; 3: s11  66 0.050972 0.821380 -0.105065 0.465364 0.000772 0.015152  1.999998</span></span>
<span id="cb41-22"><a href="#cb41-22" tabindex="-1"></a><span class="co">#&gt; 4: s12  66 0.909920 0.340136 -0.196598 0.206100 0.013692 0.204545 27.000006</span></span>
<span id="cb41-23"><a href="#cb41-23" tabindex="-1"></a><span class="co">#&gt; 5: s13  66 0.182890 0.668901  0.118876 0.277971 0.002767 0.106061 13.999986</span></span>
<span id="cb41-24"><a href="#cb41-24" tabindex="-1"></a><span class="co">#&gt;    dosage_sd</span></span>
<span id="cb41-25"><a href="#cb41-25" tabindex="-1"></a><span class="co">#&gt; 1:  0.307915</span></span>
<span id="cb41-26"><a href="#cb41-26" tabindex="-1"></a><span class="co">#&gt; 2:  0.244311</span></span>
<span id="cb41-27"><a href="#cb41-27" tabindex="-1"></a><span class="co">#&gt; 3:  0.244311</span></span>
<span id="cb41-28"><a href="#cb41-28" tabindex="-1"></a><span class="co">#&gt; 4:  0.549856</span></span>
<span id="cb41-29"><a href="#cb41-29" tabindex="-1"></a><span class="co">#&gt; 5:  0.408810</span></span>
<span id="cb41-30"><a href="#cb41-30" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb41-31"><a href="#cb41-31" tabindex="-1"></a><span class="co">#&gt;  CPU time on 1 core(s), 2 batches: 00:00:00</span></span></code></pre></div>
</div>
<div id="exploration-of-association-results" class="section level3">
<h3>Exploration of association results</h3>
<p>QQ and Manhattan plots help to explore the association results
visually.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a>A5 <span class="ot">&lt;-</span> <span class="fu">solarAssoc</span>(trait <span class="sc">~</span> <span class="dv">1</span>, phenodata, <span class="at">snpdata =</span> genodata, <span class="at">snpmap =</span> snpdata, <span class="at">kinship =</span> kin)</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;snp.geno-list&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a><span class="fu">plot</span>(A5)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAADAFBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////isF19AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAWkUlEQVR4nO2dCXwURfbHX0IgCYQbQxwimCyXgOFOArKigMKaqIjKssilBAQU5IocnlxeYREQ3ERQDhFQ+CMgARZQQFTEZVdAVxABObKYICAYyEGO+ndVz9E9fVT1dI/TmdTv88lMd/Xr1zXf9F3vVQHi0hUEugJ2FwdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEkQ0AfTl+/FHVBX0gQmepqLfTfvRPrVyyAaBMgO2qCwggzaVE30bAHj9Vyyk7A8rOXKIL6KfsCVEQ/IDGxgG0HI46QWzRkFq5qHzNHdGRzUafReIe5FyqqlTACnpAvfCvTMaAngLIRa+TXw1NfxcBOZeqamFaWnIlAOQ8iDpBGHRKvVxSHeJ3fP4owCcq56Ddr+wpRaikyLPu/MoECIQzDjqflrYZoWUAC5SANgg7U9LPaHhjz7qVClB4OZm9tHbKfZEA85WA7mtybn5Y9OPVBnrWrVSAGpG5hdUAQpupAsragND+pMg7TnvWrVSAYvHMkVDovO3aAVVAKqosgLa5Af0DYAMpkgDapr1upQC0BGBGvgvQSoDe+9ZEA7zpBCQu1VKlALTPdR+EZ3Lr4hufFgCjnYD2ad8HoUoCCD3fsHp/FyB04I6o22flhkFovghIXKqlygHI1uKAKOKAKOKAKOKAKOKAKOKAKOKAKOKAKAoUoP+u/x59L/whdPnFpOioNgO+waWpAM/i70MAfRGaTF6+hsY9dETTTVmE+IZ2qTB9aXTr6h0/srqigQI0GN5DA4U/dKqB+BNhNiKAqh1HckCCquzTcnMc3IBONcMTIYstrmigALWGw+g24Q91B3hk+YqxkRB6QGynSEESQCMzM1/tCNBBy83HkJCJdRShFAidtzMJ4D/WVjRAgK5XCS+5HhpegoqqQg9csAZgjrMhJ1sCaL2w6FoDCC1CKLtHTFSbKb/K/cyGSc6pX0NhJEKnnceodQoIoEzXsQNpFwASS4WiolWr9hNAcdC8WA4IdQU4h9/jYzW7IhS0BLgmevobDOsa1XqKwG8HwAqhIFrkbZ0CAmhFTA2IEv5qxExE8QBN09edEF/YC4DWVoU35ICu3wQRZagxxGw/MBYgA0kBJYjYkkvRUtJSJBy2zayta4AOsaHCaXUoZAlTXzUiP7HOI58hAujQRKj5ixvQqKVL5ybi81IhQMJxVDx9Kr5M9W3XroC4KQ2HZlv33QnwDloAgD10Boe1NQ0QoLbwDfkTdG11WkJVzOgtEdDVaBjmdRWLEs7B+BrVOv2zUpmbG3v2/A+h3HCB4HwRUCLcbG1NAwPoRtXQghtVwwpd88VfDhcwlBFAwsESkiUF1KjfCcHk0B1kJv5bFXdtoCl+d71ZmGwlTFqqgACKcu0aLdHR117bT8ruBcgRAZV1hJtkJ2mnTr3ZvQpAktTR7zk5JQifeNqi7YDvqtBNQXGSjo+C6jH4rxv6EaA7PmzKE6FGqQgIfYnZeQE6PHfuWYTOxUMdqaOVpL36ZBgMRL9FwDCEvgeYbG1dA3OIjYF5aJzwJ6g9QJdF/5fVA6A/cgJCjykBCdBSDh5dWw+6CzMPd+oknqRPVgHH7CXNyOmnP0D6kpYQonYMmlBgACXDHtRNbI84fbPzcGtzxQ0op4YCUJkYDARVv0DSy/xMsXQ8XqmF+4nFSgUEUGlkyJWyqJArZOZaRrfY8Piey/DJxAkIzVGegwoXJcdUa9LvIJ72AEKb7mwQ3fNjMpk38KY6966zuq4BAfRzm3vRWeGvIoi/D6KIA6KIA6KIA6KIA6KIA6KIA6KIA6KIA6KIA6KIA6KIA6KIA6KIA6KIGdBbSVEtMkqcM+s7177b4jd3dhUroFkwccvUsBfEmS0ho9f1qXHWb5WykxgBFdcaJ3xOihSbpe7ug1DBLdP9VysbiRHQSdghfK6HU3jmMrwrfI6K81+tbCRGQEUncB7khEjS1Pc9fCV8Lggp9l+17CMjV7FVYenkexfgVP/3wR2Ksr+jS7EHLK6f5frB4XB8foPdnh3QhcEwTLyK7YRjCDfaXXYtKjzo0u1fsm86ILrRUgDkeJ99BWZA2dFxG52TR+Br4XNhuIpV8n72TQdEpzEfx4vsK7ACyq4yxh1qcClkpfA59k8qZrYHVNIKA/qAfQVGQCWOwZK5Hv2EkvgpKna2B4TWC3wGlNDtXGIE9Ck8uxyrEGUNKEJbq8z4YmDdUyp29geEcjZ/XW7AnBGQK6owF6XhVt91ibV7qj5qVABABmXxwyoHRBEHRBEHRBEHRBEHRBEHRBEHRBEHRBEHRBEHRJGtAP3z6QnmqxPEgFY5HDc7PjPrJYgB9cBvfgbT7fTlBJRv4A2JnuwEqCMG9IBZL4AK1w5vFg4h9XrM+q/5StkJ0EQMaJ5ZL5BeH6BqTKu4WgDQ/ROz7uwE6Gqao2m6gfYLdQEkz/8X8fLLhlF1oadO3807N7mmNogJuSo2dgKEULEFJw7SguNS0cqmL2laliW6M7HmRuNcdbXOw+wFyAp5XcVK1F40Y51bfKcnVW1MTy13QQ9IU1u7dYtwA+ozUsssWAFd2b7xeCnFEjV1A2reu0ONtllqNkEKaG894ZxbP6NA39QNqKxa/QUb02Cue8nuui6FbVJft+KKAGrbfO+FT2fUS9YP13ADKl57UvgcUqvMtaT8skudg3MPqr0Gf16Ie07XtKk8n3gDnFDaBOkh1keMdljeVtfUDSjvIG6a3Ay5ShtbAdo/M8N8mCABtLHNVfy1tbquqRvQTsCN/082VrGxE6DV+FHjGN1OX+I56OYm715Eed1TdU1FQFkDikqTomdljwtVSzC2E6AkDGgS3U5fBNCcBx0ANaDhLt0nFxEQbpsvGN+yZlfVbrBtBKj8TxjQILNuXDeKOR9P61UHIro8Y+7pzkaA0F8xoLfMepHeSZf/+P645Oum3NkJ0Ol7HI5Ryv/3xSWGQuCD+I0iKj9zSVmYLuxW+ldrubwBmduB7AVITVdJjOIY9hVAdvd8pO/L5rZve0ArCKCO7CtA/ae/cI6V83NmV2i029z2bQ/oBwKoN/sKcBdAtfZ/Gdive0OAei+ZPMLsDwglYEAn2e0BffdM6xD8BrV2ynuFdHuK7A8IjW7f62cD5uQkffnwZ/tPUt8HsagCADKoYL7MWyIPoO9fShv6vNmeYoMY0IKWM957b2br+ebc2R/QtRl33W+k+c8NKP53/FnQxtzmVQCdfPfD3805tVTj8FVsL7u9G1DzM/gzz8hduIqUgLKF+nRWebUWIJU0x4Amsq/gBrQhdtBzzw1xmOxFTgmoC67QS+a8WigR0AT2FTwn6Usr58xa/quOqafpWTstXAGolNy5DmWvkHU6M2PcR2WK0rG4PgZGlGK/zEuanrXTwpV7UG9cob+zV8gynW2nmjmX//KdKZsNuGEFJGt61k4LVwL6toPD8ZDZJxhf9AbZd00nHrsAXUm4VZSGnbTpWSctXOUqdv2rI0bysyzTNAJI75zBJPcedLXlhhwsbVN3q4ZOWriN7oO2Yj69FMVFi4dNO61iriXPITbnIMXUDUiRFl7280mnOtgHEPp7c8dfflKUDhewtTdw22HgWUzSLuaVFr4n3qWIjerrBkQlV5VleeTAe4fdiS+AKnJa+E8E0Fy6oUsEUF9R/YbMOa9j6gZUkdPCy7piQP9mX4EAGh8elpiaXLVR71vD/6lt6gleqMhp4UdTHe3WGrAngKZ2+J/w+UvSB+UzWmibupueK3ZaODLWNEoA3SJ2/L3xNuHo0R422NP0XKHTwj+Z+LKBV9IioMbiWf3dGPQTmHs1oQZI+TwUSL2Jz0HH2e0JoGl1V+ej/LX1Jp1J0Ryoik1KQHlD4tosNOfUSpXfhgE9z74CAVQ6thpEQpUnb2Q2NvnOVQnob7hC9rk7KiKX+VHsKzjvg85vWrReOOnmm31qUgAqJhUaa9KtjypSno9FQFPZfbhuFIvPmG8UQyqAylrgCk2zwrdRXXnc4Zjg/bx4lAB6ld2LCGhju1AISbDgQFAeYi/jCh0279m4xqvdM18jgD5k90IAbYKUlTtXPQCmk31UAJUsHzo2MH1StsUo7vcuzRIK+xrtwyzxCTKdlmy6Uja6D7raBAMaqCjfO+UdI43IBFDUFjK9vabpWtkI0HZyMC3wLl4nFI4wcGtGALV8jUxntDJdKxsBIi/MHLu8SgtIqYGECQJoZsTC3PLcRRGzTNfKRoB+ixNINPV+MPDtKlb2VBiEQtgY8wEeNgK0m6Dwrs81x81C6TPsbpz3Qec2Z262ondfGwGa7lBrsjxOSvuyuwne8JdlBMUqr9I9xgEtksh0rWwEqPR2gUQ77+uV+E56BLsbQA0kMl0rGwFCxdP7vqi4Iyw0DohR0ub4ipIWrqYuBpvCWQHJmuMrclr4I/jif5lu5xIrIFlzvJG08MLXej1oJFjAzzpGDrGd7CswApI3xxtJC5+CK/Qpe4X8LPHuaAX7Ci5ArZbqmsmb4w2khZeTDq6fZq+Qn3WBADrCvoILUEP9xCpZc7wiLTx/p0utvHskLyOARlPrkTf+z49+xV5t37U83uEwMlgtIyBZc7wiLfybXi7VVrxSmoQBbadVozwFm6mkUVutkj7Cdjr5cJKmAFJpjmdMC7/+ctfeH1GrIfakbv5GlapDZEPr6YYuMQKSNcf7IS38DwP0teoDiI4YAcma4/2QFu6XQ+zb2a8pPBa0xxsy8FzOCsjdHO+ntPC8Cd36W3yTiSO0Hd7tfGWJuNTAS3JWQO7m+AqTFi7G13o/D51w+PioQQXEJjsBIgeTd18JuQSQ6j2cuoIY0JMYheKWZxh+CfILuxcXoAvWhDLbCVDe/Q7HEMXPKlgw6FmtfsjUFLxvFAXl5Jn3EdSArJDfARXM6dlP9YIXKJ3fdsiIud8BkRCCfdZuxYxwy+oTBtq3/A2otBkGZCCBzUp9PTPjjFeRGB/0MbsPfwMquUXtbuSP0Vq8aa++V8U3igY64PA7INZ2qL0Zqy0J4ZKIvJ/3Si7MJ/Xpz+7E7+egeFyh4dQV5whWPaxNKytvqhL/Ir7uuJPdi78BlcUxvVG8YvQtBIsGYJ9e8S/io0YXdif+BsQYxCkGXbxubWVIj9ufy8vEk/QQdid+P8QewxWixuMUJxhsjWHRrdjng16FqbhwJbsTvwPKG9Kk9aK5qcMpz4e7WlmePF7SCLPwjow/9WeHY5zRCDPrpAR0uJPDgXPVm+gkeyLxfrKtxUP/qu5Bm3Chge4nfWqbN5A3j/o4nNINGLhETPT7kjWqQzgvXBFB1R0XjmP34lPbvIG8+VIXH8d9eu7FgELTYzxI9QO+Rb1FkXxJAtsfZXfjU9u8kbz5WBcg3bjAi8RELUXPZ6WJm/3Cq5hcNN5gd+NL27yRvHn3HtRHfwOjBZPWlt5K9xO3630LdrqHwzHYwIYYAcna5o3kzd8gtbxrKzWHbW/6O9Ymlr0uAlJ05lp6Uv9y4SVGQLK2eUXe/Pksl+J3C7PPZklE2uYfzvLW9AxFkUILJtNtssZlaiyY5bw2CJMzTAyBxAhI1javyJs/9uRIp+7BmfwwUiKcyP/EiJHean6Xokih1EZ0m5E1HtNa8njaY08MHoan2j/ld0CytnmdvHlmn0OX0212acZpSdSI4YCZ9gqDIw0xApK1zevkzTP7DDZA8lR57bx5Zp9BB0jaNq+TN8/sM+gAydrmtfPmmX0GHyCLfXJAFHFAFFVqQBEMNsM/oNvsZekXuwlDb1IvZDA40pA/AF1ksLnKkHhcrtJnv08bu04Z90pP/gAUVOKAKOKAKOKAKOKAKOKAKOKAKOKAKOKAKOKAKOKAKOKAKPIdEFs0g2SRpg3LaALuRbtAlLLDtbeSolpklMitfXIkk8+AGKMZPIs0bZhGE3AvysF9GmSOjFA01c6CiVumhr1g3pFcPgNijGbwLNKwYRxNQL6otIMiFKK4Fo5pmRRZataRl3wFpBXNsLxj9dbvqdlpRTxojCag6YdoQUIJ8tJJ2CF8rodTZh15yVdAGtEMC6u+uHV8yNvEZFmudJFOxIPKaALafnDJxVrKPPOiE3gkwgmRhWYdeclXQOrRDNfq427QRkQTk9ivpYsUEQ8eKUcT0PGDS6YmadRqVVi6NY488hWQejTDAfjq4sWL78NZd33cixQRDx4pRxPQ8SN8/lpDvbeUC4NhWIkVjqTyFZB6NMOHzkvnkRs5OTkxm3NyrrgX6UQ8KEcT0PEjaF606okjOzpuoyWOZPIVkHo0w2dwQVx80Fmxye5FOhEPytEEdPwIaqsal55dZYw7cMyUI7l8vsyrRjNcCMeXiBfE9iyyR3vstCMelKMJ6PlBP4JabzsljsEqtfPBkZd8BqQezTCl2uyt6SELJPVxL9KOeFAZTUDHD3q7ilq/+5/Cs8uxCs068pLvjxqq0Qzlc9tUb5WJJPXxBDpoRjyojCag5+eRdmpOMp3HUK5ZR17iD6sUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAUcUAU+QHQpt6NanVcgmMFY+db792pnQwDXUYtRejbf4nfPst6QJPg4UX/6B+CQyr9B0gSGautATsRGtRX/PZZlgPaB6RH/XlwRA6o2MDwwTTJImMRKtfp/XiQSjey9KgpiSwH1FUcDzn/7tUCoHnPx9d+5BJCMR9Ornu6bHarmkl47J/YN7tUj3/7fGqdJh8I+4Kr9HDvOvUfOispkJnJolWlkbEnYFvjkJa4D0+FpzpLUTIAXBG+JV4X94Laj7LkU4uyGlBp+DzPTGzDQXvnRwgHW0ynB9cVPhXx6pbHQTh3xFZ98dPUkNg3tyZG5CNXaUHDrmuyHPcgpG4mjVbFauoBFDli4+SQl5DSkwDm0sN9cssxII/Xuo/teSOMvcclqwEdB0ln1rE4AGdIdwFQQjk6F7ZQmEtpLxQ/itAxeAahLfCdu/Qg7EJo42QNM1m0KpYE0MPCZ3rN3xWeMCByiAnfEq/tyhG6/w7mH2Q1oKMyQDiefXw3AdAU12BJK0KLUaxwQJTCavzzD7lLL9a8fdl5pGUmi1bFkgDCQz19BwcUnqSAJF6nCROT2cdFtxpQSTXnuGyLFuOzCHICEo67zFA8ptROOINi5+Jfvp78ck/pv++PgPYbNMzc0aqu7UgA4Z4Sf4P1Ck9SQBKvuJuKAAJCnTqTr6La451XMQJoPt418CgXq0IKZb/cU4pQwY7eVY6pm7mjVV3y2oN+gC8VnuR7kNRrQAHtAHJtnwZ7vQCdDcPDhz2QgGS/3F36UfNrCJ2CbepmsmhVLAkg3P/69Oq/KTxJAcm9BhQQGgUPvLW4L4xCXoDQmMi520fg9CzpL3eXHq+WsmXNvQ0uaZhJo1WxJIAixmRPC52OlJ4woMdbHyzF3zKvgQWE1na/qW7yMnzMywGVzrwtKhHH/st+ubv0k441Gtx3SMtMGq2KJQH0cUqd5q+UI6UnDGZ3XM2r+FvmNcCA/lidgIN+9c8BUcQBUVThAZWctrijdy9VeED+FgdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdEEQdE0f8DY9WTrXBmgn0AAAAASUVORK5CYII=" /><!-- --></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="fu">plot</span>(A5, <span class="st">&quot;qq&quot;</span>)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAIAAACb4TnXAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAgAElEQVR4nO2deUBN6RvHn9u+at+0EVoUJYnUlFv2srWMfUkk/EoRsgwzIYxt7FKDISOq0aKsiRZGRUiWIiVapNVtuXXvPb8/zsyZ221RuufeW72fv855z3ve93mrb+95t+ehYBgGCASCHIT4bQAC0ZtBAkMgSAQJDIEgESQwBIJEkMAQCBJBAkMgSAQJDIEgESQwBIJEkMAQCBJBAkMgSAQJDIEgESQwBIJEkMAQCBJBAkMgSAQJDIEgESQwBIJEkMAQCBJBAkMgSAQJDIEgESQwBIJEkMAEggcPHvj5+fn5+b1+/bqbRU2ZMoVCoUhKSnK95NacPHly+fLlubm5XC+594AhBIBTp07hv44bN250s6jJkycDgISEBNdL5iArK0tCQgIA7t27x92SexMi/BI2giS8vb1nzpwpLCxMUvlv377Nzc29c+dOSEhIY2MjSbX0HvitcATm7e09cOBA/NdhaGjo4eGBYZiFhQUAaGlpNTY2Llq0qF+/fqWlpRiGsVisS5cuWVtbq6qqSkpKDhkyZOXKlR8+fCBKY+/B2iy5mzg5OXH8CaEerANQD8Z/Xr169f79e/z69evX8vLy7E/XrVt3/vx54nbfvn0bN24kbvPy8vLy8m7fvv3kyRNZWdkulfx9TJw4UV1dHQBevHjx999/d7/AXg6/FY7AsLZGSngPJiIiAgAWFhZOTk6VlZXNzc1SUlIAoKend+vWreTkZDc3N/zFuLg4/MXvGIMlJSUFBQXdu3ePwWDgKc3NzY2NjR3b/Ntvv+Elox6sA1APJtAwGIyQkJBly5bhtyUlJfPmzQOA6dOnT5gwAQDevXsXEREBAPn5+d9XxdWrV52dnfHr0aNHh4eHDxgwwMvL6/bt24WFhVxoQ98GCUygERcX9/DwIG41NDRCQkIqKytv374dEBCQnZ2dlJSEP8K+N8ZAaGiorq5uampqVFSUv7//6NGjHR0dL1686OrqyoUG9HmQwAQaZWVlCoXCnnL06FF/f/+mpiYAEBISGjRoUF5eXneqmDFjxrJly7S0tNasWTN69GhfX9/w8PBRo0YFBQV1y3QEACCBCTgc6srOzvb19WWxWKNGjQoMDPzhhx9ycnJGjx7dnSo8PT2J6zFjxqB5C+6CBCZYdPyll5aWxmKxAGDTpk34ZEZWVhZXSkaQBNoqJRAQ68Lp6ek0Gq29bNLS0vhFcHBwampqeHj4tm3b8JT29NPJkhEkgQQmEBgaGuIX27dvx6cH22TixIkKCgoAcPPmzR9++GHu3Ln4LQC0tyGwkyUjSAIJTCCwsbHZunWrmpqalJSUjo5Oe9nU1NRu3LhhbW0tIyMzbNiwHTt23L9/H18rCw4ObrOD6mTJCJKgoE9zBII8UA+GQJAIEhgCQSJIYAgEiSCBIRAkggSGQJAIEhgCQSJIYAgEiSCBIRAkggTGC16+fBkVFZWTkwMAOTk5xDVOVVXV9u3bx4wZo6amJisrO2zYsLlz52ZkZLCXMG3aNAqFQqFQ2P0FAMCzZ8/w9FmzZuEp69evp7REWFhYT0/P2dk5Ozubi41isViSkpKUVvz+++9EnsrKylWrVpmYmEhLS1tYWOBnQ/sWfD1P3VdYuHAhAJw5cwbDMPxIMn6NYVh+fr6ysnKbv5qdO3cSJRCuZsTExHJzc4n0p0+f4ukzZ87EU/z9/dv7XQsLC6ekpHCrUe3tfgwNDSWaNmTIEPZHFArl+PHj3DKgR4COq/CCJ0+eAMDIkSPh3wMm+DUAuLu7f/nyBQBcXV2dnJwoFEpmZmZoaGhDQ8O2bdsmTJhgaWnJXlRTU5Ofn9+1a9e+Wamnp6e5uTkAVFVVRUZGPn78mMlkrlmz5vHjx1xpFN4JDx8+fNWqVezp1tbW+IW3t3deXp6QkND+/fuHDRu2devWR48erV692srKasSIEVyxoQfAb4X3furq6oSFhcXFxZubm+vq6oSEhPBrDMMaGxtFRUUBwN7env2VS5cu4b+dXbt24SkcztLi4+Px9A56sMjISKJAGo2G95NCQkKEN5v4+Hh7e3t1dXUZGRkTE5ONGzeWl5d3vl07d+4EgHXr1rX5tLy8XEhICAA8PT3xlIKCAtywDRs2dL6Wng4ag5FIcHAwhUKRlpZmMpl0Ol1UVFRaWprFYuHXy5cvr62tbW5uBgAajcZkMokXZ82aFRYWFhYWZm9vz1Em7ufQz88P9xrQSaSlpfX19QGAxWKVl5cDwLlz5xwdHe/evVtaWkqj0V68eLF3796xY8fW1NQQbxkZGeHDqrq6utZl4j1YRUWFtbW1rKysiYlJQEAAnU7Hn2ZlZeFnQ4kOTVdXV1VVFQAyMzM7b3lPBwmMRCQlJdXV1fFTkjIyMsS1tLS0urp6v379VFRU9PT0ACA9Pd3Q0HDDhg2RkZHv3r0TExObP3/+/Pnzx4wZw1Hm7t27RUVFc3NzDx8+3HlL6uvrcdcdEhIS/fv3B4Dt27cDgLq6+o0bNx49euTt7Q0AeXl5ISEhnSwTF9i5c+cePHhAo9FycnL27t07btw4/D/Fhw8f8GyKiorEK0pKSgBQVFTUect7PPzuQns/ixcvhn+H/vh1cHAw8fTBgweampocvxR5eXlXV9e7d+8S2YhPxKdPn65duxYAZGVlS0pKOvhE9PLyCg0NDQ0N3b9/PzGQc3R0xDCsoaEBvx0+fDg+ZUKn0zdv3hwQEHDlyhWi0pkzZ5qZmZmZmdXX13M0isFgiIuLA8CQIUMSEhJSUlJsbW3xMk+fPo1hGKF/9laMGjUKAPr378/1H7LAggRGOqampgCQnp7OcU1Ao9H+/PPPZcuWDR8+HB+SERw9ehTPwy6wmpoa/FtryZIlXZpFlJGRefXqFZ6NfX7P2Nh4/fr1d+/eJRyPfpOmpqZ79+7du3fv06dPeEppaSkuOVzDhFtSdoHhOtfQ0Pj+n2ZPAwmMXJqamkRFRYWEhOrr6/FrERGRhoaG9vLT6fS0tDTCF6KMjAyTycRaCgzDsNDQUACgUCjBwcGdEZimpqazs/Pbt2+Jip4+fUqMjgj09PSysrK+u7EmJiYAMHjwYAzDiE/N2NhYIsPQoUOJDH0ENAYjEVlZWTExsebmZhaLJSUlhV8zGAxJSUkjIyMAeP369d69e/fu3Us4SxMTExs7dmxoaOjEiRMBgEajlZSUtC7Z3d195MiRGIZt3bq1vdrZZxE/fvwYFRU1aNAg4qmpqWlqamp+fv6hQ4fs7Oxw3zj5+fleXl6dadrXr18/ffr06dMnBoNBJOKjL3ycqa2tjSfiixA4+PxKn/JcgARGIqqqqjIyMgAgJSWFz4YT18SkecC/sM8iYhhWXV0N/06HtC5ZSEjoyJEj8O+fbFd5/vz5gQMHDhw4ICIi4uvre+/evYKCAny65c2bN50pITo6WktLS0tL69y5c3hKfn4+Po9ibGwMAKNHj8ajhyUnJ+MZcnJycGvx1bm+At/6zr4Bvgh78OBBDMN8fHyIawJiydXKyurYsWNRUVHBwcHE7PyPP/6IZ+P4RMSZP38+8XvseB2Mg7S0NDyPo6NjZmbmq1evwsPD8ek+Ozs7IpuLi4uFhYWFhUXrSY53797hnV7//v137twZEhJCDOqIQdePP/6Ip6xfvz4kJAT3b0WhULrzFdrjQAIjF3yeHY8/YmNjA61ikRQUFGhoaLT5v8/ExKS6uhrP1qbAPn78SHhK7JLAmExm6zBfACAqKpqamkpkI1y+0Wi01oUEBga2LsHX15fdPAMDA44M7Pu/+gJIYCSCD7coFEp1dTWTyZSRkcGvObLRaLR9+/bZ2NhoaWmJi4vr6ek5ODicPXsW3+2B06bAMAzbtWvXdwgMw7CGhoZjx46NGTNGXV1dTExMV1fX2dk5MzOTPU/HAsMwLCYmxtbWVllZWVVV1cHB4erVqxwZysrK5s2bp6KiIi8vP3HixIiIiA5/YL2QNty20Wg0CQkJ3NseojsUFBRMmzatf//+N2/eLCoqmjp1Kn7Nb7sQvIOCYVhjY2NMTMzt27eTk5M/fPhAp9MpFIqCgoKZmRmVSnV2dsZnVxEIRFehrF+//syZMxUVFQAgKiqqpKSkqKjY0NBQUVFRW1uLZ7Kzs/P392/zqx2BQHQABQDGjBkzZ84ca2trU1NT9p0EpaWlDx8+vHXr1uXLl6uqqhwcHI4dO0Z8lyMQiG9Cef36deupHg7odPqVK1cCAwPnz5//888/88QwBKI30AXf9AwGo6ioCD8ugUAgOgMK/oBAkAjnXHxNTc3ff//d2Ng4dOhQPT09InwbAoH4DloILDk5edasWZWVlfitkpJSQEDA6tWrJSUl+WEbAtHjafGJaGZm1tDQEBISYmRklJ2dnZqaevjwYX19/fv374uJifHRSgSih9JCYPLy8qdOnZozZw6RUl5ePnr06Hnz5uEeThAIRJdocVzFysqK/XgPAKioqGzfvr0zTsIQCERrWgjMy8tr7969xAYOHFVVVfycD29oamr68uVLVVUVz2pEIMijhcC2b99eUVExfPhwYvPU58+f9+7d29p5GNcpKirasmWLnp6ehISEioqKoqKihITEkCFDNm3a9P79e7JrRyBIosUYLCgoKD09PSMjo7i4GACkpaXr6urU1NQuXrxoa2vL4Y+Fi2RlZdnY2CgpKTk5ORkaGioqKmIYVl1dnZubm5CQUFVVlZSUhLuLQSB6Fm0vNH/69CkjIwMXW2ZmZnV1tYSExIgRIywtLfft28d1pVGpVElJyaioqNbrAQwGY9GiReXl5bdv3+ZupQgED/j2Tg4Mw/Ly8nCxpaenJyYmSklJcdcIOTm5M2fOuLi4tPk0NTV12rRpaFSG6Il8+1QlhULR19fX19dfsGABSUbo6+vfvXu3PYElJiZ+czsyAiGYdO3Ycn19Pde7LwAICAhwc3N7//69q6urkZGRgoICfrQ+Nzc3Ojo6Jibm8uXLXK8UgeAFdDq9M64Fnj9/PnPmzJ9//pkUxwUYdu3aNSqVymEbhUKxt7dPSEggqVIEgmwoSkpKc+fOnTNnjoWFBe76mJ2CgoKbN2+eP38ed6EeFhY2btw48tReVVVVXFyMu9pUU1PT1NRkDx3QTdatW3f//n1ulYYQcIQxbASNVi8s/FJKSlJSMjIyUk1NjfdmUMaNG3fv3j0AEBMTMzY2VldXV1BQaGxsrKioeP36dVlZGQAoKip6e3tv2LCBjO/Db0Kn0xkMBuGf7LuxsrJasWLFsGHDuGIVQmBhMBiyHz7o/vwzU06ucOvWJjW1uXPnRkZGDh8+nPfGiCQlJb148SI0NPTOnTu4SzDimZycnKOjo4uLy9y5c3EvrXxhyZIl4eHhWCfOrVVXV3cwm19aWqqrq0uElkT0Pv76668Na9e6FhZuFBEpW7dOb/fuYRQKAPDxOIgIAJiYmOCxMKqqqoqKiioqKiQlJVVVVXV1dQXhPBiVSsWdTn+TgoKCDsJsf/z48dmzZ61HeojeQW5u7k8uLn8CVAOYMRjMsLDXW7d28i+HRPg9COQdMjIyISEh/LYCQQ5M5v3580sAPHFHTgDA5kR5+PDhz54944tdHU3T5+TkREREfPr0qbm5WVtb29nZmdTY1Uwm88OHDxoaGhyfo42NjTU1NXwZoSJ6Bu/ewdKl+lVVowE+sCXLycnxzaR/aTe6ypEjR1xdXYWEhMaOHWtnZycmJrZw4cIuhS3tPAwGY9u2bbKysnp6eoqKihs3bmQPNXL58uU2I4wgEMBiwenTYGUFU6dK3r8voqdHPLG1tRWECa12e7DDhw8/ffpUVlaWSPH397e0tFyzZg3XjTh06NDu3bvXrFljZWWVlpZ28ODB8vLyM2fOcL0iRK8iPx/c3YHFggcPYPBgOYCUlJT9+/e/ffvWwsLCz89PEGYQ2h2D6evrFxYWsqeUlZWZmpqS8Z2qr6+/adMm4jYsLAwAYmJi8Fs8AlX3a0FjsN4Di4UFB2MqKtiePRiT+c3sgjgG27Nnj7W19bhx43R1dQGgqKjozp07JH0ifvr0CQ/tgzN//vwbN274+vpOnDiRj8sDCAElPx88PKC5GdLSgC3StGDS7hhs1qxZz549mzhxopSUlISEhL29/bNnz1xdXckwYujQoYmJiewpBw4cqKur6yCeN6IvgmFw+jSMGQOTJ0NysuCrCzre7KuoqLhw4UIeGLFw4UIfHx8Gg+Hk5GRraysuLq6qqnr27Nnp06d//fq1vfh0iL7F+/fg4QF0OqSmgr4+v63pNHz5MG3Njh07+vXrBwBv374lEuPi4vr3788tO9EYrKfSxRFXawRuDFZTU2Nra8vh/QaHJA8ZW7du3bBhw8ePHzU1NYlEJyenwsLC5ORkXnrdQQgWBQXg4QENDZCSAj3wWGDbApOTk0tJSRk9enRQUJClpSVvTBETE9NjW8fAERERsbe354HXHYTAgWEQEgJbt8K6deDvD4Iw59512h2D9evXb+HChTo6OuxdCgLBIwoKYNkyqKuD5GToySHp2p1FBIDNmzejvecIXoNPFVpawoQJkJrao9UFXXUZgECQS2EhLFsGNFpP77gIWghs1qxZrXMICQnJyMgYGBi4u7ujGXNEl2hoaLh16xadTqdSqSoqKt/Iff48+Pv36BFXa1oIbMCAASdPnmQymebm5qqqql++fHn8+LGqqqqJiUlycnJgYGBsbOzEiRP5ZSuiZ1FSUvLDDz+8e/cOANTU1K5evWplZdVeVlixAsrK4P59MDLiqZUk02IMJiEhYWxsXFhY+OjRo7i4uIcPH3748EFLS2vRokX5+fmbN2/28fHhl6GIHkdgYCCuLgAoKyvz8/NrO19EBJiZwdChkJray9QF0HIBV1tbOzIykmOlLDo62sjICMOwiooKCoXy9etX3q3ScRW00Mxj7Ozs2P/SZGVlOXMUF2PTp2OmplhWFqmW8HGhuUUPRqFQiPCWBBUVFbhX3crKSvwdsjWP6B0YGxt3cAsRETBiBBgZQUYGmJnx1DIe0mIMNn/+/I0bN8rIyEybNk1GRoZGo8XHx69fv97d3f3Dhw++vr7m5ubsJ8QQiA746aefbt++je/CUVdX/+8oRmkpeHnB+/dw/TqQeUZeEGghsB07dtBotCVLljQ1NUlKSjY0NAgLCy9btmz37t1nzpzJzs6Ojo7ml6GIHoe6unp2dnZiYmJDQ4O9vb2CggIAQEQEeHvDkiVw5Qr0gbjEbQR/KCkpycjIKCoqUldXNzc3HzhwIADQaDRpaWkKhdJWIT0DWVnZQ4cOLVu2jN+G9FXKysDLC969g3PnwNyclzWbmppeuHCBP34RWydpaGhMnjy5tLRUVVWVOO/If/dXiB4N0XFdvtwXOi4Czq1SMTExI0aMkJSU1NXVlZKSMjU1jYmJ4YtliF5CWRk4O0NgIMTHw549fUpdwNGDxcbGzpw509HRce3atRoaGmVlZVeuXJk5c2ZcXJyTkxO/TOw8TCazsLCwvado/pMPRESAjw8sXgzh4X1NWjgtBLZr166lS5f+/vvvRMr8+fOXL1++a9euHiGwtLQ0d3f39p42NDS0XoRAkMXnz7ByJeTmQlwcWFjw2xq+0UJgL1++3LZtG0cOV1fXnhKey9bWltg60BpZWVkuxmpBdATRcf35J7QK2dOnaDEG09LSevHiBUeO7OxsbW1tHpqEEFzKysqampo6yvH5M7i5wS+/QGws7NnTx9UFHAKbN2/ezz//fPTo0bKyMgzDysrKjh8//tNPP82dO5df9iEEhOzsbBMTE3V1dXFx8a1bt7adKSICTE1h0CB4/BhGjeKtgYIK+74pJpO5evVqERERABASEgIAERGRVatWMRgMnu/h4j5oL2J3GDp0KPufDeee1cpKbOFCbOhQ7NEjPhnYEYLi9EZISOjYsWMBAQFZWVnFxcX9+/c3MzND34eIsrKyly9fsqe0CFp/7Rp4eYGLCzx5gr4JOWhjoVlLS0tLS4v3piAEFnl5eY6Uf05PVlfDxo2QmAiXLsEPP/DBMoFHBACOHz/+zXyrV68m3xgEl2EwGEJCQvjXPkFTU5NYO0tS7T0SFxdfu3btwYMH8VsNDQ0PDw+IjwcvL5g6FZ4/B37EFu4ZYBim3An48v3KXfrUGOzr16/z58+XkpKSkZFZsWIFnU7HMKygoAB3gKegoLBnzx72/KdPn1ZVVQUAKyurnJyc1gWyWKyLFy8uXbp006ZNH1+8wDw9sUGDsPv3edSe7sHHMZigePblAX1KYKtWrWL/N7pt2zYMw6ytrdkTo6Ki8MwpKSns6QYGBiwWq92i4+MxbW3M0xOj0XjTlu4jKAcuEb2GGzducNzW1tampaWxJ16/fh2/uHXrFnv6mzdv2l6vr6mBFSvA2xsuXIDgYJCW5rLRvREksN4JR/RUOTk5SUlJjsOyxNRF61CrbQRfvX4d8ICRz59DS18AiA5AAuudcExKrV69WlRUdMmSJeyJS5cuxS/mzJnD7pBv9uzZLVys4R3X//4Hf/yBOq6ughyP9k48PDyUlJTCw8OFhYXd3d3Hjx8PAAcPHhw6dOitW7eUlZV9fHyM/nXhpKmpmZKScujQoY8fP9ra2rYQ582b4OkJkyfDs2eAzgR+B22OzIYOHRoaGsrj4SDZ9KlJDi5QU4N5emIDB2J37/LblO4icJMcFRUVDQ0NPJY6QoC4efO/EReVym9rejAC94nY1NRUW1srLCz8j48URKfJzc3FA9XjoQy/k9paWL8ebt2CM2fAwYF71vVRBGWSo6ioaMuWLXp6ehISEioqKoqKihISEkOGDNm0aRNJIf96EywWa8GCBQYGBra2tvr6+sT8e5e5fRtwzzDPnyN1cQWB6MGysrJsbGyUlJScnJwMDQ0VFRUxDKuurs7Nzb1y5UpwcHBSUpKpqSm/zRQs7t+/v3Hjxnfv3o0YMcLBweHixYt4ellZmbu7+6dPn4S7FD+hvh42bYLYWAgNhfHjSbG4b9LmyExNTe3o0aM8GwiOGzduypQp9fX1rR81NzfPnTt3/Pjx3a+lN01yFBYWqqurE7/E1t+E796960JxycnYkCGYpydWW0uayfxEUI6r8IsnT56cOXNGUlKy9SP8QNq0adN4b5UAgmFYRkZGTU3NmzdvSktLifTW0bSJ4PHfoL4eAgMhLAxOnYKe4HalxyEQAtPX129xvqgliYmJBj0w+jXXqaurmzRpEr7dqfX5kUGDBhH7mw4cOEA4tOyI1FRwd4cRIyA7G9CUEjkIhMACAgLc3Nzev3/v6upqZGSkoKBAoVDwMVh0dHRMTExP8bpDKgcPHiQ2E1ZXV4uKijY3N+O3ZmZmycnJ0dHRpaWldnZ2345b39AAv/wCYWFw8iSgrwMyEQiBubi4xMXFHThwwMPDgz2dQqFQqdS4uLgpU6bwyzbB4enTp+y3zc3Nc+bMef/+/ciRI3/66SdZWdmFCxd2qqC0NHB3BzMzeP4ckJstkmlbYNnZ2dK83XLm6Ojo6OhYVVVVXFxcUlICAGpqapqamsjRGo1GCwkJef/+PZ1OZ0/X1dW9dOlS18rCO64LF+DECZgxg5tWItqhbYF9O5wuOSgoKCgoKHAEkqLT6QwGg8eCFxDq6+utrKwIX3p4TCn8+tChQ10r68EDcHcHU1PIzkYdF88QiE/EjlmyZEl4eDjWCcfX9+7dc3Z2bu9pXV1dj/PsGx0dze6pkkajHTx4UEJCYsKECYMHD+5sKUTHdfw4zJxJiqGIdugBAqNSqZ2M7WJnZ9eBZ18dHZ0e98HJPhePM2rUKBsbmy4U8fAhuLvD8OHw/DkoKXHTOEQn6AEC8/T09PT07ExOCoXSy3YwcmhJTU2tC0GuGhvh55/hwgU4dgxmzeK+cYhOICh7ERFtYmlpeeDAAfxaQ0Pj7Nmznd3I+/ffMGIE5OfDs2dIXXwECUzQWbt2bV1dXV5eXmFhYaeWKxobISAAnJ1h1y64cgWUlcm3EdEuSGCCSHNzc0BAgI6OzoABA7Zs2SImJjZ48GBRUdFvv/noEZibQ34+PH8O7c/3IHiGQIzBYmNjOTyHtWbfvn28MYa/lJSUJCYmxsTEREZG4ilBQUGSkpLtxlsgwEdc58/D0aPQzqYzBO8RCIFJSUklJyenp6dLSEi057W7dwvs+vXrv//+e2lpaXZ2duudu1evXv2GwJ49g8WLQV8fnj0DPq1hItpEIAQ2fvx4KpVKpVKxVk4w+wKxsbEzOtxX0Z6nawCA5mY4eBAOHoT9+6GTW6UQPERQxmDCwsKzZ8/mtxW85uvXr3v27Pmm3/924+I+fw6WlpCaCllZSF2CiUD0YDgzZszQ09PjtxW8o7Gx0dbWlmMLLztaWlr6+voLFixoQ2BEx7VjB3RukRDBF0SgczsPy8vLyTalT4VNysvLCwwM7EBdQ4cOvXfvXtu/mufPYckS6N8fsrKgkwcrEXxCBAB27tyJ3zQ2Nm7dulVOTs7NzU1HR6eioiIyMrK+vv7kyZN8NbK3ce3atTbPaEtKSjo7O8+aNUteXt7W1raNeXkGAw4cQB1XT4Ldf4CXl5ednV1jYyORwmAwJk+evGLFCt46MiAFQfDJ8eLFCxMTkzZ/EWpqap8/f+7o5efPMXNzbOpU7ONHXtnbSxAUx6NxcXHe3t7ibFFAhYWFvby84uLiyFV5r6ahoQHDsI8fP9LpdFdXV/bd8QSampq///57u9/qDAbs3Qvjx8OKFRAfD5qa5FqM4B6ckxwfP37kSPnw4QMeFh3RJb58+XLhwoXTp0+/fv1aSEiIxWK1l9PZ2TkyMpJCobT9+MULWLIEVFXh8WPoM2PU3gN7d7Z8+XI5Obm4uDgi5dq1a3JycugTsas8fvyY3a1aB1haWpaUlLRdSnMztmcPpqqKBQfzxuzeiqC4bTt48GBeXt60adMUFRX79+9fXFxcWVlpb29PbKkDHa4AACAASURBVOhGdAY6nb5kyZLWR7k4UFFRiYyMtLGx4Yih/A85ObBkCSgro46rR9NCYDIyMklJSUlJSenp6aWlpZqamqNHj/4BRY/vCjQabezYsdnZ2R1nc3R0PHnypLa2dhvP2KcKly+H9j4dET2BNgZXVCp18ODBNTU17c13ITrg9OnT31TX4MGD//zzz7ZPduXkgLs7KCpCZia0KT9Ej4JTYDdv3iQ+bzAMGz9+/IwZM7y9vflhW4+ktc8CaWlpU1NTLS0tBoMhKytrYGDg5eXVhrrwjuvAAdi5E3VcvYYWAgsLC3N3d1+6dOkPP/yAO9kbO3bsmjVrxMXFO3lov4/z+fNnjll4RUXFoqIiKSmpb7z58iW4u4O8PGRmgo4OiSYieAz7jMfQoUP9/PwwDPvy5QvxaP369SYmJryffuE6ZM8iMhiMYXjQun9RVlaOj4//xmtMJhYcjKmrY8HBGItFnnl9GUFZaC4oKJgwYQKHAqlUKorQ9U1qamosLS05Rl9+fn5Tp07t6LV374BKhbAwSEkBT0/0Wdj7aCEwQ0PDR48eceTIzMwcNGgQD03qkezcufPJkycciR2d42Kx4PRpGDMGpk6Fe/eg804OET2KFmMwb29vT09PERERBwcHACgvL4+Li9u5c+eePXv4ZF4PgMlk3r17NyoqqvWj6dOnt/1Ofj4sXQpMJjx8iKTVu2khsCVLlnz9+vWXX3756aefAEBVVVVcXHzt2rW+vr58Mq9rZGRkbN68ub2nDQ0NNTU13K2RRqNRqdTMzEyOdGlp6bi4OH19fc4XMAxCQmDrVli3DtavhzaXmBG9CM5pem9v76VLl+bk5BQUFKioqAwbNky55/j9MjIy2rhxY3tP09LSuhUdvC2OHj3aWl0AEBERQaVSOVPz88HDA5qbIS0NhgzhriUIwaSFwFavXr1gwQIrKytLS8tvx5gSPGRkZMa3H19YWFi43Q2138vLly9bJ65bt47TgSHquPoqLX7TMTExY8eO1dPT27Zt25s3b/hlU08hOTm5ze6Lcyb2/XtwcIA//oDUVNi4EamrT9Hil11UVPTw4UMXF5ewsDBDQ8NRo0b99ttv39y02jfx9fW1s7N7/fo1R/rKlSsnTZr0zw2GwenTMHo0TJoEKSnQekiG6PW0t0D25MmTLVu2GBoaCgsLT5o0iZdrcyTBxYXmNr8Mra2tc3Nz/8v0/j1mb49ZWWGvX3OlUsR3IygLzewMHjzYzMzMwsJCVFT0zp07ZGq855GXl9c6UVNTcwg+dYF3XJaWMHEipKQAiuDeh+GcRSwqKoqLi4uNjU1KSsIwjEqlHjlyZCaK2taSNs8ZuLm5AQAUFoKHB9TVQXIyGBry2jKEgNGiBzM3N9fR0Vm7dq2oqGhwcHBZWdnNmzeXL1/Or4iyAouenl5QUBBxq6SkFBwc7OriAqdPw6hRMGECpKYidSGAvQdrbm5WVVU9d+7crFmzuL5e1PvYtGnTjz/++OLFCwMDA0NDQygshIkT4etX1HEh2PmvB2OxWOnp6bKyskhdnaGurk5MTGz69OmGhoYQEQFWVjB+PKSlIXUh2PlPYOLi4kuXLj1//jzWiXDjfZmMjAxzc3MZGRkdHR1rPb1qW1vYvx8SE2HjRhAW5rd1CMGixSSHpaXl/fv3R4wYMXnyZDU1NXZnLGvWrOG5bYLI+fPnFy9ejF+7ARwrKAivqlrx+TOlg43ziD5MC4ERKvrjjz848vVxgcXFxW3fvj0nJ6epqQkANABOAegCTAJ4WlMzubh4wIAB/LYRIYi0EFhJSQm/7BBYKioq/Pz8Lly4QKS4ARwFOAfgCtAMAJ2LnoHom7TtsreoqAh5lUpLS5s9e/anT5+IFHWAUwADAaYAZP2buHbtWmlpab5YiBB8OHdy3Lx5U0NDQ0dHB3cvMX78+KNHj/LSoKampi9fvlRVVfGy0tbU1dXNmjWLXV1uAE8BXgOM+ldd/fr1O3v27P79+/llJELwaSGwsLAwJyen6dOnE19EuFep06dPk21HUVHRli1b9PT0JCQkVFRUFBUVJSQkhgwZsmnTJt57BGEymTNmzCBCoqkBXAX4CWAqQABAEwAAmJmZffz4ccmSJVw/AoPoTbT4RNy9e7e3t/fBgwcrKirwlMDAwMbGxqNHj5Lqti0rK8vGxkZJScnJycnQ0FBRURHDsOrq6tzc3CtXrgQHByclJZmampJnAAcrV65MTEzEr90AjgD8ATAboAlATExsxIgR69atc3V1RdJCfBv2nb9SUlIJCQlYS7dtCQkJ0tLSpO44Hjdu3JQpU+rr61s/am5unjt37vjx47tfS8e76VksVk5OTmRkZEBAAP6TUQP4CyAbYOS/Pytra2smk9l9SxA8RlB20/PLq9STJ088PDwkJSVbPxIREVm1alWb5xq5SEZGhpqamrGxsaurK+7hBx9x5QJYADwGMDMzu3z5cnJyctuBGhCIdhAIr1L6+vp37951cXFp82liYqIBaSc+mpubDx06tGXLFgaDgaeoApwAMASYBoDLeuzYsSkpKUhaiO+Bo0c7cuSIkpIS8VRcXDwgIIBFssdZPPzclClTfv/99wcPHrx69er169d///33+fPnnZ2dhYWFIyMju19L60/E8vJyjiUsN4ASgD0ARJDPAQMGdL9qBH/h4ydiGyeaaTTao0ePLl++fPfu3fLyct7Yce3atdZumCgUir29PT4s7D4cArtw4QJ7p6QKEAHwAmBUSxsCAgK4UjuCjwhKAD4caWlp3KvUmzdvMjMzLS0tFRUVu9FHdgpHR0dHR8eqqqri4mJ8Q4mampqmpiZJVWdnZ+PRLXDcAA4DnAdYAEBny2ZoaNiBHzgE4pu0EFhRUZGHh4epqem+ffv++uuvH3/8kclkKisr37lzhzez5AoKCgoKCsbGxuyJdDqdwWBwd7dEdHT0PzUC7AGwAZgJkA4AABoaGmPHjh04cCCVSp04cSKKT43oDi3+enx8fHJyclavXg0AO3bsGD9+/MmTJ1esWLF169a4uDg+WQhLliwJDw/HOnGI5u3bt6Ghoe09bWpqamhowK+FhYUBwAngJEACwCiAegAAcHJyioiIkJCQ4IrlCEQLgSUnJ+/Zs2fGjBklJSVPnz69ffv2wIED3d3d+buVnkqlysjIdCanhISEgoJCe0/l5eWJ9Qa3CROUt2xxAJgPkAwAAF5eXr/++qusrCwXLEYg/qWFwFgsFn6c+caNGxISEjY2NgAgKytL/OPnC56enp3cR6KlpdXBkCk6OvqfEV1CwpAVK2SnT/eoqXn39q29gcHJkyfb8COPQHQbzgOXR44cUVZWPnTo0KRJkyQkJOrq6k6dOmXIw2PwTU1NtbW1wsLCHfRF340IjQYrVkBiIly8qG5rG8/1ChCIlrQQ2K+//jpp0qTx48f369cPP3Npbm5eUFDQZmwe7lJUVHTq1KlLly4VFBTgwy1xcXFtbW1XV1dPT8+BAwd2vwqdhgZ9Z+fs0aMz/PwYr1/D69cvXrz48OEDjz8L6+vr6XQ6Gf8+OqCpqammpobH59aYTGZ5ebm6ujovK8UwrKqqasaMGeyJxN5a3kPhmDxoaGh4+fLlwIED8a+piIgIMzOzISSHAulgs29CQkJVVRVXNvueCQ5+9uhRo6gokZKUlFRTU8O+sM4Dqqqq6HQ6j//saDRaVVWVtrY2LyttbGwsKSnhyj/HzsNkMvPz85cuXcqeKCEhsWPHDv54c2q9NFZRUZGamhoVFZWRkUGj0XiwGMebzb6t2bx5865du8gouQNOnDixcuVKHlcaGxs7bdo0HlealZVlZmbG40rLy8uVlZV5XGkHtNhfx2Kx/P39NTU1bWxsXFxcRo0apaWltXfvXrJFzvfNvggESbQQ2I4dOw4ePOjj45OdnV1VVZWTk7Ny5crNmzeTfagZ3+zb3lNSN/siEKTSYpLj4sWL/v7+RJclLy8fFBTEZDJPnz7t7e1NnhEBAQFubm7v3793dXU1MjJSUFCgUCj4GCw6OjomJuby5cvk1Y5AkEcLgVVUVOBrX+zY2toGBweTaoSLi0tcXNyBAwc8PDzY0ykUCpVKjYuL4wwYiUD0EFoIzNbW9vr169OnT2dPvH79urm5Odl28HizLwLBG0QA4NmzZ/jNqlWr5s6dW11dPWfOHHV19dLS0suXL4eHh8fH82hJts3NvghEz0UEAMzMzNiTwsPDw8PD2VOmTp2KIYf1CETXEQGAt2/f8tsM/iAiIiLKtu7MG0RFRXl/BIYvlYqIiPSRSjvgv50cTCbzwoULKSkpr169qq2t1dPTmz59uru7u3DvjRhSX19PoVDaXH8jj6amJjqdzuP9WUwm8+vXr/Ly8rysFAAqKip4vFGGX5W2xz8Cy87OXrRo0dOnT/X09AwMDERERF69evX27Vtra+uLFy/q6ury204EokdCwTCsrq7O1NRUTEzs2LFj9vb2xLP79+97eXlVVlbm5eWhqHwIxHcgBABBQUFfvnxJSEhgVxcA2NnZXb9+nU6n82C3FALRKxECgJSUlIULF7YZ4WrAgAGLFy++efMmr+1CIHoFQgDw/Plzjpl6dkaMGJGbm8tDkxCI3oMQACgpKeHO6NukvLxcTU2NhyYhEL0HIQAwMzO7fft2ezlu3brVQf+GQCA6QAgAVqxYcffu3SNHjrR+fPTo0cTERFJjFyEQvZh/1sHWrFlz5MiRKVOmLFu2zNDQEMOwN2/ehISE3Lhxw8fH5/Dhw/y2E4Hokfy3kwM/DFZaWko8U1NT279//4IFC/hkGwLR42nh9IbBYLx9+zYnJ4dCoQwdOnTw4MECta0LgehxcHqVQiAQXKRPBJWLioqytLSUl5e3t7d/+vRpN7NxsdKrV69SWrJ8+fJu1gsAd+7ciY2N7aZt3K2U6y09duzYmDFjZGVlDQ0N9+/fT8RP5ICMlnaJ3i+w+Ph4Nzc3CwuL0NBQcXFxGxuboqKi787G3Urz8/NVVVVPsdH9ES+LxdqyZUtKSko3beNupdxt6c6dO729va2trcPDw2fNmrVp06bAwMDW2choaZfhm8M4XkGlUidPnoxf19fXa2trb968+buzcbfSVatWOTg4fHctHBQVFR0/ftzW1hYA/P39u2kbdyvlYkvpdHq/fv18fHyIlHXr1klKSjIYDI6c3G3p99HLezDcK7Cbmxt+Kykp6ejoeOnSpe/Lxt1KASA/P5+LAeazs7MvXbrEYrE6CL/E3ZZ2slLgaks/fvxYW1vr5OREpFhZWTU0NHz48IE9G9db+n30coEVFxcDgJGREZFiZGRUUFDQ1NT0Hdm4WykA5OfnFxYWjhw5UkZGxszM7PTp099RHcGUKVNSUlJSUlK0tLS6bxsXKwWutlRTU/Pt27d4n4mTlpYmKSmpoaHBno3rLf0+ernA8GU99kgLuOP72tra78jG3UpZLFZBQUFmZubixYsvXrw4atSoFStWHDhw4Dtq5Lpt3IW7LRUXFx80aJC4+D9x6i9evHj06NH//e9/HF0oX1raml6+zIVhGABQKBSOFA4/CJ3Mxt1KGQzG+fPnR40apaenBwAzZsxoamoKDAz08/Njj87OXbjb0k5CUkvLy8vXrVt34cKFJUuWBAUFcTzlS0tb08t7MPwcQHV1NZFSXV0tLi7OET2ok9m4W6mYmNjs2bPxvzmcmTNn1tbWvn///jsq5a5t3IWMliYkJJiYmKSmpkZHR589e7b1jgi+tLQ1vVxgmpqaFAqF/TxbXl5e69FCJ7Nxt9LPnz8/fvwYY1vox/9KOhkvl1TbuAvXW5qQkDB9+nRXV9eXL19yhAIj4EtLW9PLBaaoqEilUqOjo/FbBoMRHx/v6ur6fdm4W+nz588tLCzY57Xi4+N1dHRIPX3H3ZZ2Eu62lMFgLF++fN68ecePH+9g6pIvLW0DHi8L8J6EhARhYeFffvklNTV13rx5CgoK+fn5+KPg4OA5c+Y0NjZ2nI2kShkMxujRo1VVVXfs2BEfH+/j4yMkJBQREdH9Jg8ePJhjSYq8lnamUu62NDExEQA2bNhwriUNDQ28aWmX6P0CwzAsIiLC0tJSTk7OwcEhKyuLSF+2bBkAEEEG28tGXqX19fW+vr6GhoaysrJjx469fv16NyvFaf23TmpLO1MpF1t66tSpNruK0tJSjCct7RJosy8CQSK9fAyGQPAXJDAEgkSQwBAIEkECQyBIBAkMgSARJDAEgkSQwBAIEkECQyBIBAkMgSARJDAEgkSQwBAIEkECQyBIBAkMgSARJDAEgkSQwBAIEkECQyBIBAkMgSARJDAEgkSQwBCCwrx58zQ0NOrr6zuTOTc3V0REhGxHyN0HCQwhEDx8+DA8PHzLli1SUlKdya+vr79w4cIdO3Z8+fKFbNu6A3J6gxAIqFRqdnZ2cXGxmJhYJ1958+aNoaHh+vXrf/31V1Jt6w6oB/sHPz8/Slv89NNPvDTj6dOnmZmZXX0rODhYRUWF68aoqKgcP36c68W25sWLF/fu3Zs7d27n1QUABgYGY8aMOXPmTGNjI3m2dZNeHvyhS8jKyp49e5Yj0dDQkJc2HDhwgEajXb16lZeV8p3Q0FAAmDNnTldfnDNnjq+vb0xMzOzZs0mwiwsggf2HuLi4i4vL972LYRiLxeJx5I5ew507d8TExCwsLLr6orW1NQDcvn1bYAWGPhE7xZ07d4SEhO7du4ffRkVFiYmJPX/+/N27dxQK5caNGwMGDBAVFTUyMuIYD/zxxx8WFhbS0tImJibs3SOTydy+fbuhoaGiouKkSZNycnIAwMrKKiwsLDo6mkKh1NTUdPB6XV2dl5eXtra2trb2qlWrePCNxGKxdu3aZWxs3K9fvzFjxly7do390ZYtWwYMGKCrq7t582Y/P79Zs2Z1vuSqqqqXL18aGxsTIb8Idu3aZWZmFhsbO3XqVHl5+REjRnCEgTY1NRUVFU1OTu5O08iF986EBRNfX19lZeUOMixatMjQ0JBOp9fW1mpqam7duhXDsLdv3wKApKTk8uXLo6Oj/f39KRTK9u3b8VeOHDkiKiq6bdu2hIQEX19fCoVy4sQJ/JGHh4ecnNzx48cvXrxoY2MjIyPz4cOHiooKFxeXyZMnl5aWslisDl6nUqn9+vU7ePBgRESEnZ2djIxMx8Z/H8rKyseOHcOvV69eLSEhsXv37mvXrrm7uwNATEwM/sjHx0dBQSE4ODgiImLEiBHi4uIzZ87sfC3Pnj0DgClTprR+NGPGDA0NDWNj4wsXLvz555/6+vpKSkrV1dXsedTV1cXFxb+3iaSDBPYPvr6+bf4DunHjBp6hvLxcWVk5MDAQ97GOhxfABebi4kKUs379ellZ2draWhqNpqSktGPHDuLR8uXLVVVVMQzLzc0VEhK6cuUKnv7p0ycxMbHDhw9jGLZgwQL8r7OD1+/evQsAsbGxeHpjY6OmpiapAisqKhIRETly5AjxyNHRccSIERiG4fN+f/75J9EWERGRLgkMb868efNaP+rfv7+urm5NTQ1+e//+ffbfCM7QoUMB4OvXr11sHI9AY7D/aHOSw8zMDL9QVlY+ePDg8uXLWSxWUlIS+/fM3LlzietFixbt27fv1atXAFBRUeHg4FBRUYE/srW1DQkJKSoq+vvvv4WEhIjvqP79+3/58kVUVJS93pycnPZeT09PV1BQmDZtGp4uLi7u5uYWFhbWmTaWl5cPGzasvadKSkr4xyoHWVlZDAbjxx9/JFJ+/PFHd3f3pqamjIyMpqYmwpj+/fubm5tzvF5TUyMnJ9depZKSkgDQ+iv306dPxcXFv//+e79+/fAUPOAyR86GhgYKhdJxCHY+ggT2H9+c5JgzZ46fn5+qqurYsWPZ09XV1YlrPMRbUVERk8kEAI6cAFBdXf3hwwdlZWX2oIyysrIc2QoKCtp7vbS0VFNTkz1RR0eno4axIS0tvXXr1vae4n/rrSkuLhYSElJVVSVS+vfvz2KxSktLi4qKJCUl2UPpsWcDgNraWltbW/w7EACio6MDAgKam5vXrFnj4+ND5Cf+jxBkZGQAAHt8vezsbADAuyyCiooKBQWF1hEuBQQBNUsw2bdvn4iIyLt3786dO4ePQ3DweNs4JSUlAKChoUGn0wHg8+fPrVeo0tPTKysrWSwWEaH46dOnUlJS+vr6RB78rTZf19TULC4uZk9p/dfZHlJSUv/73/86mZkAl1N5eTkhnrKyMgqFoqqqqq6u3tDQUFdXJy0tjT/68uUL8R8nKCjowoULRJ9TWVnp7e2dlpYmLy9vaWnp4OBgbGysq6srJyf3+vVrjkrT09OlpaWVlJSIlNTUVCUlJfZQtKWlpbW1tfb29l1tEc9As4id5c2bN4GBgSdOnPD391+/fj37Dp0rV64Q12FhYVJSUkOHDjUxMREXF4+LiyMebdu2bfz48QBgYWHR1NQUHx+Ppzc2No4fP559Xg4AOnjd0tKysrKSeJ3JZEZFRXG/wWyYmZmJiIhEREQQKVeuXBk2bJiEhIS5ubmwsDBhfGlp6ZMnT4hskyZNYg9PHh8fP27cOB0dnX79+s2dOxcvUFhY2MbGpqysDO+0CTIyMurq6j5//ozflpeXHz58OCAggH0t5OHDhwBApVK53mRugXqw/6DT6TExMRyJ8vLydnZ2GIYtX77cwcHB2dl5ypQply5d8vf3P3fuHJ4nNjZ29erVjo6Oqampe/fuDQgIkJeXBwBfX9+VK1eWlJSYm5snJSXt37//t99+AwBTU1M3Nzd3d/egoKCBAwceO3aMwWDgKzmioqJ5eXmPHz82MzNr73VbW1sqlTp//vydO3fq6OicOnWqrq6O1J+Mtra2p6fn+vXrGxsbTUxMoqKiYmNj8eisAwcOXLZs2erVq+vr65WUlIKCglRUVIieeeTIkezfz0VFRbq6uvi1jo7OgwcP8GsXF5f4+PibN2+uWLECT8EwLDMzU01Nbfbs2Rs3biwrKwsMDNTX11+9ejW7Ybdu3QIAZ2dnUpvfLfg8ySIwtDeLaGFhgWHYiRMnJCUliQCk+D/spKQkfBbx6tWrjo6O8vLy+vr6QUFBLBYLz8Zisfbv329iYoL3aadOnSKqo9Pp/v7+gwYNkpWVtbOzS09Px9OTkpIGDhwoKytbU1PTwetfv3719PTU1tbW0NDw9PS8du0a2dP0DAYjMDDQyMhIRkbG0tKSmMPEMKypqcnPz09NTc3Q0PDixYuTJk1atGgR8fTjx48DBgzAr3fs2LFlyxb8+syZM4sXL8av6+vrFRUVx44dS7z15s0bALh9+7ajo6OUlNTw4cM3b95Mp9PZzWtsbFRQUMD//QksSGDdAhdYZmYmvw3hG1+/fj116lRhYSF+y2KxBgwYsGvXLiIDu8DOnj1LiCowMBBfS8TZvn07ADx//hy/vXDhgri4eFNTUwdVnz9/HgDi4uK41xrugwTWLZDAWCyWjo6Oo6Pj27dvKysrAwMDxcTEiouLiQzsAvv8+bOmpubnz58bGhqMjY0fP35MZKurq9PU1CQW0Hx8fKysrDqot7m5efDgwQ4ODiS0iZugMRiiW1AolOjo6GXLlg0ePJhCoQwZMiQ2NlZDQ6PNzCoqKocOHZowYQKDwVi2bBn7ipmUlFRYWNj9+/fr6+ulpKTS09NHjx7dQb1FRUXz58//jv3BPAadB+sWDAbj06dPGhoaXTpn0Supra0VEhJiXxDrDvX19aKiohyL7z0RJDAEgkTQOhgCQSJIYAgEiSCBIRAkggSGQJAIEhgCQSJIYAgEiSCB8QguukDDHYEQJ6w65vs8ukVFRVlaWsrLy9vb2z99+rTNPAwGY+/evfr6+jIyMiNHjoyMjGyd586dO7GxsV2tvTeBBIbgJD4+3s3NzcLCIjQ0VFxc3MbGpqioqHW2X375Zfv27UuXLr18+bKlpaWbm1tCQgJ7BtwZDoebmr4GEhiCkwMHDkyaNOnEiROurq5//fWXoqLiqVOnWmc7c+bMypUrAwICHB0dT548OWrUKMLhwsePH0+cOEGlUtPT03lru8CBBMYHvn796uvrO2TIEElJyUGDBu3cuZPYT6Otrf3bb7+NHTtWWlp60KBBJ0+eLCkpmTZtmoKCwoABA/7880+ikPLycmdnZyUlJQMDg8DAQBaLhad34NGtg3oJqqqqkpKS3Nzc8FtJSUlHR8dLly61bkVzc7OCggJxq6KiQtSVnZ196dIlFoslsK4yeAd/9xr3HdjPVjk7OysoKOzfvz82NnbNmjUAQHhl0tLSwl21JSYmOjk5USgULS2tQ4cOJSQkWFpaSkhIfP36Fd/Cr6SktGLFiujo6HXr1lEolHXr1uEldODRrYN6CV68eAEADx48IFIOHz5MoVA4zmJhGObl5aWpqfno0aMvX74EBweLioqePXuWI8/gwYP9/f258OPrsSCB8Qh2gbm4uLCfnjQwMNiwYQN+raWl5ebmhl/jbirWrFmD3+KnPLOzs3GBzZ49myhh/fr1EhISlZWVHXt066Begjt37gDAq1eviJQLFy4AQHl5OUfOpqYmdl+8q1atat1qJDB0XIUP4BNuNBrt7du3jx8/zs/PJz7wAID4qx08eDAAEKc28FvcWRW05Svu5cuXHXt067heHAzDAIBCoXCktPYKvnr16uLi4rNnzw4ZMiQ5OXn37t1mZmbLly//7h9LrwQJjA+kpaWtWrUqOztbQ0PD1NSUYxqd40+5vYMwrX3FlZWVdezRreN6cdTU1ACgurqaSKmurhYXF2cfbgHAmzdvQkJCbt26NWHCBACwtrZmMpmbNm3y8PAgHHIgAE1y8J6qqioqlWpjY1NaWvrp06eEhITOezVkh91XHO7FTUdHpwOPbp2sV1NTk0Kh5ObmEil5eXm4gNnBpwfZT0xaWFhUVFTk5+d/R1t6MUhgvCYzM7O5uXntKzpbdwAAAXZJREFU2rW4j8HGxsbCwsLvKIfDV5y8vLyRkVEHHt06Wa+ioiKVSsU9RgEAg8GIj493dXXlyGZgYAAAaWlpREpaWpqEhAThNAqBgz4ReY2+vr6oqOimTZt8fHwqKyuDgoJqa2tfvnxZUVHB7mTzm1y6dElRUXHKlCnJycn79u0LCgqSlpbuwKNb5+v19/efNm1aYGCgg4PDiRMnqqqqCG9qp0+fTkpKOnfunKWl5dSpU5cuXbp9+3Z9ff2UlJRff/1127ZtveAMMpfh9yxLX4F9FjE8PNzAwEBaWtrS0jIuLu6PP/5QUFDYtm0bhmFaWlr79+/HszEYDACIjIzEb/FJxadPn+KziAkJCePGjZOTkzM2Nv7tt9+Iijrw6NZBvRxERERYWlrKyck5ODhkZWUR6cuWLQMAGo2GYVhdXd2mTZv09fWlpaXNzMxCQkIIf3UEaBYRuQxAIEgEjcEQCBJBAkMgSAQJDIEgESQwBIJEkMAQCBL5P6a8WTxJ6d24AAAAAElFTkSuQmCC" /><!-- --></p>
</div>
<div id="parallel-computation" class="section level3">
<h3>Parallel computation</h3>
<p>Parallelization of the association analysis is an easy task with
<code>solareclipser</code>. The user only needs to specify how many
cores are to be used for the computation.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a>A6 <span class="ot">&lt;-</span> <span class="fu">solarAssoc</span>(trait <span class="sc">~</span> <span class="dv">1</span>, phenodata, <span class="at">snpdata =</span> genodata, <span class="at">kinship =</span> kin, <span class="at">cores =</span> <span class="dv">2</span>)</span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;snp.geno-list&quot;</span></span></code></pre></div>
<p>The gain in CPU time is not obvious, as the number of SNPs (50) is
too small.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a>A1<span class="sc">$</span>assoc<span class="sc">$</span>tprofile<span class="sc">$</span>cputime.sec</span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a><span class="co">#&gt; [1] 1.213</span></span>
<span id="cb46-3"><a href="#cb46-3" tabindex="-1"></a>A6<span class="sc">$</span>assoc<span class="sc">$</span>tprofile<span class="sc">$</span>cputime.sec</span>
<span id="cb46-4"><a href="#cb46-4" tabindex="-1"></a><span class="co">#&gt; [1] 0.805</span></span></code></pre></div>
<p><code>dat30</code> data set contains 100 SNPs
(<code>genocovdat30</code> matrix). The same test on speed up by
parallel computing looks as following.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" tabindex="-1"></a>A7 <span class="ot">&lt;-</span> <span class="fu">solarAssoc</span>(trait1 <span class="sc">~</span> <span class="dv">1</span>, dat30, </span>
<span id="cb47-2"><a href="#cb47-2" tabindex="-1"></a>  <span class="at">snpcovdata =</span> genocovdat30, <span class="at">snpmap =</span> mapdat30)</span>
<span id="cb47-3"><a href="#cb47-3" tabindex="-1"></a><span class="co">#&gt; [1] &quot;snp.geno-list&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a>A8 <span class="ot">&lt;-</span> <span class="fu">solarAssoc</span>(trait1 <span class="sc">~</span> <span class="dv">1</span>, dat30, </span>
<span id="cb48-2"><a href="#cb48-2" tabindex="-1"></a>  <span class="at">snpcovdata =</span> genocovdat30, <span class="at">snpmap =</span> mapdat30,</span>
<span id="cb48-3"><a href="#cb48-3" tabindex="-1"></a>  <span class="at">cores =</span> <span class="dv">2</span>)</span>
<span id="cb48-4"><a href="#cb48-4" tabindex="-1"></a><span class="co">#&gt; [1] &quot;snp.geno-list&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" tabindex="-1"></a>A7<span class="sc">$</span>assoc<span class="sc">$</span>tprofile<span class="sc">$</span>cputime.sec</span>
<span id="cb49-2"><a href="#cb49-2" tabindex="-1"></a><span class="co">#&gt; [1] 1.995</span></span>
<span id="cb49-3"><a href="#cb49-3" tabindex="-1"></a>A8<span class="sc">$</span>assoc<span class="sc">$</span>tprofile<span class="sc">$</span>cputime.sec</span>
<span id="cb49-4"><a href="#cb49-4" tabindex="-1"></a><span class="co">#&gt; [1] 1.064</span></span></code></pre></div>
</div>
</div>
<div id="linkage-model" class="section level2">
<h2>Linkage model</h2>
<p><code>solarMultipoint</code> function performs the linkage analysis
by calling <code>SOLAR</code> command <a href="http://helix.nih.gov/Documentation/solar-6.6.2-doc/91.appendix_1_text.html#multipoint">multipoint</a>.
The function looks for multipoint IBD matrices in the directory pointed
out by <code>mibddir</code> argument. The analysis can be customized by
combination of options via <code>multipoint.options</code> and
<code>multipoint.settings</code> arguments.</p>
<p>Phenotypes in <code>dat30</code> data set and IBD matrices are placed
in <a href="https://github.com/ugcd/solarius/tree/master/inst/extdata/solarOutput">inst/extdata/solarOutput</a>
directory of the package.</p>
<div id="basic-linkage-model" class="section level3">
<h3>Basic linkage model</h3>
<p>The basic model requires only the path to a directory with IBD
matrices (<code>mibddir</code> argument). By default, all
<code>SOLAR</code> specific files, which represent IBD matrices, will be
used in the analysis.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a>mibddir <span class="ot">&lt;-</span> <span class="fu">package.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;solarOutput&quot;</span>, <span class="st">&quot;solarMibdsCsv&quot;</span>, <span class="at">package =</span> <span class="st">&quot;solareclipser&quot;</span>)  </span>
<span id="cb50-2"><a href="#cb50-2" tabindex="-1"></a><span class="fu">list.files</span>(mibddir)</span>
<span id="cb50-3"><a href="#cb50-3" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;mibd.2.0.gz&quot; &quot;mibd.2.1.gz&quot; &quot;mibd.2.2.gz&quot; &quot;mibd.2.3.gz&quot; &quot;mibd.2.4.gz&quot;</span></span>
<span id="cb50-4"><a href="#cb50-4" tabindex="-1"></a><span class="co">#&gt;  [6] &quot;mibd.2.5.gz&quot; &quot;mibd.5.0.gz&quot; &quot;mibd.5.1.gz&quot; &quot;mibd.5.2.gz&quot; &quot;mibd.5.3.gz&quot;</span></span>
<span id="cb50-5"><a href="#cb50-5" tabindex="-1"></a><span class="co">#&gt; [11] &quot;mibd.5.4.gz&quot; &quot;mibd.5.5.gz&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" tabindex="-1"></a>L1 <span class="ot">&lt;-</span> <span class="fu">solarMultipoint</span>(trait1 <span class="sc">~</span> <span class="dv">1</span>, dat30, <span class="at">mibddir =</span> mibddir)</span>
<span id="cb51-2"><a href="#cb51-2" tabindex="-1"></a>L1</span>
<span id="cb51-3"><a href="#cb51-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb51-4"><a href="#cb51-4" tabindex="-1"></a><span class="co">#&gt; Call: solarMultipoint(formula = trait1 ~ 1, data = dat30, mibddir = mibddir)</span></span>
<span id="cb51-5"><a href="#cb51-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb51-6"><a href="#cb51-6" tabindex="-1"></a><span class="co">#&gt;  Input IBD data:</span></span>
<span id="cb51-7"><a href="#cb51-7" tabindex="-1"></a><span class="co">#&gt;   *  directory /tmp/Rtmp8xhLLr/Rinst2e9d41898f786/solareclipser/extdata/solarOutput/solarMibdsCsv</span></span>
<span id="cb51-8"><a href="#cb51-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb51-9"><a href="#cb51-9" tabindex="-1"></a><span class="co">#&gt;  Output results of association:</span></span>
<span id="cb51-10"><a href="#cb51-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb51-11"><a href="#cb51-11" tabindex="-1"></a><span class="co">#&gt;   * Table of association results (first 5 out of 12 rows):</span></span>
<span id="cb51-12"><a href="#cb51-12" tabindex="-1"></a><span class="co">#&gt;   chr pos    LOD  Loglike      H2r     H2q1</span></span>
<span id="cb51-13"><a href="#cb51-13" tabindex="-1"></a><span class="co">#&gt; 1   2   0 1.4888 -208.503 0.448464 0.403780</span></span>
<span id="cb51-14"><a href="#cb51-14" tabindex="-1"></a><span class="co">#&gt; 2   2   1 2.4232 -206.352 0.244560 0.614310</span></span>
<span id="cb51-15"><a href="#cb51-15" tabindex="-1"></a><span class="co">#&gt; 3   2   2 3.2759 -204.388 0.123184 0.742126</span></span>
<span id="cb51-16"><a href="#cb51-16" tabindex="-1"></a><span class="co">#&gt; 4   2   3 3.5569 -203.741 0.097187 0.763608</span></span>
<span id="cb51-17"><a href="#cb51-17" tabindex="-1"></a><span class="co">#&gt; 5   2   4 3.1610 -204.653 0.179701 0.668166</span></span>
<span id="cb51-18"><a href="#cb51-18" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb51-19"><a href="#cb51-19" tabindex="-1"></a><span class="co">#&gt;  CPU time on 1 core(s): 00:00:01</span></span></code></pre></div>
<p>Methods <code>print</code>, <code>summary</code> and
<code>plot</code> help to explore the results of analysis. The data
table of results is stored in <code>lodf</code> slot of the returned
object.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" tabindex="-1"></a><span class="fu">summary</span>(L1)</span>
<span id="cb52-2"><a href="#cb52-2" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb52-3"><a href="#cb52-3" tabindex="-1"></a><span class="co">#&gt; Call: solarMultipoint(formula = trait1 ~ 1, data = dat30, mibddir = mibddir)</span></span>
<span id="cb52-4"><a href="#cb52-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb52-5"><a href="#cb52-5" tabindex="-1"></a><span class="co">#&gt; Multipoint model</span></span>
<span id="cb52-6"><a href="#cb52-6" tabindex="-1"></a><span class="co">#&gt;  * Number of used markers: 12 </span></span>
<span id="cb52-7"><a href="#cb52-7" tabindex="-1"></a><span class="co">#&gt;  * Number of passes: 1 </span></span>
<span id="cb52-8"><a href="#cb52-8" tabindex="-1"></a><span class="co">#&gt;  * Maximum LOD score: 3.56 </span></span>
<span id="cb52-9"><a href="#cb52-9" tabindex="-1"></a><span class="co">#&gt;   -- chr: 2 </span></span>
<span id="cb52-10"><a href="#cb52-10" tabindex="-1"></a><span class="co">#&gt;   -- position: 3 cM</span></span></code></pre></div>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" tabindex="-1"></a><span class="fu">plot</span>(L1)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAADAFBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////isF19AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAW1klEQVR4nO2deXwTRf/HNz1sESjIoS2HB3IpHmBBQVFRHxAVUESe1lIpCA+oIBTEBxGwKGI5BZEbCsULRFF4RA45ytFygyBnBbQcAiV30jY90sxvN9nZ2SSTzG6O3fxezOePzWdnd+b77XuzyWZ2Z8oAKr9i1E4g0kUBEUQBEUQBEUQBEUQBESQd0JaPMIWtB6ANJS8a/FT3vzVyJR3QhFqYwm7/RRsWMNf9VPe/NXIVACAzbkPpLyPjfSPwvzWiJRlQZ4ZhZoL2mVs6pAOwvENC/ecK2NIHBvAbDtaqdYtvBP63RrQkAzr7Zo3D10H7bvVHbwXZTN/FY+9J0DsBuTawmukXgf+tkSuZp1h7Jo9dPtbCDkABs80JSDj3xAj0nRpNqQQV2aj6zQKoJWcrqtjFKmaDb0BTu69u9sDW9Q+i6jcLoG5Ov21Mj5Yaf4Am7wLmN+M0S1D1mwXQK5wdwzw18dut/gA5Zb0sWrmZAOmjstjlKSIgN90EgGoCHtBBZjG7zBYA1XTtcLMDymaW/ekCVHrHnbO+eb0pM+SaE5BzA6CALj4Wv4D/DDrUuVbzkdaMWt84ATk3AAroJhUFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRJBUQIVrVVChK7ZOjdh7ZQLK/ewrxfXZSlfs399VPnbuO3IBrStUXOsgoGnKxz5NAfkXBUQQBUQQBUQQBUSQYoBynmjVa3eASQYN6NlGjRoNC6yqUoB2tVh3bEjvwHIMHtD9h06ePB1YVaUAzf93YeHWVoHlGDSgY20CDFyoHKDjR1hIvQJMMlhAm+7v3Pq1fYHVVfBDeuGDPwVYM1hAP6fsPd3/9cDqKgbo0GtPbw6sZmi+xda1CKyeUoBOdsk8E1iGhcEDWr++sHBDgJ9DSgFa8GJg+TkVLKAlbXedHjQ0sLpKAcpkr0Qa3RdYjsGfYh8nPzjwaGBV6ZU0QRQQQRQQQRQQQRQQQRQQQfIBfbRAcX0EAQ1VPvZ8uYDWDGDVP2MAVH/k+mMKM0RWVElyfVfhD67Yf0gKMwCzXVoYbKWpct9BR9iFoURYR2OirEZkq6Er1yJbDl21FlmrUMmILGq0hBt9dxi+g3LYhQWFsThQGMHaRGEqoLWjtvG5o0F+HrlXjebXKCB87hSQK2EKSKhEAVFAwkYK6GYFtPmxVcIfoTSgs90+FHaNUEDaN5gGUV/CP0JZQBWfxDXUZFR65x5BgNY0rJl9dSgzwlWuLKCCNlFDLi2O/ZfZK/fQAcrZY7FYdAYLlF5wBpE1Q2fUCtZkYhcnnmO6nzZrjdM0qc7dzaKWsI3q2MXu5a7YRxZw23SY3UxawRqN0Jm1JsFybV8fFd1mB5v7+lrtLuDCYHM3jJQJKKh3UNWcWokrXR/SK2Of46oq+A765c74rApn7gdvv6fQI/fIOMWOddC8oQf8t9hvtdsXKwjo2hvMU2cAn/uFFvX3gogDZBwT3Xw7nzn3RziPo0KAqhbXq7vYtYsz92uP1Pw10gDlNY8dWw4zdxruOCoD6NwzTN9i3rtyt3aPWRpRgPRDmI5HUeauP4I9jhsUAFQ5Na6xcOEFc6/op8mKIEBrGtSZU+Z9Jc0ex3lCpXAB4r7bjd5X0o4sZoiQpsqALnRlelzC/tSoSNNkwdLwADKNiHroAP6nxhdRvctgwmoCqppTM4nrM8X+FrOPY4bzO4cF0P+a1uC+2/G/xb6O76TjE1YR0JH2ru92Xz9WheMYBkAX+zJPn+Wsjx+rO+q0ueRKWDVA5qHRbQr4eD5+za+Ke9pZM/SAZiY0WOkq9/Vr/ljSnaecCasGaHzsJPhd67O7Y3vCg1dAGABtYVJu8NZnd8ffrevlAxUB6RIGSegPOtGk0fEwAOp4nw1a3/1Bhs5xa1QE9H78H1I6zIpa37Y75IDWMd9K6TArfy16gWqArt46SlqPInccQwzI8XCHG5J6FO1vMWMtKgEaVuuyxC5X9jjOCi2gVcw2qV2uU5l+xUKhkoCK4iZK7pOuGsxMxzQaMKCqVs9I75NeHP2yZ+5ACUBv1jPJ6LTvX+OCd6MBA1rOFMjotF/E/OCRO1AA0J8x2XLualxKet670UABVTZ7SdZdja5JJrfcXY2GGVBqQ4us2z7LmR+9Gg0U0DzNUVmADsdnuuXualQqoPUD+02HlxTSAZ2ImiPzvtiL8DAGDcjWuK/M+2KToo+KcucblQjor4Fa64ewQ0U6oJcblckEdC5+lGejAQKaHn1aJqCK1h3sKHe+UYmA9rL5rZvDr0gGdMg5/aa8O6vwMAYLyHp7fyD3zupOzQIhd9io5M8g07FRB7mIZrN56WGHw2EocUCZBWc1Imt3OLrdXc46m9YOC2026OxaZK1CJaPVYWMPo3ujJQZ2cSiXB7SMXbGgMJZqoW0tsmzbk2LPOcOUw8IqFAafOxsmLeEfmDtslLWVkgAdeOfta+zLteTk5EE7tJK0gVkgbUc3rdfMwJRuX+ZK49CX0lo5X3dAALFP1+2LKb02QgogAH7OYhe2rVu3zthfXl6uN5dDGQVnMiBrK3/y/lLOWbU2WGi1QmfTImsSKhk4m1KnyK1Rs55d7FvhSuLIYm4bCmMU2i5xC/N+/HlXmBJYWIbC4HPnwnzBbHLlDlviGi2B325+AP20DYAz/+FXJH4GbWLWOp3cpzuu100HwX4G3ag9ig8j7+mO6o4tbAF9Bu3OvFoyaza/IhHQo484UBA+nqTHX+Zx03cHB2hEzet8GJmPvxyP+SSwD+nVg9KnWXgvDdBaZqMoCG8lAXIexqAA/V1jPAwj9/mgkXFnlbmSNj/8hDgIb6U9QMUdxqAAvVlHD8PIBWRp3E0ZQMuYHeIgvJX4hNmIuMJgAJ2M/VQII/sJs++Z75UAZG/xnFsQ3koExB7GYACl1oefBoE8gvdS4hUFAC1lCtyC8FbqM4rfM7lCoWxAJ6OmoTDyARXVfDv8gCrueRH3dIf0hzhfukOAIBvQq4lC72lAD3FOjoKTloUP0BeaguAAna8BL19lAzqsmRvcU64VrZLt4tzDAKiscUqwjwGPjzrAO7mAut9lDvIx4M0a+DxFuABNjT4dLCDtfe35wygTUD6TG/Rz0v0S/hHlHnpA1oYDg3+QfKdmvsvJBPRMy6qgARXXf12Ue+gBZcVeCMGT9un8YZQH6DdmTQietF/I/wwIDyBdwjv+n5MGkgBdvy3N6eQB6vhgdQgAVXdqbhNyDzmg9+MvhwIQWMhs5ZwsQOuYX0IyVuOP2Cwh91ADunbreyAkgPjDKAeQo20HR2gGs4yKOwvCBGh4rWIQEkDsYZwE5AFaxWwHoQFkadLFER5AF+MmwiAw80ABgUzuMMoAZG/9JGdDMhzqR+a78AAaVNcgBOEzDxiQpUlXWYCWM7s4G5rxYj3YXzthAPRnzGeiIK7MAwYEfmBWyQDE3WzmFBpARTWHhwPQ6w0toiCuzAMHBHokGqUD4m42cwrRiMMpUftCD+hk1Gy3ICBIQOxhlAyojLvZzClEgCruTy4N+XixHknFpPFi/EAuNKbL73ixiVFbpY4X+zT6IGm8mEv43L3Hi23WTAtovNgK33dW8zSLRHdWRXcnJd9Z9Wy0/P52Wml3Votu78/bMlEYWXdWPXPPSPhD8p1VkfycYs/eDd/SIRv1vFvDzS0i4RQbG3uetyEb9ayr/0poP4O2MF95BQkWEJi0E0gBdDVhKLShGxb+1YrQAurSEnbFKT9ufnzcJWgjdtz8QWaRryC8DR+gknoZkT+xwCtNbwiFSgOaEX0g4gGdifpctakpyhulRv7UFP1vv6EaoMWa/REP6FJstmqTm9hb9Ij8yU2GJRhVA/Qdkx/xgIprjFNvepx2T/kf9QwiANC4+KuqAdrAbIp4QOa6wzBP2gtBeBsmQJ0fdkQ8oOzYv1UDtJv53us5aRgmUgDZkrjntlUC9MK99ogHNE9zAqgF6JhmKYh0QPZ7nf98Xh1AKY0rQKQD+opxPnOkCqDzMZ+7wkQwIMcDz7paVgPQ4HpWVxhVAe0bnjYR/lT3BrTOdRddFUDX4ifxYRQEVDjh9S5pE/8U1kFxSqFt3sf8ijegx9u5klMD0OiaWj6McoBGRd/Ta3CvZtGjhSp5UwC4nMaveAHawY/KUAOQrvZoGEYxQAtjvuX2cKyOFToIbWzrOyfyK16AurbmW1QBUFbsRRhGMUAdP+DNhE4AaXcGN4McbrzYDo3EQVxByNd4sYv10sMe22u8WPwm3mytgY7U5JFFzsOZm5s7eW9ZWZneVAbVswn0Rr1QaCqFzqJF1gJdqRZZ1JLeKFjkTFyjBStceRxZ5B5mSvRx/2Gs0JaIwiArCqjzlbvFc7wYs583h4VPpcrMHOEN7HmKnY2aK7w3lT7FKpukoDCKnWIYQLvGASQPQAPqCVZxQMuYwyiMcoD+lepSNwFQTk9W/fgVd0CXb4Ef3soDsrfsKnmiyRAC6oEEcHIHNKL2RdSywoC+ZzarAYgkN0C6WmN9DMl0C8LbEANK7qjSjORXx77QqsekywAvN0AT4q6qBmgTs0EdQF/Vbpo2Lv2ueusBVmJAltve9jFmVQFATz3kUAXQybhxXISKD6OOA5zEgKZFn1MN0H7mO3Um/U95iTc9+gCcRIDKG/XDjzhUAlCPZlXqAGrJJwO+vhvgJAK0UPOHaoBORS3CjhcLP6CELbzZlgBwQoDszXsC1QClJZapBOhxONXajCcATgjQt0w+UAvQXzEzgEqAPkl0zZhXnPgJwEkA5GjXxSOIgoDeqsPNVaUKoIqHm63WA/3qZg8JrbtJAPQLs8UjiHKArtdw/sJR5zpIO1DD3MpoBqDnxdwkAHqircMjiHKA3r/VmZ1a/9vnysYFG31dSAuAdvIzD6oByFTH9XS3uv/86FRbzxKnIKDn77V7BlEM0ORYZ/+dyoAO43++8oCOaXK8gigFqLThQNd6JAPq26TCK4hSgL6IOuVaj2BAJ6LhjFTKA7pxF/wRFMGAMurBiMoDms8c4jeqAugDqAw/gE7dAm+0Kg/I3Lob3KgKoAeQAE45+Var9e1bL1p5GayC0yNrgc6kRdYEnUWLLKqvFzUlOKOOXeyB48UWsisrmY3eu5m0yIrCmDFhdEZMGIPOV+5GCdMEumnFIbvdfmClHcokOIsB2UroyrTIlkFXqUXWIlQyIIsatXKNHuTHix1dyq78NVPYaK5CYQRbKgpjg7ZCFMaKy13vK/fyUTIB0f+SSRAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFRBAFhFVBJXQUEE5XUkqhpYAwmvpaTwrI/ynWhwKSAuhGr169hu80GAw6nQFK5HCFeq1esHpMIbE+Z/NyXFkcnsdt02LDIIvaxobBx/bZ6A3P8WL+AFnmzp2bVVBSUqIzlkDpBWdE1mCFzqxF1gydVYusAbWEbVTHLvKFG4dc26IwgjNrBWsihMHmbtD5yt0k7cYhPcUoIB+5U0CuhOmVtFCJAqKAhI0UEAVEAVFAFJCHKCCCKCCCKCCCKCCCKCCCKCCCKCCCKCCCKCCCKCCCKCCCKCCClntME2hAU+2FbppA1KjfaQJRBVEYM3aaQLQrPnc0TaCRME0gSSsOVlVV6S1VUCbBmQ3IVkBXqkW2FLoKLbJmoZIBWdSoRc8uDgjjxTzCVKIwyIrClEFbjtrG5673lbttlExA9BQjiAIiiAIiiAIiiAIiiAIiiAIiiAIiiAIiiAIiiAIiiAIiiAIiiAIiiAIiiAIiiAIiiALC6OTw/kthNQrIW/aBx23/zeNXKCBvHXkfgD1Z/AoF5K1NcwEoeodfoYC8tWYpAMXprNGlp6eP3GU0GnV6I5ROcHqRFZxBaxAsclo9Zlffje7McSVyeD63TYfZzaBFNYQwRlFsQhidqD5yXH3tSDKgjV+y76C3OOJTpkwZn+8+WaNodsnQTTSJWvKeaFIcRnCBTTQpaimoiSaPfADA3ix+hZ5i3rJnXKj+aDu/QgFhdGbEUOF/1FFABFFABFFABFFABOV+c+DAge27DkDlCW7ndmT3Q5f/G7L50O3/DdmdQqXtyKJGd3GNfg0BfcxtQ2HycGH2iMIUQLtPFEZe7gVyAZ3IZTVsYi7UMsGNf1ewOSugmz5oCbTLl0O3ZNB0aFfkCJXeHY9pdOJwbnnCFbuY8+MyMWGmDRLaQWEWD5qJCTMcl/uEEZhGZzhzh1/fUgE59fIsTOGUVEzhlmSjd6EheYt3IUiZgimc2duzJOsNzG6/JJd6FxYn52F27T0DU5jdF1O4NVkvWqOAvEQBIYUa0HUTptBYjCksvVLtXVh9BfPngGIMSmC67llikBzGfqUMs6v03MvcGpUF6GYUBUSQDEDiLmqx0ERngvYNT5vo9Y/K1g/sN93mXd22wbvsvZ49ey6SEjzI2LjgHrGlA3LrohZJNNEZVHFKoW3exx6Ffw3UWj9c5V1/7hDvsnRLZaVdQvBgY+OCe8SWDsitixpJPNEZVB77xXQ5zaNwL3tdvG6OV/19I71ztKVLCx5sbFxwz9jSAbl1UYvVxytJG/sTaOdErx1Nx0Yd9CwzjDjpDejv9HfTJhvEJb6CBxUbG9wztnRAQhc1OUlWuzPOeJUdeOftax5FjqzfL3oDOvuZ3j4rW0rwYGLjg3vGlg5I6KKWkKRl8sgiXBs/Z3kUbFgCMIA4FaZICR5MbN/BxbFlfAaJu6jF8k6yMlPoiUT6aRsAZ/7jUTg7NfXfvVI9r+EKCwG44PZ+8RU8mNj44J6xZXyLibuo/Se5axxmt92ZV0tmzfYuxxzE/IHF1V/MkxI8yNiY4J6xZVwHibuo/SeZw15K9OznWbp6UPo0i5QcAVg7OGO2e6s+ggcZGxfcIza9kiaIAiKIAiKIAiKIAiKIAiKIAiIowgCVM8fADavzBauKR7TQJjJ/ci+OxszZ/NfCl1GEAaoa8w/oPNP5gtWUoYJN1Dh7+/dFMWdBh41hyyjCAHFiAflS+R3od3pi53bcy5gnWUDfdQlbNmoB2t9gaVLtZ09zz/clJvVjz5v/ta1x92zu3GrPMN25UwxuiPm5TY17f3TVWvMUu7iR0qDZODtInB57jl27Zw4LqOq2c+FKVDVAUUk/7H6pgdnR4dG8vEc7gIu3fHBoOrOXI8O+g9gXuAHENP7hZGq8q0N58HgAqts+X/DNHaNB4rLu2QAcrXWGBQReWBCuRFUDxLBvirLb5+ZFXwSgKHrXNs15ADYUIUBwA4j5DIBC5ryzVrtvAfi1pgGA3DdZQMuSARifWsQBGjMoXImqB0jHLl95a0Fzbq3ZorJOca/MvQQQILgBxGxhz0MeUNJmAKY+5mohcZku5gJo/aMTUPaL4UpUXUB9Bs93cmg+Dzj2TGh3y88IkLAhZicC1PRXACY/7mohcRnoOu1UjVInoE97hStR9QCtBcCW9PmOmMsAXIrJ286eSODl3ggQ3OAG6LFcANYmWABY0oEDtLjDJ32AE9BIXF9wSKQeoKbr9r1cz+ho3yl/T6f2jh1RX55cc9sUDtBTw4zchzS/wQ1Q5ij2UqlVz0OrG7/HAboR3WSVC1CXleFKVD1Avz5Uu8tJALRpdyRy3+ZzmsXdNa6KA7Ss/qvcC9wgBrSlLbu40rteUqaNAwSeibM4AZUkXA1XouoBspN38lL1vQXY8nmvBpeNH/3/AgRy+uBKq1sdDi4bP1IL0O93BwTIkarFlO6dEGQ2fhSBv8UiSxQQQRQQQRQQQRQQQRQQQRQQQRQQQf8HjdjFhnGWw44AAAAASUVORK5CYII=" /><!-- --></p>
<p>The results for chromosomes 2 and 5 are identical, since the IBD
matrices for chromosome 2 are duplicates of the matrices for chromosome
5.</p>
</div>
<div id="linkage-options" class="section level3">
<h3>Linkage options</h3>
<p>The following code shows an example of a custom linkage analysis. In
particular, the linkage is run on chromosome 5 with a 2 cM interval,
Fine mapping is disabled and the second pass is enables via
<code>multipoint.options</code> argument.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" tabindex="-1"></a>L2 <span class="ot">&lt;-</span> <span class="fu">solarMultipoint</span>(trait1 <span class="sc">~</span> <span class="dv">1</span>, dat30, <span class="at">mibddir =</span> mibddir, <span class="at">chr =</span> <span class="dv">5</span>, <span class="at">interval =</span> <span class="dv">2</span>, </span>
<span id="cb54-2"><a href="#cb54-2" tabindex="-1"></a>  <span class="at">multipoint.settings =</span> <span class="st">&quot;finemap off&quot;</span>, <span class="at">multipoint.options =</span> <span class="st">&quot;3&quot;</span>)</span>
<span id="cb54-3"><a href="#cb54-3" tabindex="-1"></a><span class="fu">summary</span>(L2)</span>
<span id="cb54-4"><a href="#cb54-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb54-5"><a href="#cb54-5" tabindex="-1"></a><span class="co">#&gt; Call: solarMultipoint(formula = trait1 ~ 1, data = dat30, mibddir = mibddir, </span></span>
<span id="cb54-6"><a href="#cb54-6" tabindex="-1"></a><span class="co">#&gt;     chr = 5, interval = 2, multipoint.options = &quot;3&quot;, multipoint.settings = &quot;finemap off&quot;)</span></span>
<span id="cb54-7"><a href="#cb54-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb54-8"><a href="#cb54-8" tabindex="-1"></a><span class="co">#&gt; Multipoint model</span></span>
<span id="cb54-9"><a href="#cb54-9" tabindex="-1"></a><span class="co">#&gt;  * Number of used markers: 3 </span></span>
<span id="cb54-10"><a href="#cb54-10" tabindex="-1"></a><span class="co">#&gt;  * Number of passes: 2 </span></span>
<span id="cb54-11"><a href="#cb54-11" tabindex="-1"></a><span class="co">#&gt;  * Maximum LOD score: 3.28 </span></span>
<span id="cb54-12"><a href="#cb54-12" tabindex="-1"></a><span class="co">#&gt;   -- chr: 5 </span></span>
<span id="cb54-13"><a href="#cb54-13" tabindex="-1"></a><span class="co">#&gt;   -- position: 2 cM</span></span></code></pre></div>
<p>The LOD scores for the second path (stored in <code>lodf2</code>) are
plotted with the code given below.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" tabindex="-1"></a><span class="fu">plot</span>(L2, <span class="at">pass =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAC6FBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8RERESEhITExMUFBQWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj5AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxeXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKiqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///82bxQOAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMIUlEQVR4nO2dfVwUdR7HByWVQs/qRDsvo0d70kPy7Ly8zvN6ULLMgxCBNHuywM5LyacrjzDP0kPz0qDDTk8sU+/MztBQDHwuO5SoTNMsWHYXFhQR5fvvzezOPuDOb77MKs7yms/7j53fbz77nfnxfs1vdndYBomALpLZAwh3IIgBghggiAGCGCCIoe2CNs/WWHlLmj9oeMChU66fhi9tFzQzWmPliBf9wVLpR51y/TR8CUFQnVZw8j8Z3cQK9NOwps2ChkqStIAGZW6OTyF6J77HVb8tldfelqYGe6Kju4gV6KdhTZsFffl41L4fadCIq14oohxp7LJp1/WwuwV5ApkFugr00/DF4BQbJBXLj4NvPEtUKm1xC/LNvUAF9iF9ss9QU46/3CqCblKaTc3yw2ppo1jQvPsKY28r2nC7v9wqgka421umjrwpQk/Q3O1U93jXiOX+cqsIGq00p0r3zFpVpCfITf2xgI6VBNk7zZEfD7KCWmEBQVeQKmiPtEx+zPEJusLzBKsLypHyvvIIOtnr56+vfKyvNOkHtyB3QBB0dHC3peo5aO/Q6Bsy6lOjV7oFuQOCIIsCQQwQxABBDBDEAEEMEMQAQQwQxABBDBDE0FZBlWstRplBQQWvvmspCp42Kmh9paWogCB9IIgBghggiAGCGDqKoHv79OnzrBk77iiC+u8tLz9kxo47iKADt5qxV4UOIuij/kNveXSnGXvuIILWJZYdGv+YGXvuIIIU1t9oxl47iKANGyorN5pyHuoggpYP2H5owpNm7LmDCKp8Oe729M/M2HFHEWQaEMQAQQwQxABBDBDEYFzQ7KWW4k2jgtak+RifJiRVJ9Ip04subZn/B5hn9Aja720125qFzzp1Rhg12sQbd7UIo3qnuOz8v8sKwOkSRi06ZbZT3tbZKWoDggKBILqogjakj5vfqLYhKJhv0231L61WOxAUTNkKovWL1A4EaVF7YMoetQlBWux+5ukf5EV1QkLC5G0OLzaHELtdHOmU1ehEepk4soVY5hu/7bm2CCJap/xlmCs3N3dOaYOKy+ZqEFGrE9mEUYNDHDnt4kwv0tmkTpmt1tuqy+AFfbCFqOIJtYMpFkxJ5omG1xeqHQjSoHBCymveHUEQAwQxQBADBDFAEAMEMUAQAwQxQBADBDFAEAMEMUAQAwQxQBADBDFAEAMEMUAQAwQxQBADBDFAEAMEMUAQQ15JrYrT5qwV4dCJbMKo1q4T1YRUViMuc+ps0ebwbft5g4JwBDFAEAMEMUAQAwQxQBADBDFAEAMEMUAQAwQxQBADBDFAEAMEMUAQAwQxQBADBDFAEAMEMUAQAwRpsHNy8qxqtQ1BwVQlVjYueVntQFAwxdlEx5LVDgQF01hPtG2W2oEgTUpSKwg3FhDimptxRFnWZWdnz/i0XsVlq6sX4RRHtTZhVO/QieziTCeq0dmkTpnN6W3VtuHGAmcy8/2HPqZYMNuzAjoQFEz+KJlxageCGCCIAYIYIIgBghggiAGCGCCIAYIYIIgBgqhy5mPDkmd9JSiFoCmdr0uYmBDb+QXtUssL+nvkqnNKdeFlb2mWWl7QL6erjZlDNEstL6jbR2qjKEqz1PKCpF1qY5/26xoEQRDpCxqe5GEEBCkECRrpR7PU8oI4IIhOTLv/5pF/PiYohaB3u/dNzkq59soN2qWWF1TeNeu0vGh6qdMXmqWWF5T4oNoYOUaz1PKCblqhNv7ZT7PU8jcW6LFZbWzpoSnI8kfQ3fPVxl9/pVlqeUGvxFS5l1Uxr2iWWl5Q052xhXayF8be0aRZanlBZEuPkC6XItKqg6rcQBDR8U1LN4neSEOQj4MDNEshyAuuB7mBIE0giCDIw8UXNN1LKgQpBAm6zY9mqeUFcUAQAwQxQBADBDFAEAMEMUAQAwQxQBADBGlS6vuBIUiL44knvU0I0mDeo6MgSH+KjYGgtgiqSUlJydjuVHHYHE4RdnHksAkjZ404sutkelFoZTa770nn/25eTxBuLMAIUsAU0wSCIKgVeCdNEOQBghQgiCDIAwQRBHmAIIIgARDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQwQxABBDBDEAEEMEMQAQQz5O1wqdbY6lwhnrTiyCSOXXRw5dDKdqEac1emU2ZzelrNtX17wgyOIAYIYIIgBghggiAGCGCCIAYIYIIgBghggiAGCGCwhqPlrCNIW5Ni3Zt6kkXFR90JQa0GNBzcunjL6ju6SJHXr/+Bzb3wAQaog9yEzPLaTbKZn3Nhpy4oOK//yAOcgcpQXvZU5Nu5yRUz/4ZPmrdl3MjC2rqCmw0XLpo2N6ymL6dLPLUZzuxYUpM6lzv651IyXeRlnteeQiVZOv7HuQ6ZejSz+PuiMLGbOpOH9IiTpMkXMiqLD5z3DqoLUuRTpmUuZf5PnknaV1QSd9swl5a1MV89cUk4/l+ajRvnk8W+fU9vhJ8ixb4VyyET4T78BW7kkgs6mf9H4YrHaCR9BnrnUPyrgdSm47JII2v8noh1z1I75gtS3Mj/xzyXhnXkvkaCPcomOPKN2TBQk+Fhg/uWONW8TVaVQ0I0FFgwQcqc40sv0otguipiBozPeWLu/KrxuLLBpsXwEPUVBNxZYmZYuIk0nGi+M0lN1oonZq0q/D88bC+yfTlQWPuegYMyeYmdTD5+bvVXtQJAGFc8/me8dOgQxQBADBDFAEEPByt0qZR+X7RZRUiqMdnwsjHYX7xJG27bqlImjreJsl07ZxyXeVplRQf8r8LJ4wpICEe/8QxjlTBBn+eJoxnPCqCBPHGVmibN8cfREtq+51aCgAFVxBw3XyLwfdzaUsuzEUKpo/OyQygavCloFQYFAEMNFEdR0XPv/RzE0HBe/VOngrAqliqrsIZV9Xx+0yrggiwFBDIYFBV6oNkSp+E2kmJ2Tk2cJ/uOZDhvSx81vDGFv1LgxaJVRQa0uVBsh4Ma5bacqsbJxyctGq75Nt9W/tNr43ohyJwWtMiqo1YVqAwTeOLftFGcTHUs2WlW2gmj9IuN7o50ZFy6o1YVqQ4wJQVCj/KqybZbxutoDU/YYr3I8X37hgnwXqg0TiiCZktQK40W7n3n6B8NFLXM+P3rhgnwXqg0TkiDX3IwjoeyM1hk/DWxcThdBUKsL1YYIRdCZzPwQ3l5+sIWo4gnDZQuTkv6QkFR73lrDr2KBF6oNEYqg7Vmh7Kkk80TD6wtDqbwIR1CrC9WGCEVQ/iiZcYbLCiekvCa+YqbDxRBkNSCIAYIYIIgBghggiAGCGMJM0GnpAFXXuxeaNP3C9/vHGOkrZdFyjfTlp4+234jCTFDz1O9p6AL3QpPsJ33NmIhsZbGzk/QlxW9qtxGFmSAFWZCI0738H+1jhg5UFlN/LQv617B2G41ZgnZd/Xbv7vceUr7fF9N7nDxv/j0gqt9CZW4NkqT7lCnmDSLX3Rp1/fueqjX3yA/ViVfHZp2lmPmXfS33rlskC2ru+XV7DdQ0QZ16v1fy4NV1LfF3FRffFU9Hu0zfO18qU8zIR5C88AYUec175UndPJeYJ84gOjfg96Ure71AMXn35RB9Fl0hC6L7l7bXQE0TJMkHxamf5hZ3Pkp0pPP2LRHfEG084hfkDSjyVaJK6Rt31cBVRB9e4SAqeFwWlBdHNCPpiCJo6oT2Gqh5gmrkx9FPLb1B6cW+dWpI19G535FfkDegyM3yPFQF9f4v0bzBni3E5NVEHqZb3ncLynmgvQZqrqAxE990e7hhCbXsmDmwyzq/IF8Quc0vqO+HRHPv9mwhJo9+99rBqJNuQX9JaK+BmidoLVFj7zc+iTxG9F1k8VZ5ItFDD/sFeYNWggYXEK3t4SJaHq8IWhb/yhhyC8oI6SpwWzBPUN/1Ox+60tkyaMinO4YMavmk0+LyNT2zFUH3POtUTtJq0EpQ5hT5rdLNo/YWXvNHRVB155+t9ggatqK9BmqeoA/v6D6snMiW3CtGeTVfFNv12qxmRVDeVY8oC28QKGjzAPnh+MNX9s5sVATRb7q63IIaepxor4GaJyiUb8Ocu75Uc/2SRy5sNDp0LEGUP0Zr7bmb913YaHQwS9Dn/UIS1JKk9WXZspkXNhg9wvCzWHgBQQwQxABBDBDEAEEMEMQAQQz/B5sb4m27WQhHAAAAAElFTkSuQmCC" /><!-- --></p>
</div>
<div id="bivariate-linkage" class="section level3">
<h3>Bivariate linkage</h3>
<p>The bivariate linkage analysis is performed in the same way as the
bivariate polygenic analysis.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" tabindex="-1"></a>L3 <span class="ot">&lt;-</span> <span class="fu">solarMultipoint</span>(trait1 <span class="sc">+</span> trait2 <span class="sc">~</span> <span class="dv">1</span>, dat30, <span class="at">mibddir =</span> mibddir, <span class="at">chr =</span> <span class="dv">5</span>, </span>
<span id="cb56-2"><a href="#cb56-2" tabindex="-1"></a>  <span class="at">interval =</span> <span class="dv">2</span>, <span class="at">multipoint.settings =</span> <span class="st">&quot;finemap off&quot;</span>)</span>
<span id="cb56-3"><a href="#cb56-3" tabindex="-1"></a><span class="fu">summary</span>(L3)</span>
<span id="cb56-4"><a href="#cb56-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb56-5"><a href="#cb56-5" tabindex="-1"></a><span class="co">#&gt; Call: solarMultipoint(formula = trait1 + trait2 ~ 1, data = dat30, </span></span>
<span id="cb56-6"><a href="#cb56-6" tabindex="-1"></a><span class="co">#&gt;     mibddir = mibddir, chr = 5, interval = 2, multipoint.settings = &quot;finemap off&quot;)</span></span>
<span id="cb56-7"><a href="#cb56-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb56-8"><a href="#cb56-8" tabindex="-1"></a><span class="co">#&gt; Multipoint model</span></span>
<span id="cb56-9"><a href="#cb56-9" tabindex="-1"></a><span class="co">#&gt;  * Number of used markers: 3 </span></span>
<span id="cb56-10"><a href="#cb56-10" tabindex="-1"></a><span class="co">#&gt;  * Number of passes: 1 </span></span>
<span id="cb56-11"><a href="#cb56-11" tabindex="-1"></a><span class="co">#&gt;  * Maximum LOD score: 2.47 </span></span>
<span id="cb56-12"><a href="#cb56-12" tabindex="-1"></a><span class="co">#&gt;   -- chr: 5 </span></span>
<span id="cb56-13"><a href="#cb56-13" tabindex="-1"></a><span class="co">#&gt;   -- position: 2 cM</span></span></code></pre></div>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" tabindex="-1"></a>L4 <span class="ot">&lt;-</span> <span class="fu">solarMultipoint</span>(trait1 <span class="sc">+</span> trait2 <span class="sc">~</span> age <span class="sc">+</span> <span class="fu">sex</span>(trait1), dat30, </span>
<span id="cb57-2"><a href="#cb57-2" tabindex="-1"></a>  <span class="at">mibddir =</span> mibddir, <span class="at">chr =</span> <span class="dv">5</span>, <span class="at">interval =</span> <span class="dv">2</span>, <span class="at">multipoint.settings =</span> <span class="st">&quot;finemap off&quot;</span>)</span>
<span id="cb57-3"><a href="#cb57-3" tabindex="-1"></a><span class="fu">summary</span>(L4)</span>
<span id="cb57-4"><a href="#cb57-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb57-5"><a href="#cb57-5" tabindex="-1"></a><span class="co">#&gt; Call: solarMultipoint(formula = trait1 + trait2 ~ age + sex(trait1), </span></span>
<span id="cb57-6"><a href="#cb57-6" tabindex="-1"></a><span class="co">#&gt;     data = dat30, mibddir = mibddir, chr = 5, interval = 2, multipoint.settings = &quot;finemap off&quot;)</span></span>
<span id="cb57-7"><a href="#cb57-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb57-8"><a href="#cb57-8" tabindex="-1"></a><span class="co">#&gt; Multipoint model</span></span>
<span id="cb57-9"><a href="#cb57-9" tabindex="-1"></a><span class="co">#&gt;  * Number of used markers: 3 </span></span>
<span id="cb57-10"><a href="#cb57-10" tabindex="-1"></a><span class="co">#&gt;  * Number of passes: 1 </span></span>
<span id="cb57-11"><a href="#cb57-11" tabindex="-1"></a><span class="co">#&gt;  * Maximum LOD score: 2.68 </span></span>
<span id="cb57-12"><a href="#cb57-12" tabindex="-1"></a><span class="co">#&gt;   -- chr: 5 </span></span>
<span id="cb57-13"><a href="#cb57-13" tabindex="-1"></a><span class="co">#&gt;   -- position: 2 cM</span></span></code></pre></div>
</div>
<div id="parallel-computation-1" class="section level3">
<h3>Parallel computation</h3>
<p>Parallelization of the linkage analysis is again parametrized with
<code>cores</code> argument, but the chromosome values are expected to
be a vector of values. That means that each core does the job on a
single chromosome.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" tabindex="-1"></a>L5 <span class="ot">&lt;-</span> <span class="fu">solarMultipoint</span>(trait1 <span class="sc">~</span> <span class="dv">1</span>, dat30, <span class="at">mibddir =</span> mibddir,</span>
<span id="cb58-2"><a href="#cb58-2" tabindex="-1"></a>  <span class="at">chr =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">5</span>), <span class="at">cores =</span> <span class="dv">2</span>)</span></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" tabindex="-1"></a>L1<span class="sc">$</span>multipoint<span class="sc">$</span>tprofile<span class="sc">$</span>cputime.sec</span>
<span id="cb59-2"><a href="#cb59-2" tabindex="-1"></a><span class="co">#&gt; [1] 1.063</span></span>
<span id="cb59-3"><a href="#cb59-3" tabindex="-1"></a>L5<span class="sc">$</span>multipoint<span class="sc">$</span>tprofile<span class="sc">$</span>cputime.sec</span>
<span id="cb59-4"><a href="#cb59-4" tabindex="-1"></a><span class="co">#&gt; [1] 0.662</span></span></code></pre></div>
</div>
</div>
</div>
<div id="miscellaneous-material" class="section level1">
<h1>Miscellaneous material</h1>
<div id="solar-references" class="section level2">
<h2><code>SOLAR</code> references</h2>
<p>A list of references given on the official web page of
<code>SOLAR</code> <a href="https://solar-eclipse-genetics.org">solar-eclipse-genetics.org</a>:</p>
<ul>
<li>The main reference for SOLAR, including theoretical explanations of
the variance component linkage method and the approximate multipoint IBD
calculations in pedigrees, is <span class="citation">[@Almasy1998]</span>;</li>
<li>Bivariate Quantitative trait linkage is described in the following
papers: <span class="citation">[@Almasy1997]</span>, <span class="citation">[@Williams1999]</span>;</li>
<li>The Liability Threshold model for discrete traits is described in
the preceding paper as well as the following one: <span class="citation">[@Duggirala1997]</span>;</li>
<li>Gene By Environment Interaction is discussed in: <span class="citation">[@Towne1997]</span>, <span class="citation">[@Blangero2009]</span>;</li>
<li>An examination of LOD adjustment is given in: <span class="citation">[@Blangero2000]</span>, <span class="citation">[@Blangero2001]</span>;</li>
<li>Additional references include: <span class="citation">[@Blangero1997]</span>, <span class="citation">[@Williams2004]</span>, <span class="citation">[@Williams1999b]</span>.</li>
</ul>
</div>
</div>
<div id="r-session-info" class="section level1">
<h1>R session info</h1>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb60-2"><a href="#cb60-2" tabindex="-1"></a><span class="co">#&gt; R version 4.1.2 (2021-11-01)</span></span>
<span id="cb60-3"><a href="#cb60-3" tabindex="-1"></a><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span id="cb60-4"><a href="#cb60-4" tabindex="-1"></a><span class="co">#&gt; Running under: Pop!_OS 22.04 LTS</span></span>
<span id="cb60-5"><a href="#cb60-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb60-6"><a href="#cb60-6" tabindex="-1"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb60-7"><a href="#cb60-7" tabindex="-1"></a><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0</span></span>
<span id="cb60-8"><a href="#cb60-8" tabindex="-1"></a><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span></span>
<span id="cb60-9"><a href="#cb60-9" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb60-10"><a href="#cb60-10" tabindex="-1"></a><span class="co">#&gt; locale:</span></span>
<span id="cb60-11"><a href="#cb60-11" tabindex="-1"></a><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span id="cb60-12"><a href="#cb60-12" tabindex="-1"></a><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=C              </span></span>
<span id="cb60-13"><a href="#cb60-13" tabindex="-1"></a><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span id="cb60-14"><a href="#cb60-14" tabindex="-1"></a><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span id="cb60-15"><a href="#cb60-15" tabindex="-1"></a><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span id="cb60-16"><a href="#cb60-16" tabindex="-1"></a><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span id="cb60-17"><a href="#cb60-17" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb60-18"><a href="#cb60-18" tabindex="-1"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb60-19"><a href="#cb60-19" tabindex="-1"></a><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb60-20"><a href="#cb60-20" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb60-21"><a href="#cb60-21" tabindex="-1"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb60-22"><a href="#cb60-22" tabindex="-1"></a><span class="co">#&gt; [1] gridExtra_2.3       ggplot2_3.4.4       plyr_1.8.9         </span></span>
<span id="cb60-23"><a href="#cb60-23" tabindex="-1"></a><span class="co">#&gt; [4] solareclipser_0.3.3</span></span>
<span id="cb60-24"><a href="#cb60-24" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb60-25"><a href="#cb60-25" tabindex="-1"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb60-26"><a href="#cb60-26" tabindex="-1"></a><span class="co">#&gt;  [1] tidyselect_1.2.0  xfun_0.43         bslib_0.5.1       qqman_0.1.9      </span></span>
<span id="cb60-27"><a href="#cb60-27" tabindex="-1"></a><span class="co">#&gt;  [5] lattice_0.20-45   colorspace_2.1-0  vctrs_0.6.4       generics_0.1.3   </span></span>
<span id="cb60-28"><a href="#cb60-28" tabindex="-1"></a><span class="co">#&gt;  [9] htmltools_0.5.8.1 yaml_2.3.7        utf8_1.2.3        rlang_1.1.1      </span></span>
<span id="cb60-29"><a href="#cb60-29" tabindex="-1"></a><span class="co">#&gt; [13] jquerylib_0.1.4   pillar_1.9.0      glue_1.6.2        withr_2.5.1      </span></span>
<span id="cb60-30"><a href="#cb60-30" tabindex="-1"></a><span class="co">#&gt; [17] calibrate_1.7.7   foreach_1.5.2     lifecycle_1.0.3   munsell_0.5.0    </span></span>
<span id="cb60-31"><a href="#cb60-31" tabindex="-1"></a><span class="co">#&gt; [21] gtable_0.3.4      codetools_0.2-18  evaluate_0.22     labeling_0.4.3   </span></span>
<span id="cb60-32"><a href="#cb60-32" tabindex="-1"></a><span class="co">#&gt; [25] knitr_1.46        fastmap_1.1.1     doParallel_1.0.17 parallel_4.1.2   </span></span>
<span id="cb60-33"><a href="#cb60-33" tabindex="-1"></a><span class="co">#&gt; [29] fansi_1.0.5       highr_0.10        Rcpp_1.0.12       scales_1.2.1     </span></span>
<span id="cb60-34"><a href="#cb60-34" tabindex="-1"></a><span class="co">#&gt; [33] cachem_1.0.8      jsonlite_1.8.7    farver_2.1.1      digest_0.6.33    </span></span>
<span id="cb60-35"><a href="#cb60-35" tabindex="-1"></a><span class="co">#&gt; [37] dplyr_1.1.3       grid_4.1.2        quadprog_1.5-8    kinship2_1.9.6.1 </span></span>
<span id="cb60-36"><a href="#cb60-36" tabindex="-1"></a><span class="co">#&gt; [41] cli_3.6.1         tools_4.1.2       magrittr_2.0.3    sass_0.4.7       </span></span>
<span id="cb60-37"><a href="#cb60-37" tabindex="-1"></a><span class="co">#&gt; [45] tibble_3.2.1      crayon_1.5.2      pkgconfig_2.0.3   MASS_7.3-55      </span></span>
<span id="cb60-38"><a href="#cb60-38" tabindex="-1"></a><span class="co">#&gt; [49] Matrix_1.4-0      data.table_1.14.8 rmarkdown_2.25    rstudioapi_0.15.0</span></span>
<span id="cb60-39"><a href="#cb60-39" tabindex="-1"></a><span class="co">#&gt; [53] iterators_1.0.14  R6_2.5.1          compiler_4.1.2</span></span></code></pre></div>
</div>
<div id="license" class="section level1">
<h1>License</h1>
<p>This document is licensed under the Creative Commons Attribution 4.0
International Public License.</p>
<p><a href="http://creativecommons.org/licenses/by/4.0/"><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFgAAAAfCAMAAABUFvrSAAAAAXNSR0IB2cksfwAAAARnQU1BAACxjnz7UZMAAAAgY0hSTQAAeiUAAICDAAD5/wAAgOkAAHUwAADqYAAAOpgAABdvkl/FRgAAAW5QTFRFAAAAAAAA////////////7+/v39/f1tXV09bS0tXS0tXR0dTR0dTQ0NTQ0NPPz9PPztLOztHNzdHNzdHMz8/PzdDMzNDMzNDLzM/Ly8/Ly8/Ky87Kys3Jyc3Jyc3IyMzIyMzHx8vHxsrGxsrFxcnFxcnExMnExMjDw8jDxMfDw8fCwsfCwcXAwMXAwMW/wMS/v8S+v8O+vsO+vsK9vcK9vcK8v7+/vMG8vMG7vMC8u8C7u8C6ur+6ur+5ub65ub64uL23t7y2tru1tbq0tLqztLmzs7iysrixsrexsbewsbawsLavsLWvr7Wur7SusLOvrrStrrOtr7KvrbOsrLKrr6+vq7Gqn6OenqCdn5+flpmWk5iTkZSRkZORj4+PiYyJhIaEhIWEgoWCgICAfX98fH98eXx5cHJvcHBwYGBgXV5dUFFQUFBQQ0RDQEBAPj8+NTY1MjMxMDAwKSkpKCkoICAgGxsbEBAQDg4ODQ4N2y3MbAAAAAR0Uk5T/wAKDnDBpeYAAALVSURBVHjatZX9V9JgFMdvNQh1Tme2zU1othSl5WsQZoqIrxmmqaUpqS2JKczibfLfd5+hHCYybB6/B859ftg+5+57Xx54Ag+iR/hPxGPRSXVYkf2SwHGC2C8rQXUiOhdPLK992tze3vn6/wKLOxseCykBkWXoNp+vrYNh+wJKaDQ8S8jJze0dl2TkvgkOSCx9kqsAqpI7prtFeUh9i+SV9eRWY8rpAqAKaWdwDLkyz6TKUFP5kOECg2p4Lr60ut6Q8kERwMhkDIDigRM4OhaUOfocT6auRSKabuLxfORZYGg0GltAM26k/O0Ssl4K5c3C5YEDeDI0wBOuqXmoqvYRXRzh5NDEzHwCU7aDi6DjMwDkQSg6gFVFYpCb91I1efKYMyMqahhT/rhh8yIN2adVMCoL6ebg4RdsCrmYrp182B0Ijs/EF1eTW/XgAtRl4IVCc7Ai0mUweymbPCaU6T5FfRdLrNi9AINC6QA6iQY0B8s9JwCanUvsg2NWDk3Nohd2cIZCAYrEjAPY35UDE42IZA1Dq4Yf+IoJOaafeLG0tuEOLNEV0BEIRMauFXTysZUOcXD0/fziWtKdFUI7AKaYh5I3UtrPY3vslvYtL9oERY1iwyG4oXgALYvH+wAiFGX5XAvWF/i4l41gbDfKod3cgu0DksEBuYsVnt6ShlaQ0GCF00g7F++qaleBZFOhpcbiEXIRIK9n8q2WkL/rHEyq1m67hhWoErabPzj+odpubtam3JNCL24fkNdTcwvLZPLcLPpX1kh7HUbaJRiX0N5tSyjFBoYncPDIEnIFhlOJOUNy3RryWmtT+gn31G+hs37RezQT4A/Nn8K99YvrPLtxNdHPv0MzVeeDHFqTBWav/jJNMXxzLlBXP+oubkgsnbq+/g9pVnLy4Xptwl3094vY00W3+3ztNMP2fb4AB9mMaK2Lo2lJ4HlBmj4iWLcZP7wePwz20T8rQcP0CuFIbQAAAABJRU5ErkJggg==" alt="Creative Commons License" /></a></p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
